<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Peter Selby" />
  <meta name="author" content="Rafael Abbeloos" />
  <meta name="author" content="Anne-Francoise Adam-Blondon" />
  <meta name="author" content="Francisco J. Agosto-Pérez" />
  <meta name="author" content="Michael Alaux" />
  <meta name="author" content="Isabelle Alic" />
  <meta name="author" content="Khaled Al-Shamaa" />
  <meta name="author" content="Johan Steven Aparicio" />
  <meta name="author" content="Jan Erik Backlund" />
  <meta name="author" content="Aldrin Batac" />
  <meta name="author" content="Sebastian Beier" />
  <meta name="author" content="Gabriel Besombes" />
  <meta name="author" content="Alice Boizet" />
  <meta name="author" content="Matthijs Brouwer" />
  <meta name="author" content="Terry Casstevens" />
  <meta name="author" content="Arnaud Charleroy" />
  <meta name="author" content="Keo Corak" />
  <meta name="author" content="Chaney Courtney" />
  <meta name="author" content="Mariano Crimi" />
  <meta name="author" content="Gouripriya Davuluri" />
  <meta name="author" content="Kauê de Sousa" />
  <meta name="author" content="Jeremy Destin" />
  <meta name="author" content="Stijn Dhondt" />
  <meta name="author" content="Ajay Dhungana" />
  <meta name="author" content="Bert Droesbeke" />
  <meta name="author" content="Manuel Feser" />
  <meta name="author" content="Mirella Flores-Gonzalez" />
  <meta name="author" content="Valentin Guignon" />
  <meta name="author" content="Corina Habito" />
  <meta name="author" content="Asis Hallab" />
  <meta name="author" content="Puthick Hok" />
  <meta name="author" content="Lynn Carol Johnson" />
  <meta name="author" content="Sook Jung" />
  <meta name="author" content="Paul Kersey" />
  <meta name="author" content="Andrzej Kilian" />
  <meta name="author" content="Patrick König" />
  <meta name="author" content="Suman Kumar" />
  <meta name="author" content="Josh Lamos-Sweeney" />
  <meta name="author" content="Laszlo Lang" />
  <meta name="author" content="Matthias Lange" />
  <meta name="author" content="Marie-Angélique Laporte" />
  <meta name="author" content="Taein Lee" />
  <meta name="author" content="Erwan Le-Floch" />
  <meta name="author" content="Francisco López" />
  <meta name="author" content="Brandon Madriz" />
  <meta name="author" content="Dorrie Main" />
  <meta name="author" content="Marco Marsella" />
  <meta name="author" content="Maud Marty" />
  <meta name="author" content="Célia Michotey" />
  <meta name="author" content="Zachary Miller" />
  <meta name="author" content="Iain Milne" />
  <meta name="author" content="Lukas A. Mueller" />
  <meta name="author" content="Moses Nderitu" />
  <meta name="author" content="Pascal Neveu" />
  <meta name="author" content="Nick Palladino" />
  <meta name="author" content="Tim Parsons" />
  <meta name="author" content="Cyril Pommier" />
  <meta name="author" content="Jean-François Rami" />
  <meta name="author" content="Sebastian Raubach" />
  <meta name="author" content="Trevor Rife" />
  <meta name="author" content="Kelly Robbins" />
  <meta name="author" content="Mathieu Rouard" />
  <meta name="author" content="Joseph Ruff" />
  <meta name="author" content="Guilhem Sempéré" />
  <meta name="author" content="Romil Mayank Shah" />
  <meta name="author" content="Paul Shaw" />
  <meta name="author" content="Becky Smith" />
  <meta name="author" content="Nahuel Soldevilla" />
  <meta name="author" content="Anne Tireau" />
  <meta name="author" content="Clarysabel Tovar" />
  <meta name="author" content="Grzegorz Uszynski" />
  <meta name="author" content="Vivian Bass Vega" />
  <meta name="author" content="Stephan Weise" />
  <meta name="author" content="Shawn C. Yarnes" />
  <meta name="dcterms.date" content="2024-07-15" />
  <meta name="keywords" content="brapi, breeding, api, standards, community" />
  <title>BrAPI Success Stories</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta property="og:type" content="article" />
  <meta name="dc.title" content="BrAPI Success Stories" />
  <meta name="citation_title" content="BrAPI Success Stories" />
  <meta property="og:title" content="BrAPI Success Stories" />
  <meta property="twitter:title" content="BrAPI Success Stories" />
  <meta name="dc.date" content="2024-07-15" />
  <meta name="citation_publication_date" content="2024-07-15" />
  <meta property="article:published_time" content="2024-07-15" />
  <meta name="dc.modified" content="2024-07-15T16:32:22+00:00" />
  <meta property="article:modified_time" content="2024-07-15T16:32:22+00:00" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Peter Selby" />
  <meta name="citation_author_institution" content="Cornell University" />
  <meta name="citation_author_orcid" content="0000-0001-7151-4445" />
  <meta name="citation_author" content="Rafael Abbeloos" />
  <meta name="citation_author_institution" content="VIB Agro-Incubator" />
  <meta name="citation_author_orcid" content="0000-0002-0177-3887" />
  <meta name="citation_author" content="Anne-Francoise Adam-Blondon" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, 78026, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, 78026, Versailles, France" />
  <meta name="citation_author_orcid" content="0000-0002-3412-9086" />
  <meta name="citation_author" content="Francisco J. Agosto-Pérez" />
  <meta name="citation_author_institution" content="Cornell University" />
  <meta name="citation_author_orcid" content="0000-0002-5059-253X" />
  <meta name="citation_author" content="Michael Alaux" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, 78026, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, 78026, Versailles, France" />
  <meta name="citation_author_orcid" content="0000-0001-9356-4072" />
  <meta name="citation_author" content="Isabelle Alic" />
  <meta name="citation_author_institution" content="MISTEA, University of Montpellier, INRAE, Institut Agro, Montpellier, France" />
  <meta name="citation_author_orcid" content="0000-0002-8961-6068" />
  <meta name="citation_author" content="Khaled Al-Shamaa" />
  <meta name="citation_author_institution" content="ICARDA" />
  <meta name="citation_author_orcid" content="0000-0002-7668-3798" />
  <meta name="citation_author" content="Johan Steven Aparicio" />
  <meta name="citation_author_institution" content="Alliance Bioversity-CIAT" />
  <meta name="citation_author" content="Jan Erik Backlund" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author" content="Aldrin Batac" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author_institution" content="Leafnode LLC" />
  <meta name="citation_author" content="Sebastian Beier" />
  <meta name="citation_author_institution" content="Institute of Bio- and Geosciences (IBG-4: Bioinformatics), CEPLAS, Forschungszentrum Jülich GmbH, Wilhelm Johnen Straße, 52428 Jülich, Germany" />
  <meta name="citation_author_institution" content="Bioeconomy Science Center (BioSC), Forschungszentrum Jülich GmbH, 52428 Jülich, Germany" />
  <meta name="citation_author_orcid" content="0000-0002-2177-8781" />
  <meta name="citation_author" content="Gabriel Besombes" />
  <meta name="citation_author_institution" content="MISTEA, University of Montpellier, INRAE, Institut Agro, Montpellier, France" />
  <meta name="citation_author_orcid" content="0009-0004-1359-2145" />
  <meta name="citation_author" content="Alice Boizet" />
  <meta name="citation_author_institution" content="CIRAD, UMR AGAP Institut, Montpellier, France." />
  <meta name="citation_author_institution" content="AGAP Institut, CIRAD, INRAE, Institut Agro, Université de Montpellier, Montpellier, France." />
  <meta name="citation_author_orcid" content="0000-0003-4096-6689" />
  <meta name="citation_author" content="Matthijs Brouwer" />
  <meta name="citation_author_institution" content="Wageningen University and Research" />
  <meta name="citation_author_orcid" content="0000-0001-8183-0484" />
  <meta name="citation_author" content="Terry Casstevens" />
  <meta name="citation_author_institution" content="Buckler Lab and Institute for Genomic Diversity, Cornell University" />
  <meta name="citation_author_orcid" content="0000-0001-7602-0487" />
  <meta name="citation_author" content="Arnaud Charleroy" />
  <meta name="citation_author_institution" content="MISTEA, University of Montpellier, INRAE, Institut Agro, Montpellier, France" />
  <meta name="citation_author_orcid" content="0009-0009-7575-3617" />
  <meta name="citation_author" content="Keo Corak" />
  <meta name="citation_author_institution" content="USDA-ARS Genomics and Bioinformatics Research Unit" />
  <meta name="citation_author_orcid" content="0000-0002-4129-3319" />
  <meta name="citation_author" content="Chaney Courtney" />
  <meta name="citation_author_institution" content="Clemson University" />
  <meta name="citation_author" content="Mariano Crimi" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author" content="Gouripriya Davuluri" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author_orcid" content="0009-0005-2308-0773" />
  <meta name="citation_author" content="Kauê de Sousa" />
  <meta name="citation_author_institution" content="Digital Inclusion, Bioversity International, Montpellier, France" />
  <meta name="citation_author_orcid" content="0000-0002-7571-7845" />
  <meta name="citation_author" content="Jeremy Destin" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, 78026, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, 78026, Versailles, France" />
  <meta name="citation_author" content="Stijn Dhondt" />
  <meta name="citation_author_institution" content="VIB AgroIncubator" />
  <meta name="citation_author_orcid" content="0000-0003-4402-2191" />
  <meta name="citation_author" content="Ajay Dhungana" />
  <meta name="citation_author_institution" content="Louisiana State University (LSU)" />
  <meta name="citation_author" content="Bert Droesbeke" />
  <meta name="citation_author_institution" content="VIB Data Core" />
  <meta name="citation_author_orcid" content="0000-0003-0522-5674" />
  <meta name="citation_author" content="Manuel Feser" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author_institution" content="Graduate School DILS, Bielefeld Institute for Bioinformatics Infrastructure (BIBI)" />
  <meta name="citation_author_orcid" content="0000-0001-6546-1818" />
  <meta name="citation_author" content="Mirella Flores-Gonzalez" />
  <meta name="citation_author_institution" content="Boyce Thompson Institute" />
  <meta name="citation_author_orcid" content="0000-0002-7759-1617" />
  <meta name="citation_author" content="Valentin Guignon" />
  <meta name="citation_author_institution" content="Bioversity International, Parc Scientifique Agropolis II, 34397 Montpellier, France" />
  <meta name="citation_author_orcid" content="0000-0003-0903-6811" />
  <meta name="citation_author" content="Corina Habito" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author" content="Asis Hallab" />
  <meta name="citation_author_institution" content="Jülich research center, Institute of Bio- and Geosciences (IBG), Bioinformatics (IBG-4) and Bingen Technical University of Applied Sciences, Germany" />
  <meta name="citation_author_orcid" content="0000-0002-2421-5431" />
  <meta name="citation_author" content="Puthick Hok" />
  <meta name="citation_author_institution" content="Diversity Arrays Technology (DArT)" />
  <meta name="citation_author" content="Lynn Carol Johnson" />
  <meta name="citation_author_institution" content="Buckler Lab and Institute for Genomic Diversity, Cornell University" />
  <meta name="citation_author_orcid" content="0000-0001-8103-2722" />
  <meta name="citation_author" content="Sook Jung" />
  <meta name="citation_author_institution" content="Department of Horticulture, Washington State University" />
  <meta name="citation_author_orcid" content="0000-0003-3968-2769" />
  <meta name="citation_author" content="Paul Kersey" />
  <meta name="citation_author_institution" content="Royal Botanic Gardens, Kew" />
  <meta name="citation_author" content="Andrzej Kilian" />
  <meta name="citation_author_institution" content="Diversity Arrays Technology (DArT)" />
  <meta name="citation_author" content="Patrick König" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author_orcid" content="0000-0002-8948-6793" />
  <meta name="citation_author" content="Suman Kumar" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author_orcid" content="0009-0005-5832-7190" />
  <meta name="citation_author" content="Josh Lamos-Sweeney" />
  <meta name="citation_author_institution" content="Cornell University" />
  <meta name="citation_author" content="Laszlo Lang" />
  <meta name="citation_author_institution" content="Bingen Technical University of Applied Sciences, Berlinstraße 109, 55411 Bingen am Rhein, Germany" />
  <meta name="citation_author_orcid" content="0009-0009-8936-4532" />
  <meta name="citation_author" content="Matthias Lange" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author_orcid" content="0000-0002-4316-078X" />
  <meta name="citation_author" content="Marie-Angélique Laporte" />
  <meta name="citation_author_institution" content="Digital Inclusion, Bioversity International, Montpellier, France." />
  <meta name="citation_author_orcid" content="0000-0002-8461-9745" />
  <meta name="citation_author" content="Taein Lee" />
  <meta name="citation_author_institution" content="Department of Horticulture, Washington State University" />
  <meta name="citation_author_orcid" content="0000-0002-2000-2586" />
  <meta name="citation_author" content="Erwan Le-Floch" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, 78026, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, 78026, Versailles, France" />
  <meta name="citation_author_orcid" content="0000-0002-1010-6859" />
  <meta name="citation_author" content="Francisco López" />
  <meta name="citation_author_institution" content="International Treaty on Plant Genetic Resources for Food and Agriculture, FAO" />
  <meta name="citation_author" content="Brandon Madriz" />
  <meta name="citation_author_institution" content="MrBot Software Solutions, Cartago, Costa Rica" />
  <meta name="citation_author" content="Dorrie Main" />
  <meta name="citation_author_institution" content="Department of Horticulture, Washington State University" />
  <meta name="citation_author_orcid" content="0000-0002-1162-2724" />
  <meta name="citation_author" content="Marco Marsella" />
  <meta name="citation_author_institution" content="International Treaty on Plant Genetic Resources for Food and Agriculture, FAO" />
  <meta name="citation_author_orcid" content="0000-0003-0334-8785" />
  <meta name="citation_author" content="Maud Marty" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, 78026, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, 78026, Versailles, France" />
  <meta name="citation_author_orcid" content="0009-0001-8764-6466" />
  <meta name="citation_author" content="Célia Michotey" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, 78026, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, 78026, Versailles, France" />
  <meta name="citation_author_orcid" content="0000-0003-1877-1703" />
  <meta name="citation_author" content="Zachary Miller" />
  <meta name="citation_author_institution" content="Buckler Lab and Institute for Genomic Diversity, Cornell University" />
  <meta name="citation_author_orcid" content="0000-0002-5454-4527" />
  <meta name="citation_author" content="Iain Milne" />
  <meta name="citation_author_institution" content="The James Hutton Institute" />
  <meta name="citation_author_orcid" content="0000-0002-4126-0859" />
  <meta name="citation_author" content="Lukas A. Mueller" />
  <meta name="citation_author_institution" content="The Boyce Thompson Institute" />
  <meta name="citation_author_orcid" content="0000-0001-8640-1750" />
  <meta name="citation_author" content="Moses Nderitu" />
  <meta name="citation_author_institution" content="SEQART AFRICA" />
  <meta name="citation_author" content="Pascal Neveu" />
  <meta name="citation_author_institution" content="MISTEA, University of Montpellier, INRAE, Institut Agro, Montpellier, France" />
  <meta name="citation_author_orcid" content="0000-0003-4189-7793" />
  <meta name="citation_author" content="Nick Palladino" />
  <meta name="citation_author_institution" content="Breeding Insight, Cornell University" />
  <meta name="citation_author_orcid" content="0009-0009-1645-297X" />
  <meta name="citation_author" content="Tim Parsons" />
  <meta name="citation_author_institution" content="Breeding Insight, Cornell University" />
  <meta name="citation_author_orcid" content="0000-0001-6209-2455" />
  <meta name="citation_author" content="Cyril Pommier" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, 78026, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, 78026, Versailles, France" />
  <meta name="citation_author_orcid" content="0000-0002-9040-8733" />
  <meta name="citation_author" content="Jean-François Rami" />
  <meta name="citation_author_institution" content="CIRAD, UMR AGAP Institut, Montpellier, France." />
  <meta name="citation_author_institution" content="AGAP Institut, CIRAD, INRAE, Institut Agro, Université de Montpellier, Montpellier, France." />
  <meta name="citation_author_orcid" content="0000-0002-5679-3877" />
  <meta name="citation_author" content="Sebastian Raubach" />
  <meta name="citation_author_institution" content="The James Hutton Institute" />
  <meta name="citation_author_orcid" content="0000-0001-5659-247X" />
  <meta name="citation_author" content="Trevor Rife" />
  <meta name="citation_author_institution" content="Clemson University" />
  <meta name="citation_author_orcid" content="0000-0002-5974-6523" />
  <meta name="citation_author" content="Kelly Robbins" />
  <meta name="citation_author_institution" content="Cornell University" />
  <meta name="citation_author_orcid" content="0000-0001-9522-9585" />
  <meta name="citation_author" content="Mathieu Rouard" />
  <meta name="citation_author_institution" content="Bioversity International, Parc Scientifique Agropolis II, 34397 Montpellier, France" />
  <meta name="citation_author_orcid" content="0000-0003-0284-1885" />
  <meta name="citation_author" content="Joseph Ruff" />
  <meta name="citation_author_institution" content="Royal Botanic Gardens, Kew" />
  <meta name="citation_author" content="Guilhem Sempéré" />
  <meta name="citation_author_institution" content="CIRAD, UMR INTERTRYP, Montpellier, France INTERTRYP, Univ Montpellier, CIRAD, IRD, Montpellier, France French Institute of Bioinformatics (IFB)" />
  <meta name="citation_author_institution" content="South Green Bioinformatics Platform, Bioversity, CIRAD, INRAE, IRD, Montpellier, France" />
  <meta name="citation_author_orcid" content="0000-0001-7429-2091" />
  <meta name="citation_author" content="Romil Mayank Shah" />
  <meta name="citation_author_institution" content="North Carolina State University" />
  <meta name="citation_author" content="Paul Shaw" />
  <meta name="citation_author_institution" content="The James Hutton Institute" />
  <meta name="citation_author_orcid" content="0000-0002-0202-1150" />
  <meta name="citation_author" content="Becky Smith" />
  <meta name="citation_author_institution" content="The James Hutton Institute" />
  <meta name="citation_author_orcid" content="0000-0002-8968-3383" />
  <meta name="citation_author" content="Nahuel Soldevilla" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author_institution" content="Leafnode LLC" />
  <meta name="citation_author" content="Anne Tireau" />
  <meta name="citation_author_institution" content="MISTEA, University of Montpellier, INRAE, Institut Agro, Montpellier, France" />
  <meta name="citation_author_orcid" content="0000-0001-8501-6922" />
  <meta name="citation_author" content="Clarysabel Tovar" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author_institution" content="Leafnode LLC" />
  <meta name="citation_author" content="Grzegorz Uszynski" />
  <meta name="citation_author_institution" content="Diversity Arrays Technology (DArT)" />
  <meta name="citation_author" content="Vivian Bass Vega" />
  <meta name="citation_author_institution" content="Bingen Technical University of Applied Sciences, Berlinstraße 109, 55411 Bingen am Rhein, Germany" />
  <meta name="citation_author_orcid" content="0009-0002-2476-9888" />
  <meta name="citation_author" content="Stephan Weise" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author_orcid" content="0000-0003-4031-9131" />
  <meta name="citation_author" content="Shawn C. Yarnes" />
  <meta name="citation_author_institution" content="Breeding Insight, Cornell University" />
  <meta name="citation_author_orcid" content="0000-0002-1090-0403" />
  <link rel="canonical" href="https://plantbreeding.github.io/BrAPI-Manuscript2/" />
  <meta property="og:url" content="https://plantbreeding.github.io/BrAPI-Manuscript2/" />
  <meta property="twitter:url" content="https://plantbreeding.github.io/BrAPI-Manuscript2/" />
  <meta name="citation_fulltext_html_url" content="https://plantbreeding.github.io/BrAPI-Manuscript2/" />
  <meta name="citation_pdf_url" content="https://plantbreeding.github.io/BrAPI-Manuscript2/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://plantbreeding.github.io/BrAPI-Manuscript2/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://plantbreeding.github.io/BrAPI-Manuscript2/v/511faa4c28e126a274c0f733c9fcaab35c0dfe49/" />
  <meta name="manubot_html_url_versioned" content="https://plantbreeding.github.io/BrAPI-Manuscript2/v/511faa4c28e126a274c0f733c9fcaab35c0dfe49/" />
  <meta name="manubot_pdf_url_versioned" content="https://plantbreeding.github.io/BrAPI-Manuscript2/v/511faa4c28e126a274c0f733c9fcaab35c0dfe49/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">BrAPI Success Stories</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://plantbreeding.github.io/BrAPI-Manuscript2/v/511faa4c28e126a274c0f733c9fcaab35c0dfe49/">permalink</a>)
was automatically generated
from <a href="https://github.com/plantbreeding/BrAPI-Manuscript2/tree/511faa4c28e126a274c0f733c9fcaab35c0dfe49">plantbreeding/BrAPI-Manuscript2@511faa4</a>
on July 15, 2024.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Peter Selby</strong>
<sup><a href="#correspondence">✉</a></sup><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7151-4445">0000-0001-7151-4445</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/BrapiCoordinatorSelby">BrapiCoordinatorSelby</a>
<br>
<small>
Cornell University
· Funded by The BrAPI Project is funded by the USDA grant NIFA-DSFAS 2022-67021-37024.
</small></p></li>
<li><p><strong>Rafael Abbeloos</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-0177-3887">0000-0002-0177-3887</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/raabb">raabb</a>
<br>
<small>
VIB Agro-Incubator
</small></p></li>
<li><p><strong>Anne-Francoise Adam-Blondon</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-3412-9086">0000-0002-3412-9086</a>
<br>
<small>
Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, 78026, Versailles, France; Université Paris-Saclay, INRAE, URGI, 78026, Versailles, France
</small></p></li>
<li><p><strong>Francisco J. Agosto-Pérez</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-5059-253X">0000-0002-5059-253X</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/agostof">agostof</a>
<br>
<small>
Cornell University
· Funded by Innovation Lab for Crop Improvement Cornell, FFAR
</small></p></li>
<li><p><strong>Michael Alaux</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-9356-4072">0000-0001-9356-4072</a>
<br>
<small>
Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, 78026, Versailles, France; Université Paris-Saclay, INRAE, URGI, 78026, Versailles, France
· Funded by The AGENT project is funded by the European Union’s Horizon 2020 research and innovation programme under grant agreement no. 862613.
</small></p></li>
<li><p><strong>Isabelle Alic</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-8961-6068">0000-0002-8961-6068</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/Isabelle-inrae">Isabelle-inrae</a>
<br>
<small>
MISTEA, University of Montpellier, INRAE, Institut Agro, Montpellier, France
</small></p></li>
<li><p><strong>Khaled Al-Shamaa</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-7668-3798">0000-0002-7668-3798</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/khaled-alshamaa">khaled-alshamaa</a>
<br>
<small>
ICARDA
</small></p></li>
<li><p><strong>Johan Steven Aparicio</strong>
<br>
<small>
Alliance Bioversity-CIAT
</small></p></li>
<li><p><strong>Jan Erik Backlund</strong>
<br>
<small>
Integrated Breeding Platform
</small></p></li>
<li><p><strong>Aldrin Batac</strong>
<br>
<small>
Integrated Breeding Platform; Leafnode LLC
</small></p></li>
<li><p><strong>Sebastian Beier</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2177-8781">0000-0002-2177-8781</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/sebeier">sebeier</a>
<br>
<small>
Institute of Bio- and Geosciences (IBG-4: Bioinformatics), CEPLAS, Forschungszentrum Jülich GmbH, Wilhelm Johnen Straße, 52428 Jülich, Germany; Bioeconomy Science Center (BioSC), Forschungszentrum Jülich GmbH, 52428 Jülich, Germany
</small></p></li>
<li><p><strong>Gabriel Besombes</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0009-0004-1359-2145">0009-0004-1359-2145</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/Gabriel-Besombes">Gabriel-Besombes</a>
<br>
<small>
MISTEA, University of Montpellier, INRAE, Institut Agro, Montpellier, France
</small></p></li>
<li><p><strong>Alice Boizet</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-4096-6689">0000-0003-4096-6689</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/aliceboizet">aliceboizet</a>
<br>
<small>
CIRAD, UMR AGAP Institut, Montpellier, France.; AGAP Institut, CIRAD, INRAE, Institut Agro, Université de Montpellier, Montpellier, France.
</small></p></li>
<li><p><strong>Matthijs Brouwer</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-8183-0484">0000-0001-8183-0484</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/matthijsbrouwer">matthijsbrouwer</a>
<br>
<small>
Wageningen University and Research
</small></p></li>
<li><p><strong>Terry Casstevens</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7602-0487">0000-0001-7602-0487</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/tcasstevens">tcasstevens</a>
<br>
<small>
Buckler Lab and Institute for Genomic Diversity, Cornell University
</small></p></li>
<li><p><strong>Arnaud Charleroy</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0009-0009-7575-3617">0009-0009-7575-3617</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/niio972">niio972</a>
<br>
<small>
MISTEA, University of Montpellier, INRAE, Institut Agro, Montpellier, France
</small></p></li>
<li><p><strong>Keo Corak</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-4129-3319">0000-0002-4129-3319</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/keocorak">keocorak</a>
<br>
<small>
USDA-ARS Genomics and Bioinformatics Research Unit
</small></p></li>
<li><p><strong>Chaney Courtney</strong>
<br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/chaneylc">chaneylc</a>
<br>
<small>
Clemson University
</small></p></li>
<li><p><strong>Mariano Crimi</strong>
<br>
<small>
Integrated Breeding Platform
</small></p></li>
<li><p><strong>Gouripriya Davuluri</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0009-0005-2308-0773">0009-0005-2308-0773</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/Gouripriya5">Gouripriya5</a>
<br>
<small>
Leibniz Institute of Plant Genetics and Crop Plant Research
· Funded by The AGENT project is funded by the European Union’s Horizon 2020 research and innovation programme under grant agreement no. 862613.
</small></p></li>
<li><p><strong>Kauê de Sousa</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-7571-7845">0000-0002-7571-7845</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/kauedesousa">kauedesousa</a>
<br>
<small>
Digital Inclusion, Bioversity International, Montpellier, France
</small></p></li>
<li><p><strong>Jeremy Destin</strong>
<br>
<small>
Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, 78026, Versailles, France; Université Paris-Saclay, INRAE, URGI, 78026, Versailles, France
</small></p></li>
<li><p><strong>Stijn Dhondt</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-4402-2191">0000-0003-4402-2191</a>
<br>
<small>
VIB AgroIncubator
</small></p></li>
<li><p><strong>Ajay Dhungana</strong>
<br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/GenAjay">GenAjay</a>
<br>
<small>
Louisiana State University (LSU)
</small></p></li>
<li><p><strong>Bert Droesbeke</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-0522-5674">0000-0003-0522-5674</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/bedroesb">bedroesb</a>
<br>
<small>
VIB Data Core
</small></p></li>
<li><p><strong>Manuel Feser</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-6546-1818">0000-0001-6546-1818</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/feserm">feserm</a>
<br>
<small>
Leibniz Institute of Plant Genetics and Crop Plant Research; Graduate School DILS, Bielefeld Institute for Bioinformatics Infrastructure (BIBI)
</small></p></li>
<li><p><strong>Mirella Flores-Gonzalez</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-7759-1617">0000-0002-7759-1617</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/mflores202">mflores202</a>
<br>
<small>
Boyce Thompson Institute
</small></p></li>
<li><p><strong>Valentin Guignon</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-0903-6811">0000-0003-0903-6811</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/guignonv">guignonv</a>
<br>
<small>
Bioversity International, Parc Scientifique Agropolis II, 34397 Montpellier, France
</small></p></li>
<li><p><strong>Corina Habito</strong>
<br>
<small>
Integrated Breeding Platform
</small></p></li>
<li><p><strong>Asis Hallab</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2421-5431">0000-0002-2421-5431</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/asishallab">asishallab</a>
<br>
<small>
Jülich research center, Institute of Bio- and Geosciences (IBG), Bioinformatics (IBG-4) and Bingen Technical University of Applied Sciences, Germany
</small></p></li>
<li><p><strong>Puthick Hok</strong>
<br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/puthick">puthick</a>
<br>
<small>
Diversity Arrays Technology (DArT)
</small></p></li>
<li><p><strong>Lynn Carol Johnson</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-8103-2722">0000-0001-8103-2722</a>
<br>
<small>
Buckler Lab and Institute for Genomic Diversity, Cornell University
</small></p></li>
<li><p><strong>Sook Jung</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-3968-2769">0000-0003-3968-2769</a>
<br>
<small>
Department of Horticulture, Washington State University
· Funded by USDA National Institute of Food and Agriculture National Research Support Project 10
</small></p></li>
<li><p><strong>Paul Kersey</strong>
<br>
<small>
Royal Botanic Gardens, Kew
· Funded by The AGENT project is funded by the European Union’s Horizon 2020 research and innovation programme under grant agreement no. 862613.
</small></p></li>
<li><p><strong>Andrzej Kilian</strong>
<br>
<small>
Diversity Arrays Technology (DArT)
</small></p></li>
<li><p><strong>Patrick König</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-8948-6793">0000-0002-8948-6793</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/patrick-koenig">patrick-koenig</a>
<br>
<small>
Leibniz Institute of Plant Genetics and Crop Plant Research
· Funded by The AGENT project is funded by the European Union’s Horizon 2020 research and innovation programme under grant agreement no. 862613.
</small></p></li>
<li><p><strong>Suman Kumar</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0009-0005-5832-7190">0009-0005-5832-7190</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/sumankumar1">sumankumar1</a>
<br>
<small>
Leibniz Institute of Plant Genetics and Crop Plant Research
· Funded by The AGENT project is funded by the European Union’s Horizon 2020 research and innovation programme under grant agreement no. 862613.
</small></p></li>
<li><p><strong>Josh Lamos-Sweeney</strong>
<br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/jlamossweeney">jlamossweeney</a>
<br>
<small>
Cornell University
</small></p></li>
<li><p><strong>Laszlo Lang</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0009-0009-8936-4532">0009-0009-8936-4532</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/LzLang">LzLang</a>
<br>
<small>
Bingen Technical University of Applied Sciences, Berlinstraße 109, 55411 Bingen am Rhein, Germany
</small></p></li>
<li><p><strong>Matthias Lange</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-4316-078X">0000-0002-4316-078X</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/langeipk">langeipk</a>
<br>
<small>
Leibniz Institute of Plant Genetics and Crop Plant Research
· Funded by The AGENT project is funded by the European Union’s Horizon 2020 research and innovation programme under grant agreement no. 862613.
</small></p></li>
<li><p><strong>Marie-Angélique Laporte</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-8461-9745">0000-0002-8461-9745</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/marieALaporte">marieALaporte</a>
<br>
<small>
Digital Inclusion, Bioversity International, Montpellier, France.
</small></p></li>
<li><p><strong>Taein Lee</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2000-2586">0000-0002-2000-2586</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/leetaei">leetaei</a>
<br>
<small>
Department of Horticulture, Washington State University
</small></p></li>
<li><p><strong>Erwan Le-Floch</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-1010-6859">0000-0002-1010-6859</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/erlefloch">erlefloch</a>
<br>
<small>
Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, 78026, Versailles, France; Université Paris-Saclay, INRAE, URGI, 78026, Versailles, France
</small></p></li>
<li><p><strong>Francisco López</strong>
<br>
<small>
International Treaty on Plant Genetic Resources for Food and Agriculture, FAO
</small></p></li>
<li><p><strong>Brandon Madriz</strong>
<br>
<small>
MrBot Software Solutions, Cartago, Costa Rica
</small></p></li>
<li><p><strong>Dorrie Main</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-1162-2724">0000-0002-1162-2724</a>
<br>
<small>
Department of Horticulture, Washington State University
· Funded by {‘USDA NIFA’: ‘2022-51181-38449’}
</small></p></li>
<li><p><strong>Marco Marsella</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-0334-8785">0000-0003-0334-8785</a>
<br>
<small>
International Treaty on Plant Genetic Resources for Food and Agriculture, FAO
</small></p></li>
<li><p><strong>Maud Marty</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0009-0001-8764-6466">0009-0001-8764-6466</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/maudmarty">maudmarty</a>
<br>
<small>
Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, 78026, Versailles, France; Université Paris-Saclay, INRAE, URGI, 78026, Versailles, France
</small></p></li>
<li><p><strong>Célia Michotey</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-1877-1703">0000-0003-1877-1703</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/cmichotey">cmichotey</a>
<br>
<small>
Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, 78026, Versailles, France; Université Paris-Saclay, INRAE, URGI, 78026, Versailles, France
</small></p></li>
<li><p><strong>Zachary Miller</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-5454-4527">0000-0002-5454-4527</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/zrm22">zrm22</a>
<br>
<small>
Buckler Lab and Institute for Genomic Diversity, Cornell University
</small></p></li>
<li><p><strong>Iain Milne</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-4126-0859">0000-0002-4126-0859</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/imilne">imilne</a>
<br>
<small>
The James Hutton Institute
</small></p></li>
<li><p><strong>Lukas A. Mueller</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-8640-1750">0000-0001-8640-1750</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/lukasmueller">lukasmueller</a>
<br>
<small>
The Boyce Thompson Institute
</small></p></li>
<li><p><strong>Moses Nderitu</strong>
<br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/jndmose">jndmose</a>
<br>
<small>
SEQART AFRICA
</small></p></li>
<li><p><strong>Pascal Neveu</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-4189-7793">0000-0003-4189-7793</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/pascalneveu">pascalneveu</a>
<br>
<small>
MISTEA, University of Montpellier, INRAE, Institut Agro, Montpellier, France
· Funded by This work was supported by the Agence Nationale de la Recherche, programme Investissements d’avenir, ANR-11-INBS-0012 (Phenome)
</small></p></li>
<li><p><strong>Nick Palladino</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0009-0009-1645-297X">0009-0009-1645-297X</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/nickpalladino">nickpalladino</a>
<br>
<small>
Breeding Insight, Cornell University
· Funded by U.S. Department of Agriculture, under agreement numbers [8062-21000-043-004-A, 8062-21000-052-002-A, and 8062-21000-052-003-A]
</small></p></li>
<li><p><strong>Tim Parsons</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-6209-2455">0000-0001-6209-2455</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/timparsons">timparsons</a>
<br>
<small>
Breeding Insight, Cornell University
· Funded by U.S. Department of Agriculture, under agreement numbers [8062-21000-043-004-A, 8062-21000-052-002-A, and 8062-21000-052-003-A]
</small></p></li>
<li><p><strong>Cyril Pommier</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-9040-8733">0000-0002-9040-8733</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/cpommier">cpommier</a>
<br>
<small>
Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, 78026, Versailles, France; Université Paris-Saclay, INRAE, URGI, 78026, Versailles, France
</small></p></li>
<li><p><strong>Jean-François Rami</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-5679-3877">0000-0002-5679-3877</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/jframi">jframi</a>
<br>
<small>
CIRAD, UMR AGAP Institut, Montpellier, France.; AGAP Institut, CIRAD, INRAE, Institut Agro, Université de Montpellier, Montpellier, France.
</small></p></li>
<li><p><strong>Sebastian Raubach</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-5659-247X">0000-0001-5659-247X</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/sebastian-raubach">sebastian-raubach</a>
<br>
<small>
The James Hutton Institute
</small></p></li>
<li><p><strong>Trevor Rife</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-5974-6523">0000-0002-5974-6523</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/trife">trife</a>
<br>
<small>
Clemson University
</small></p></li>
<li><p><strong>Kelly Robbins</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-9522-9585">0000-0001-9522-9585</a>
<br>
<small>
Cornell University
· Funded by The Bill and Malinda Gates Foundation in cooperation with the Excellence in Breeding Platform of the CGIAR
</small></p></li>
<li><p><strong>Mathieu Rouard</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-0284-1885">0000-0003-0284-1885</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/mrouard">mrouard</a>
<br>
<small>
Bioversity International, Parc Scientifique Agropolis II, 34397 Montpellier, France
</small></p></li>
<li><p><strong>Joseph Ruff</strong>
<br>
<small>
Royal Botanic Gardens, Kew
· Funded by The AGENT project is funded by the European Union’s Horizon 2020 research and innovation programme under grant agreement no. 862613.
</small></p></li>
<li><p><strong>Guilhem Sempéré</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7429-2091">0000-0001-7429-2091</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/GuilhemSempere">GuilhemSempere</a>
<br>
<small>
CIRAD, UMR INTERTRYP, Montpellier, France INTERTRYP, Univ Montpellier, CIRAD, IRD, Montpellier, France French Institute of Bioinformatics (IFB); South Green Bioinformatics Platform, Bioversity, CIRAD, INRAE, IRD, Montpellier, France
</small></p></li>
<li><p><strong>Romil Mayank Shah</strong>
<br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/romil2807">romil2807</a>
<br>
<small>
North Carolina State University
</small></p></li>
<li><p><strong>Paul Shaw</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-0202-1150">0000-0002-0202-1150</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/cardinalb">cardinalb</a>
<br>
<small>
The James Hutton Institute
</small></p></li>
<li><p><strong>Becky Smith</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-8968-3383">0000-0002-8968-3383</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/Batbaby91">Batbaby91</a>
<br>
<small>
The James Hutton Institute
</small></p></li>
<li><p><strong>Nahuel Soldevilla</strong>
<br>
<small>
Integrated Breeding Platform; Leafnode LLC
</small></p></li>
<li><p><strong>Anne Tireau</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-8501-6922">0000-0001-8501-6922</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/annetireau">annetireau</a>
<br>
<small>
MISTEA, University of Montpellier, INRAE, Institut Agro, Montpellier, France
</small></p></li>
<li><p><strong>Clarysabel Tovar</strong>
<br>
<small>
Integrated Breeding Platform; Leafnode LLC
</small></p></li>
<li><p><strong>Grzegorz Uszynski</strong>
<br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/grzegorz69">grzegorz69</a>
<br>
<small>
Diversity Arrays Technology (DArT)
</small></p></li>
<li><p><strong>Vivian Bass Vega</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0009-0002-2476-9888">0009-0002-2476-9888</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/VivianBass">VivianBass</a>
<br>
<small>
Bingen Technical University of Applied Sciences, Berlinstraße 109, 55411 Bingen am Rhein, Germany
</small></p></li>
<li><p><strong>Stephan Weise</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-4031-9131">0000-0003-4031-9131</a>
<br>
<small>
Leibniz Institute of Plant Genetics and Crop Plant Research
· Funded by The AGENT project is funded by the European Union’s Horizon 2020 research and innovation programme under grant agreement no. 862613.
</small></p></li>
<li><p><strong>Shawn C. Yarnes</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-1090-0403">0000-0002-1090-0403</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/SCYarnes">SCYarnes</a>
<br>
<small>
Breeding Insight, Cornell University
· Funded by U.S. Department of Agriculture, under agreement numbers [8062-21000-043-004-A, 8062-21000-052-002-A, and 8062-21000-052-003-A]
</small></p></li>
</ul>
<div id="correspondence">
<p>✉ — Correspondence possible via <a href="https://github.com/plantbreeding/BrAPI-Manuscript2/issues">GitHub Issues</a>
or email to
Peter Selby &lt;ps664@cornell.edu&gt;.</p>
</div>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<p>Population growth and climate change require extraordinary efforts to increase efficiency in breeding programs around the world. In the last few years, new phenotyping techniques, genomics technologies, and genetic approaches such as genomic prediction have provided a boost in genetic gain in breeding, but have also created a flood of data that needs careful management to be fully harnessed. In particular, data integration is a challenge due to the multiple types of data being handled by a variety of disparate and dispersed systems.
The Breeding API (BrAPI) project is an international, grass-roots effort to enable more efficient data management by enabling interoperability among research databases and tools, using a standardized RESTful web service API specification for exchanging breeding related data.
This community driven standard is software agnostic and free to be used by anyone interested in plant breeding, genetics and agronomy data management, including trial, germplasm, phenotyping, and genotyping data management.
This manuscript presents the current version of BrAPI, the substantial growth of the project, and a wide variety of open source plant genetics research tools with active BrAPI implementations.</p>
<h2 id="introduction">Introduction</h2>
<!-- 
Notes:
* DIGITAL ECOSYSTEM
* Justification for a second paper
* Highlight Community Growth
* Hackathons
* sabbaticals
* Updates for V2
* Auth Data Access Control - super simplified - highlight data privacy and security elements
* Project structure - Modules
* Standardization of endpoints across objects - said in a non-tech way
-->
<p>To address consequences of climate change and population growth, plant and animal breeding needs to become more efficient and data driven to ensure a healthy, resilient, and sustainable agricultural production system. Modern breeding techniques require large amounts of high quality data to be effective, requiring digital methods for data collection, management, and analysis. They rely on several research disciplines, including plant phenomics, genetics, genomics, and agronomy, involving research institutes, genebanks, and breeding companies. Interoperability between research software tools, systems, and databases can substantially increase the efficiency of a breeding program. The ability to efficiently share data means access to larger and more complete datasets, enabling more accurate computational models, more accurate predictions, and improved selections.</p>
<p>The Breeding API (BrAPI) project is an effort to enable interoperability among breeding tools, systems, and databases. BrAPI is a standardized Representational State Transfer (REST), web service, Application Programming Interface (API), specification for breeding, genetics, phenomics and related agricultural data.<span class="citation" data-cites="QfGEJ6km"><sup><a href="#ref-QfGEJ6km" role="doc-biblioref">1</a></sup></span> By using the BrAPI standard, breeding software can more easily become interoperable, allowing groups to more easily share data and software tools.</p>
<p>Since its first publication<span class="citation" data-cites="QfGEJ6km"><sup><a href="#ref-QfGEJ6km" role="doc-biblioref">1</a></sup></span>, BrAPI has seen a significant increase in community services, compatible tools, and participating organizations. The community has organized numerous hackathons to evolve the specifications, resulting in continuous improvements and enhancements. This report includes a short technical description of the standard and a showcase of the applications, services, and tools available in the BrAPI community. BrAPI has become an essential part of the digital infrastructure for breeding, genetics and phenomics applications and related agricultural projects. It is the intention of this manuscript to demonstrate the value of BrAPI to the wider scientific community as an effective and efficient means to collaborate and share resources.</p>
<h3 id="how-it-works">How it works</h3>
<p>An API is a technical connection between two pieces of software. Just as a Graphical User Interface (GUI) or a Command Line Interface (CLI) allows a human user to interact with a piece of software, an API allows one software application to interact with another. A GUI or CLI might allow a user to input data, read data, and start processes within an application. An API allows one piece of software (sometimes called a client, user agent, or service consumer) to programmatically input data, read data, and start process within another piece of software (sometimes called a server or service provider).</p>
<p>A REST style web service is a type of API commonly used in today’s web infrastructure. REST is a technical architecture that describes the stateless transmission of data between applications. Typically, RESTful web service APIs are implemented using the standard HTTP protocol that most of the modern internet is built upon. These implementations generally use JavaScript Object Notation (JSON) to represent the data being transferred. Both HTTP and JSON are programming language agnostic, very stable, and very flexible. This means BrAPI can be implemented in almost any piece of software, and can solve a wide range of use cases.</p>
<p>Data repositories and service providers can choose to represent their data as a BrAPI compatible API. By mapping their internal data structures to the standard models, data repositories can easily expose data to the outside world. Similarly, they can accept new data from external sources and automatically map the new data into an existing database. Client application developers can take advantage of this standardization by building tools and connectors that integrate with all BrAPI compatible data repositories. Visualization, reporting, analytics, data collection, and quality control tools can be built once and shared with other organizations following the standards. As the number of BrAPI compatible databases, tools, and organizations grows, so does the value added by implementing the standard into a given application.</p>
<h3 id="project-updates">Project Updates</h3>
<p>Over its lifetime, the BrAPI project has grown and changed substantially. The latest stable version of the specification (v2.1) looks vastly different from the first version (v1.0) released in 2017. The total size of the specification has almost quadrupled in that time, going from 51 endpoints documented in v1.0 to 201 endpoints documented in v2.1. Because of this growth, the specification documents were reorganized into four modules: BrAPI-Core, BrAPI-Germplasm, BrAPI-Genotyping, and BrAPI-Phenotyping. Figure <a href="#fig:domains">1</a> shows a simplified domain map of the whole BrAPI v2.1 data model, divided into the organizational modules. The early versions of the specification focused on read-only phenotype data, with a small consideration to the other domains. Now the specification has a full representation of most of the major concepts applicable to the breeding and research process. The new specification is also internally consistent, easier to navigate, and allows for read, write, and update capabilities. None of those qualities were a guarantee for the earlier versions.</p>
<div id="fig:domains" class="fignos">
<figure>
<img src="images/BrAPI_Domains_v2-1_vertical.png" style="width:100.0%" alt="Figure 1: A simplified domain map of the whole BrAPI data model, divided into organizational modules. A more detailed Entity Relationship Diagram (ERD) is available on brapi.org." />
<figcaption aria-hidden="true"><span>Figure 1:</span> A simplified domain map of the whole BrAPI data model, divided into organizational modules. A more detailed Entity Relationship Diagram (ERD) is available on brapi.org.</figcaption>
</figure>
</div>
<p>As the specification has matured, so have the tools, services, and libraries available to the community to work with the specification. Every version of the specification is now released with a change log to guide developers upgrading from a previous version, an Entity Relationship Diagram (ERD) to describe the whole data model visually, and a JSON Schema data model to be used in some automated development and validation efforts. For groups who are using Java, Java Script, Python, R, or Drupal, there are community maintained libraries available that contain full BrAPI implementations ready to be added to some existing code. The BrAPI Test Server and the BRAVA validation tool are both still available to the community for testing purposes, and they have been maintained to support every version of the specification. Finally, there are three new resource list pages on brapi.org to advertise the BrAPI compatible software available in the community. The BrAPPs list page, servers list page, and compatible software list page showcase many of the BrAPI compatible applications and data resources available in the community.</p>
<h3 id="community-growth">Community Growth</h3>
<p>The international BrAPI Community consists of software developers, biologists, and other scientists working on BrAPI related projects and data sources. This community is what sustains the BrAPI project, builds implementations, maintains development tools, and provides input to enhance the specification. As the project has grown, so has the community. The BrAPI project started in June 2014 with less than ten people coming together to discuss the idea. Over the next ten years, the community has grown to between 200 and 250 members.</p>
<p>The BrAPI Hackathons are a major staple of the BrAPI community. Twice a year, the community gathers to discuss the specification and collaborate on BrAPI related projects. This time is very valuable to the community; for some organizations, the hackathon is the only time during the year when they can collaboratively work on anything related to BrAPI. During the COVID-19 pandemic, virtual hackathons took the place of in-person events. While the virtual hackathons do not provide the same level of face-to-face time that is crucial to collaborative work, they did allow for more attendees to gather and share their opinions. Going forward, the community leadership has decided to have one in-person hackathon and one virtual hackathon each year, to balance the advantages of both.</p>
<h2 id="results">Results</h2>
<!-- success stories highlighting BrAPI usefulness in breeding cycle. Perhaps reference the original BrAPI paper where possible use cases were proposed.  -->
<p>Below are a number of short success stories from the BrAPI community. These tools, applications, and infrastructure projects serve as another indicator of community growth and success over the past 5-10 years. These stories clearly illustrate all the different ways the BrAPI Standard can be used productively and in practice. Figure <a href="#fig:apps">2</a> contains a summary of the tools described below.</p>
<div id="fig:apps" class="fignos">
<figure>
<img src="images/BrAPI_Paper_Applications_Chart.png" style="width:100.0%" alt="Figure 2: A summary of all the tools described below and the general areas each tool is designed to handle" />
<figcaption aria-hidden="true"><span>Figure 2:</span> A summary of all the tools described below and the general areas each tool is designed to handle</figcaption>
</figure>
</div>
<!-- NOTES TO UPDATE THE  BrAPI_Paper_Applications_Chart.png :
FAIDARE: Core, Phenotyping, Genotyping, Germplasm Visualization .
-->
<h3 id="phenotyping">Phenotyping</h3>
<!-- Ajay and Trevor -->
<p>Phenotyping is fundamental to plant breeding and genetics research, providing the accurate, high-quality data needed for downstream analyses and selection decisions.
Effective phenotyping requires a thorough understanding of both biological research questions and operational data gathering to ensure successful outcomes.
Collected data, subsequent analyses, and data visualizations all impact and shape downstream experiments.
The BrAPI specification supports phenotypic data throughout the entire breeding pipeline, including collection, analyses, publication, and archiving.
The BrAPI community has developed several BrAPI-compatible tools to facilitate data standardization, efficient storage, and curation of phenotypic data and trait metadata.
Ongoing development efforts are creating tools to manage images and other high-throughput phenotypic data sources, further enhancing the precision and efficiency of plant breeding research.
By supporting the accurate and efficient collection and storage of phenotypic data, BrAPI compatible tools simplify the conversion of phenotypes into insights that are necessary to help digitize and boost modern breeding and genetics research programs.</p>
<h4 id="field-book">Field Book</h4>
<!-- Trevor and Chaney -->
<p>Data from plant breeding and genetics experiments has traditionally been collected using pen and paper, but this approach often results in transcription errors and delayed analysis.
<a href="fieldbook.phenoapps.org/">Field Book</a><span class="citation" data-cites="MlqG0w6i"><sup><a href="#ref-MlqG0w6i" role="doc-biblioref">2</a></sup></span>, a highly-customizable Android app, was developed to help scientists digitize and organize their phenotypic data as measurements are collected.
This effectively improves data collection speed, reduces errors, and enables larger and more robust breeding populations and data sets.
Field Book has added support for BrAPI to streamline data transfer to and from BrAPI-compatible servers.
Removing the need to manually transfer data files simplifies data exchange between these systems and reduces the opportunities for human error and data loss.</p>
<h4 id="gridscore">GridScore</h4>
<!-- Sebastian R -->
<!-- Make distinct from Field Book, highlight what makes GridScore different -->
<!-- Better flow is needed between Field Book and GridScore -->
<p><a href="https://ics.hutton.ac.uk/get-gridscore/">GridScore</a><span class="citation" data-cites="Ne0E9HbC"><sup><a href="#ref-Ne0E9HbC" role="doc-biblioref">3</a></sup></span> is a modern mobile application for phenotypic observations that harnesses technological advancements in the area of mobile devices to enrich the data collection process. GridScore focuses on user experience by closely mirroring the look and feel of printed field plans. It also enriches the experience with a wide range of functionalities including data validation, data visualizations, georeferencing, cross-platform support, and data synchronization across multiple devices. GridScore is a multi-platform tool which works on any reasonably modern device with a web browser including laptops, PCs, tablets and phones. Once created, trials can be transferred to collection devices using a Quick Response (QR) code. Its approach towards data collection uses a top-down view onto the trial and offers field navigation mechanisms using barcodes, QR codes, or guided walks which take the data collector through the field in one of 16 pre-defined orders.</p>
<p>BrAPI has further increased the value of GridScore by integrating it into the overarching workflow, from trial creation, through data collection, and to its ultimate data storage for further processing. Specifically, trial designs as well as trait definitions can be imported into GridScore using BrAPI and a finalized trial can be exported via BrAPI to any compatible database.</p>
<h4 id="climmob">ClimMob</h4>
<!-- Marie-Angelique -->
<p><a href="https://climmob.net/">ClimMob</a><span class="citation" data-cites="LCpCba21"><sup><a href="#ref-LCpCba21" role="doc-biblioref">4</a></sup></span> is a software suite for a different research paradigm in experimental agriculture. In traditional breeding, a few researchers design complicated trials in search of the best solutions for a few target environments. ClimMob enables many participants to carry out reasonably simple experiments across many environments. Taken together, this data across many environments can be very informative. It applies the principles of citizen science and choice experiments to scale the data collection process, mostly in the format of rankings. Although this data may not be as detailed as from a centralized experiment, it can be very useful to inform decisions to a wide range of locations and environments with increased external validity. ClimMob applications include testing crop varieties, evaluating agronomic practices, and investigating climate resilience strategies. The platform supports experiment design, data collection through mobile apps, and data analysis to provide actionable insights.</p>
<p>During a crop trial, all farmer-collected data is stored in ClimMob. When data collection is complete, the raw data is automatically uploaded via BrAPI to a central breeding database for long-term storage and analysis. To facilitate this synchronization, ClimMob uses BrAPI to retrieve curated germplasm information from breeding databases when designing a trial, significantly enhancing data quality. Additionally, a process has been developed to push analyzed data from ClimMob to the breeding databases, providing breeders with insights into the potential adoption of the tested crop varieties.</p>
<h4 id="imagebreed">ImageBreed</h4>
<p><a href="https://imagebreed.org/">ImageBreed</a><span class="citation" data-cites="Dhlmsxng"><sup><a href="#ref-Dhlmsxng" role="doc-biblioref">5</a></sup></span> is an image collection pipeline tool to support regular use of UAVs and UGVs. High-throughput phenotyping has been gaining significant traction lately as a way to collect lots of data very quickly. Image collection from unmanned aerial and ground vehicles (UAVs and UGVs) are a great way to collect a lot of raw data all at once, then analyze it later.</p>
<p>When the raw images have been processed through the standardization pipelines in ImageBreed, useful phenotypes can be extracted from the images. The BrAPI standard is used to push these phenotypes back to a central breeding database where they can be analyzed with other data. In addition to this, ImageBreed has the ability to use BrAPI to upload the raw images to the central breeding database, or any other BrAPI compatible long term storage service. In the current version of the standard (V2.1), the BrAPI data models for images are rudimentary, but effective. The ImageBreed team has put in some work into enhancing the BrAPI image data standards, collaborating with others in the community.</p>
<h4 id="phis">PHIS</h4>
<!-- Isabelle -->
<p><a href="http://www.phis.inrae.fr/">PHIS</a><span class="citation" data-cites="3TbA7PwE"><sup><a href="#ref-3TbA7PwE" role="doc-biblioref">6</a></sup></span>, the Hybrid Phenotyping Information System, is an ontology-driven information system based on semantic web technologies, based on the <a href="https://github.com/OpenSILEX/">OpenSILEX</a> framework. PHIS is deployed in several field and greenhouse platforms of the French national <a href="https://www.phenome-emphasis.fr/">PHENOME</a> and European <a href="https://emphasis.plant-phenotyping.eu/">EMPHASIS</a> infrastructures. It manages and collects data from basic phenotyping and high throughput phenotyping experiments on a day to day basis. PHIS unambiguously identifies all the objects and traits in an experiment, and establishes their types and relationships via ontologies and semantics.</p>
<p>PHIS has been designed to be BrAPI-compliant. PHIS adheres to the standards and protocols specified by BrAPI and implements various services aligning with the BrAPI standards, encompassing the Core, Phenotyping, and Germplasm modules. This enables integration and compatibility with BrAPI-compliant systems and platforms, such as OLGA, a genebank accessions management system, to retrieve accession information. This prerequisite served as the basis for formalizing the data model, while also facilitating compatibility with other standards, such as the Minimal Information About a Plant Phenotyping Experiment (<a href="https://www.miappe.org/">MIAPPE</a><span class="citation" data-cites="kWLsm7g6"><sup><a href="#ref-kWLsm7g6" role="doc-biblioref">7</a></sup></span>). By integrating BrAPI requirements into its structure, PHIS not only meets the standards of the phenotyping field, but also strengthens its capacity for interoperability and effective collaboration in the wider context of plant breeding and related fields. The fact that data within a PHIS instance can be queried through BrAPI services enables indexing of PHIS in the <a href="https://urgi.versailles.inrae.fr/faidare/">FAIDARE</a> data portal<span class="citation" data-cites="HQFMUyfV"><sup><a href="#ref-HQFMUyfV" role="doc-biblioref">8</a></sup></span>.</p>
<p>Furthermore, as PHIS offers BrAPI-compliant Web Services, it simplifies the integration and data exchange with other European information systems that handle phenotyping data. The adhesion to BrAPI standards ensures a common interface and compatibility, facilitating communication and collaboration between PHIS and other systems in the European context. This interoperability not only eases data sharing, but also promotes a more coherent and efficient approach to the management and use of phenotyping data on various platforms and research initiatives within the European scientific community.</p>
<h4 id="pippa">PIPPA</h4>
<!-- Rafael -->
<p><a href="https://pippa.psb.ugent.be">PIPPA</a><span class="citation" data-cites="okf3YI9z"><sup><a href="#ref-okf3YI9z" role="doc-biblioref">9</a></sup></span> is a data management system used for collecting data from the <a href="https://www.wiwam.be/">WIWAM</a><span class="citation" data-cites="r84WbX0P"><sup><a href="#ref-r84WbX0P" role="doc-biblioref">10</a></sup></span> range of automated high throughput phenotyping platforms. These platforms have been deployed at different research institutes and commercial breeders across Europe. They can be setup in a variety of configurations with different types of equipment including weighing scales, cameras, and environment sensors. The software features a web interface with functionality for setting up new experiments, planning imaging and irrigation treatments, linking metadata to pots (genotype, growth media, manual treatments), importing data, exporting data, and visualizing data. It also supports the integration of image analysis scripts and connections to a compute cluster for job submission.</p>
<p>To share the phenotype data of the experiments linked to publications, an implementation of BrAPI v1.3 was developed which allowed read only access to the data in the BrAPI standardized format. This server was registered on <a href="https://urgi.versailles.inrae.fr/faidare/">FAIDARE</a><span class="citation" data-cites="HQFMUyfV"><sup><a href="#ref-HQFMUyfV" role="doc-biblioref">8</a></sup></span> which allows the data to be found alongside data from other BrAPI compatible repositories.</p>
<p>As the BrAPI ecosystem has matured, it has created a clear path for the further development of PIPPA. The BrAPI specification demonstrates how to share data in a manner consistent with the FAIR principles,<span class="citation" data-cites="6DjakjNS"><sup><a href="#ref-6DjakjNS" role="doc-biblioref">11</a></sup></span> which are becoming best practices in plant research data management. The BrAPI technical standard, in combination with the <a href="https://www.miappe.org/">MIAPPE</a><span class="citation" data-cites="kWLsm7g6"><sup><a href="#ref-kWLsm7g6" role="doc-biblioref">7</a></sup></span> scientific standard, have served as guidelines in the current development effort of the PIPPA project. This development is focused on delivering a public BrAPI v2.1 endpoint and making more high throughput datasets publicly available via BrAPI.</p>
<h4 id="trait-selector-brapp">Trait Selector BrAPP</h4>
<!-- Mirella & Valentin -->
<!-- the "BrAPP" description part should better fit at the beginning of the section. -->
<!-- BrAPPs are simple tools developed by the BrAPI community that are entirely reliant on BrAPI for their data requirements. Often, they are JavaScript based applications or visualizations that fit on a single web page. This means a single BrAPP can be easily shared and used by many organizations and systems, as long as those organizations have the standard BrAPI endpoints available. -->
<p>The <a href="https://github.com/solgenomics/BrAPI-Trait-selector">Trait Selector BrAPP</a> is used to search and select useful traits, using a visual aid to help the user find exactly what they need. Instead of searching through a long list of possible traits, the user is presented with a cartoon image of a species. They can then click on pieces of the image to show traits associated to that part of the plant. For a breeder, they might use it to quickly find specific traits to study. For a genebank user, they might use it to find varieties that have a specific trait they are interested in.</p>
<p>The Trait Selector can be integrated into any website or system, assuming there is a BrAPI compatible data source available to connect to. A breeding database would need to only implement the BrAPI endpoints for Traits, Observations, and Variables, while a genebank would require Traits and Germplasm Attributes. Any BrAPI server with either of these sets of endpoints implemented could use this BrAPP. CassavaBase and MGIS are two successful examples of the Trait Selector BrAPP in use.</p>
<!-- (example screenshots coming + supplementary data: links to the git and the doc) -->
<h3 id="genotyping">Genotyping</h3>
<!-- Ajay -->
<p>Genotyping has become a cornerstone of most breeding processes, but managing the data can be challenging. Understanding different genotyping protocols for various crops is crucial due to the unique genetic structures of each species. Techniques such as SNP genotyping, Genotyping-by-Sequencing (GBS), SSRs, Whole Genome Sequencing (WGS), and array-based genotyping each offer specific advantages depending on the crop and research objectives. BrAPI supports genotypic data by utilizing existing standards such as VCF<span class="citation" data-cites="PzHxvkzH"><sup><a href="#ref-PzHxvkzH" role="doc-biblioref">12</a></sup></span> and the GA4GH Variants schema<span class="citation" data-cites="186zclKvK"><sup><a href="#ref-186zclKvK" role="doc-biblioref">13</a></sup></span>. The BrAPI community has developed compatible tools for storing, searching, visualizing, and analyzing genotypic data, making it easier to integrate and utilize this information in breeding programs. Mastery of the various genotyping protocols ensures efficient and effective breeding, while BrAPI compliant tools streamline data management and analysis, enhancing the ability to make data-driven decisions in developing superior crop varieties.</p>
<h4 id="flapjack">Flapjack</h4>
<p><a href="https://ics.hutton.ac.uk/flapjack">Flapjack</a><span class="citation" data-cites="Wz8l0HlN"><sup><a href="#ref-Wz8l0HlN" role="doc-biblioref">14</a></sup></span> is a multi-platform desktop application for data visualization and breeding analysis (eg, pedigree verification, marker-assisted backcrossing and forward breeding) using high-throughput genotype data. Data can be easily imported into Flapjack from any BrAPI compatible data source with genotype data available. <a href="https://github.com/cropgeeks/flapjack-bytes">Flapjack Bytes</a> is a smaller, lightweight and fully web-based counterpart to Flapjack, which can be easily embedded into a database website to provide similar visualizations online. Traditionally supporting its own text-based data formats, Flapjack’s use of BrAPI has streamlined the end-user experience for data import and work is underway to determine the best methods to exchange analysis results using future versions of the API.</p>
<h4 id="dartview">DArTView</h4>
<!-- Moses N -->
<p><a href="https://software.kddart.com/kdxplore/dartview/dartviewdocs/KDXplore-DartView.html">DArTView</a> is a desktop application for marker data curation via metadata filtering. DArTView enables genotype variant data visualization and users can easily identify trends or correlations within their data using the tool. Its primary goal is to overcome tedious manual calculation of marker data through common spreadsheet applications like Excel. Users are able to import marker data from csv files, but DArTView has been recently enhanced to be BrAPI compatible. Users can now use any BrAPI compatible server as an input data source. BrAPI provides a consistent data standard across databases and data resources. DArTView’s compatibility with BrAPI also ensures easy integration with other tools and pipelines that would use DArTView for marker filtering and exploration.</p>
<p>Initially developed by Diversity Arrays Technology (DArT), the tool is gaining popularity within the breeding community, especially in Africa. Future releases will focus on enhancing the BrAPI compatibility, making it accessible to more breeders and researchers in the region. A web enabled version of DArTView is in development. This new version will allow for further collaboration opportunities with other interested partners who would like to integrate it as part of their pipelines.</p>
<h4 id="divbrowse">DivBrowse</h4>
<!-- Sebastian B -->
<p><a href="https://divbrowse.ipk-gatersleben.de/">DivBrowse</a><span class="citation" data-cites="9fRbvOj2"><sup><a href="#ref-9fRbvOj2" role="doc-biblioref">15</a></sup></span> is a web platform for exploratory data analysis of huge genotyping studies. The software can be run standalone or integrated as a plugin into existing data web portals. It provides a powerful interactive visualization of variant call matrices with hundreds of millions of variants and thousands of samples. It enables easy data import and export by using well established, standardized, bioinformatics file formats.</p>
<p>At its core, DivBrowse combines the convenience of a genome browser with features tailored to the diversity analysis of germplasm. It is able to display genomic features such as nucleotide sequence, associated gene models, and short genomic variants. DivBrowse provides visual access to large VCF files obtained through genotyping experiments. In addition, DivBrowse also calculates and displays variant statistics such as minor allele frequencies, proportion of heterozygous calls, and proportion missing variant calls. Dynamic Principal Component Analyses (PCA) can be performed on a user specified genomic area to provide information on local genomic diversity.</p>
<p>Parts of the BrAPI Genotyping module are implemented in DivBrowse. There is a server-side component which provides genotypic data that the DivBrowse database can consume. There is also a client-side GUI component which can visualize genotypic data via any external BrAPI endpoint. In addition to BrAPI, DivBrowse has an internal API to control the tool from a hosting web portal. DivBrowse also has an interface to BLAST, which can be used to directly access genes or other genomic features. The modular structure of DivBrowse allows developers to configure and easily embed links to other external information systems.</p>
<h4 id="gigwa">Gigwa</h4>
<!-- Guilhem -->
<p><a href="https://southgreen.fr/content/gigwa">Gigwa</a> is a Java EE web application providing a means to centralize, share, finely filter, and visualize high-throughput genotyping data<span class="citation" data-cites="AVjkjraw"><sup><a href="#ref-AVjkjraw" role="doc-biblioref">16</a></sup></span>. Built on top of MongoDB, it is scalable and can support working smoothly with datasets containing billions of genotypes. It is installable as a Docker image or as an all-in-one bundle archive. It is straightforward to deploy on servers or local computers and has thus been adopted by numerous research institutes from around the world. Notably, Gigwa serves as a collaborative management tool and a portal for exposing public data for genebanks and breeding programs at some CGIAR centers<span class="citation" data-cites="bvSFyrtt"><sup><a href="#ref-bvSFyrtt" role="doc-biblioref">17</a></sup></span>. The total amount of data hosted and made widely accessible using this system has continued to grow over the last few years.</p>
<p>The Gigwa development team has been involved in the BrAPI community since 2016 and took part in designing the genotype-related section of the BrAPI standard. Gigwa’s first BrAPI-compliant features were designed for compatibility with the Flapjack visualization tool<span class="citation" data-cites="Wz8l0HlN"><sup><a href="#ref-Wz8l0HlN" role="doc-biblioref">14</a></sup></span>. Over time, Gigwa has established itself as the first and most reliable implementation of the BrAPI-Genotyping endpoints. Local collaborators and external partners used it as a reference solution to design a number of tools taking advantage of the BrAPI-Genotyping features (e.g., <a href="https://webtools.southgreen.fr/BrAPI/Beegmac/">BeegMac</a>, <a href="https://github.com/jframi/snpclust">SnpClust</a>, <a href="https://github.com/icarda-git/QBMS">QBMS</a>).</p>
<p>Additional use-cases required Gigwa to also consume data from other BrAPI servers. This led to the implementation of BrAPI client features within Gigwa. A close collaboration was established with the Integrated Breeding Platform team developing the widely used Breeding Management System (BMS). This collaboration means both applications are now frequently deployed together; Gigwa pulling germplasm or sample metadata from BMS, and BMS displaying Gigwa-hosted genotypes within its own UI.
<!-- 
Community members typically write adhoc scripts federating data from multiple BrAPI sources using BrAPI client libraries available for R, python, and other programming languages.  For instance, phenotypes from one data source and genotypes from another in order to run various kinds of analyses such as GWAS, genomic selection or phylogenetic investigations. The most generic and widely-used of those pipelines are at least publicly distributed, and possibly web-interfaced using solutions like R-Shiny. This provides new, excitingly useful, online services, based on Gigwa-hosted data. --></p>
<h4 id="phg">PHG</h4>
<!-- Lynn J. -->
<p>The <a href="https://www.maizegenetics.net/phg">Practical Haplotype Graph</a> (PHG) is a graph-based computational framework that represents large-scale genetic variation and is optimized for plant breeding and genetics<span class="citation" data-cites="VFIpO8nU"><sup><a href="#ref-VFIpO8nU" role="doc-biblioref">18</a></sup></span>. Using a pangenome approach, each PHG stores haplotypes (the sequence of part of an individual chromosome) to represent the collected genes of a species. This allows for a simplified approach for dealing with large scale variation in plant genomes. The PHG pipeline provides support for a range of genomic analyses and allows for the use of graph data to impute complete genomes from low density sequence or variant data.</p>
<p>Users access the crop databases either with direct calls to the PHG embedded server or indirectly using the rPHG library from an R environment. The PHG server accepts BrAPI queries to return information on sample lists and the variants used to define the graph’s haplotypes. In addition, PHG users utilize the BrAPI Variant Sets endpoint query to return links to VCF files containing haplotype data. Work on the PHG is ongoing and it is expected to support additional BrAPI endpoints that allow for fine tuned slicing genotypic data in the near future.</p>
<h4 id="dart-sample-submission">DArT Sample Submission</h4>
<!-- Grzegorz Uszynski and Puthick Hok - Diversity Arrays Technology DArT -->
<p>The DArT genotyping lab is heavily used world wide when it comes to plant genotyping. With over 1200 available organisms and species, client base on every continent and already many million samples processed, DArT provides services for several generic and bespoke genotyping technologies and solutions. Processes of sample tracking and fast data delivery are at the core of the ordering system developed at DArT. The ordering system is tightly integrated with DArTdb - DArT’s custom LIMS operational system, which drives laboratory, quality, and analytical processes.</p>
<p>Diversity Arrays Technology was a part of BrAPI community since its inception. DArT developers have worked with the BrAPI community contributing to various aspects of the API specification. One key aspect was establishing a standard API for sending sample metadata to the lab for genotyping. This solution eliminates much of the human error involved with sending samples to an external lab and also allows for an automated process of sample batch transfers. Beyond sample submission, the current implementation also allows for an order status verification, automated data discovery, and data downloads. Data are delivered as standard data packages with self-describing metadata.</p>
<p>The current BrAPI implementation at DArT is in production and it is compatible with the newest BrAPI specification. Further details about DArT’s ordering system can be found at <a href="https://ordering.diversityarrays.com">DArT Ordering System</a> and also at <a href="https://help.diversityarrays.com/docs/ordering">DArT Help</a>.</p>
<h3 id="germplasm-management">Germplasm Management</h3>
<!-- Ajay -->
<p>Germplasm data management is crucial due to the vast quantity of new accessions, variants, and lines created yearly. Germplasm is the basis of variation on which plant breeders rely to upgrade and optimize their breeding programs. This is essential at any scale including individual breeding programs, national initiatives, and international collaborations. BrAPI supports the transmission of germplasm passport data, pedigree trees, and crossing metadata. The BrAPI community has developed compliant tools for storing, searching, and visualizing this metadata, facilitating efficient management. Additionally, there are plans to establish federated networks of genebank data connected via BrAPI, enhancing global accessibility and collaboration in germplasm management.</p>
<!-- Contribution BrAPI 2.0 paper
Suggested Authors: Matthias Lange, Patrick König, Stephan Weise, Gouripriya Davuluri, Suman Kumar, Joseph Ruff, Paul Kersey, Cyril Pommier, Michael Alaux, Erwan Le-Floch -->
<h4 id="agent">AGENT</h4>
<!-- Matthias Lange -->
<p>In the global system for ex situ conservation of plant genetic resources (PGR)<span class="citation" data-cites="S1uQPBPG"><sup><a href="#ref-S1uQPBPG" role="doc-biblioref">19</a></sup></span>, material is being conserved in about 1750 collections<span class="citation" data-cites="USun3Z8l"><sup><a href="#ref-USun3Z8l" role="doc-biblioref">20</a></sup></span> totalling ~5.8 million accessions. Unique and permanent identifiers in the form of DOIs are available for more than 1.7 million accessions via the Global Information System<span class="citation" data-cites="i0Aj9IwD"><sup><a href="#ref-i0Aj9IwD" role="doc-biblioref">21</a></sup></span> of the International Treaty on Plant Genetic Resources for Food and Agriculture. Each DOI is linked to some basic descriptive data that facilitates the use of these resources, mainly passport data. Many DOIs are also linked to additional data from different domains or will be in the future. However, a data space beyond the most basic information is needed that includes genotypic and phenotypic data. This space will help to answer questions about the global biological diversity of plant species, the detection of duplicates, the tracking of provenance for the identification of genetic integrity, the selection of the most suitable material for different purposes, and to support further applications in the field of data mining or AI. In this context, the aim of the <a href="https://www.agent-project.eu/">AGENT project</a>, funded by the European Commission, is to develop a concept for the digital exploitation and activation of this PGR via European ex situ genebanks according to the FAIR principles<span class="citation" data-cites="6DjakjNS"><sup><a href="#ref-6DjakjNS" role="doc-biblioref">11</a></sup></span> and to test it in practice using two important crops, barley and wheat. Thirteen European genebanks and five bioinformatics centers are working together and have agreed on standards and protocols for data flow and data formats<span class="citation" data-cites="31NWAHsO"><sup><a href="#ref-31NWAHsO" role="doc-biblioref">22</a></sup></span> for the collection, integration, and archiving of genotypic and phenotypic data.</p>
<p>The BrAPI specification is one of the agreed standards, that are detailed in the AGENT guidelines for dataflow<span class="citation" data-cites="QRzg4OSs"><sup><a href="#ref-QRzg4OSs" role="doc-biblioref">23</a></sup></span>. The implemented BrAPI interface enables the analysis of current and historic genotypic and phenotypic information. This will drive the discovery of genes, traits, and knowledge for future missions, complement existing information for wheat and barley, and use the new data standards and infrastructure to promote better access and use of PGR for other crops in European genebanks. The AGENT database backend aggregates curated passport data, phenotypic data, and genotypic data on wheat and barley accessions of 18 project partners. This data is accessible via <a href="https://github.com/AGENTproject/BrAPI">BrAPI endpoints</a> and explorable in a <a href="https://agent.ipk-gatersleben.de">web portal</a>. Genotyping data uses the DivBrowse<span class="citation" data-cites="9fRbvOj2"><sup><a href="#ref-9fRbvOj2" role="doc-biblioref">15</a></sup></span> storage engine and its BrAPI interface. Soon, the BrAPI implementation will be expanded to enable the integration of analysis pipelines in the AGENT portal, such as the FIGS+ pipeline developed by ICARDA<span class="citation" data-cites="6VGwG9za"><sup><a href="#ref-6VGwG9za" role="doc-biblioref">24</a></sup></span>. There is also a plan to integrate the data collected by the AGENT project into the European Search Catalogue for Plant Genetic Resources (EURISCO)<span class="citation" data-cites="pk5OwSG6"><sup><a href="#ref-pk5OwSG6" role="doc-biblioref">25</a></sup></span>.
<!-- ![Figure Data flow of genotypic data from AGENT partner databases](images/AGENT_Genotyping_Data_Flow.png){#fig:AGENT_Genotyping_Data_Flow width="100%"} -->
<!-- ![Figure AGENT Portal](images/AGENT_Web_Frontend.png){#fig:AGENT_Web_Frontend width="100%"} --></p>
<h4 id="mgis">MGIS</h4>
<!-- authors: Mathieu Rouard, Valentin Guignon -->
<p>The Musa Germplasm Information System (<a href="https://www.crop-diversity.org/mgis/">MGIS</a>) serves as a comprehensive community portal dedicated to banana diversity, a crop critical to global food security<span class="citation" data-cites="84Yn3FWj"><sup><a href="#ref-84Yn3FWj" role="doc-biblioref">26</a></sup></span>. MGIS offers detailed information on banana germplasm, focusing on the collections held by the CGIAR International Banana Genebank (ITC)<span class="citation" data-cites="JWi7AKhJ"><sup><a href="#ref-JWi7AKhJ" role="doc-biblioref">27</a></sup></span>. It is built on the Drupal/Tripal technology, like BIMS<span class="citation" data-cites="ArPSsaJQ"><sup><a href="#ref-ArPSsaJQ" role="doc-biblioref">28</a></sup></span> and Florilège.</p>
<p>Since its inception, MGIS developers have actively participated in the BrAPI community. The MGIS team pushed for the integration of the Multi-Crop Passport Data (MCPD) standard into the Germplasm module of the API. MCPD support was added in BrAPI v1.3, and MGIS now provides passport data information on ITC banana genebank accessions (with GLIS DOI), synchronized with <a href="https://www.genesys-pgr.org/a/overview/v2YdWZGrZjD">Genesys</a>. MGIS also enriches the passport data by incorporating additional information from other germplasm collections worldwide. All the germplasm data is available through the BrAPI Germplasm module implementation. For genotyping data, MGIS integrates with Gigwa<span class="citation" data-cites="AVjkjraw"><sup><a href="#ref-AVjkjraw" role="doc-biblioref">16</a></sup></span>, which provides a tailored implementation of the BrAPI genotyping module. Furthermore, MGIS supports a set of BrAPI phenotyping endpoints, facilitating the exposure of morphological descriptors and trait information supported by ontologies like the Crop Ontology<span class="citation" data-cites="VndIxnl3"><sup><a href="#ref-VndIxnl3" role="doc-biblioref">29</a></sup></span>. MGIS has integrated the Trait Selector BrAPP, and there are use cases implemented to interlink genebank and breeding data between MGIS and the breeding database MusaBase.</p>
<h4 id="helium">Helium</h4>
<!-- Paul S -->
<p><a href="https://helium.hutton.ac.uk">Helium</a><span class="citation" data-cites="X22hwO1v"><sup><a href="#ref-X22hwO1v" role="doc-biblioref">30</a></sup></span> is a plant pedigree visualization platform designed to account for the specific problems that are unique to plant pedigrees. A pedigree is a representation of how genetically discrete individuals are related to one another and is therefore a representation of the genetic relationship between individual plant lines, their parents and progeny. Plant pedigrees are often used to check for potential genotyping or phenotyping errors, since these errors, by the very nature of Mendelian inheritance, are constrained by the pedigree structure in which they exist<span class="citation" data-cites="1CKLIIaug"><sup><a href="#ref-1CKLIIaug" role="doc-biblioref">31</a></sup></span>. The accurate representation of pedigrees, and the ability to pull pedigree data from different data sources is important in plant breeding and genetics. Therefore, ways to visualize and interact this complex data in meaningful ways is critical.</p>
<p>From its <a href="https://github.com/cardinalb/helium-docs/wiki">original desktop interface</a>, Helium has developed into a web-based visualization platform implementing BrAPI calls to allow users to import data from other BrAPI compliant databases. The ability to pull data from BrAPI compliant data sources has significantly expanded Helium’s capability and utility within the community. Helium is used in projects ranging in size from tens to tens of thousands of lines and across a wide variety of crops and species. While originally designed for plant data<span class="citation" data-cites="1HmEHgW8h"><sup><a href="#ref-1HmEHgW8h" role="doc-biblioref">32</a></sup></span> it has also found utility in other non-plant projects<span class="citation" data-cites="x8DIc6zn"><sup><a href="#ref-x8DIc6zn" role="doc-biblioref">33</a></sup></span> highlighting its broad utility. BrAPI also allows Helium to provide direct dataset links to collaborators, allowing the original data to be held with the data provider and utilizing Helium for its visualization functionality. Our current Helium deployment includes example BrAPI calls to a barley dataset at the James Hutton Institute to allow users to test the system and features it offers.</p>
<h4 id="glis">GLIS</h4>
<!-- Marco M -->
<p>The <a href="https://glis.fao.org/glis/">Global Information System (GLIS)</a> on Plant Genetic Resources for Food and Agriculture (PGRFA) of the International Treaty on Plant Genetic Resources for Food and Agriculture (ITPGRFA) is a web-based global entry point for PGRFA data<span class="citation" data-cites="i0Aj9IwD"><sup><a href="#ref-i0Aj9IwD" role="doc-biblioref">21</a></sup></span>. It allows users and third-party systems to access information and knowledge on scientific, technical, and environmental matters to strengthen PGRFA conservation, management, and utilization activities. The system and its portal also enable recipients of PGRFA to make available all non-confidential information on germplasm according to the provisions of the Treaty and facilitates access to the results of their research and development.</p>
<p>Thanks to the adoption of Digital Object Identifiers (DOIs) for Multi-Crop Passport Descriptors (MCPD) of PGRFA accessions, the GLIS Portal provides access to 1.7 million PGRFA in collections conserved worldwide. Of these, over 1.5 million are accessible for research, training and plant breeding in the food and agriculture domain.</p>
<p>The Scientific Advisory Committee of the ITPGRFA have repeatedly welcomed efforts on interoperability among germplasm information systems. In this context, the GLIS Portal adopted the BrAPI v1 in 2022. Integrating BrAPI among the GLIS content negotiators facilitates queries and the exchange of content for data management in plant breeding. The Portal also offers other protocols (XML, DarwinCore, JSON and JSON-LD) to increase data and metadata connectivity. In the near future, depending on the availability of resources, upgrading to BrAPI v2 is planned.</p>
<h4 id="florilège">Florilège</h4>
<!-- Alice B -->
<p><a href="https://florilege.arcad-project.org/">Florilège</a> is a web portal designed primarily for the general public to access public plant genetic resources held by Biological Resource Centers across France. Through this portal, users can browse accessions from over 50 plant genera, spread across 19 genebanks. It allows users to view available seeds and plant material, including options for ordering material. Florilège provides a centralized access to the various French collections of plant genetic resources available to the public.</p>
<p>Florilège retrieves accession information from several BrAPI compliant systems. Key among these are OLGA, a genebank accessions management system, and GnpIS, an INRAE data repository for plant genetic resources, phenomics, and genetics<span class="citation" data-cites="m7yuFGsd 15XwZ2Vb0"><sup><a href="#ref-m7yuFGsd" role="doc-biblioref">34</a>,<a href="#ref-15XwZ2Vb0" role="doc-biblioref">35</a></sup></span>. Using BrAPI to gather data from these systems reduced development efforts and enabled standardized data retrieval. As a result, BrAPI has become the de facto standard within the French plant genetic resources community for exchanging information. During development, the Florilège team also proposed several enhancements to the BrAPI specifications themselves, such as additional support for Collection objects or improved reference linking, to better accommodate their specific use case.</p>
<!-- ![Figure Florilege Workflow](images/Schema_Florilege.jpg){#fig:Schema_Florilege width="100%"} -->
<h3 id="breeding-and-genetics-data-management">Breeding and Genetics Data Management</h3>
<p>While specialty data management is important for some use cases, often breeders want a central repository or access point of critical data. General breeding and genetics data management systems and web portals support some level of phenotypic, genotypic, and germplasm data, as well as trial, equipment, and people management. By enabling BrAPI support, these larger systems can connect with smaller tools and specialty systems to provide more functionality under the same user interface. There are several breeding data management systems developed in the BrAPI community, each with their own strengths.</p>
<h4 id="deltabreed">DeltaBreed</h4>
<!-- Shawn Y.-->
<p><a href="https://app.breedinginsight.net/">DeltaBreed</a> is an open-source breeding data management system designed and developed by Breeding Insight to support USDA-ARS specialty crop and animal breeders. DeltaBreed differs from other related systems in that is customizable to small breeding teams and generalized enough to support the workflows of diverse niche species. DeltaBreed is a unified system that connects a variety of BrAPI applications (see list below). BrAPI integration allows the complexity underlying interoperability to be hidden, shielding users from multifactorial differences between various applications. DeltaBreed, adhering to the BrAPI model, establishes data standards and validations for users and provides a singular framework for data management and user training.</p>
<p>DeltaBreed users need not be aware of BrAPI or the specifics of underlying applications but will notice that BrAPI interoperability reduces the need for human-mediated file transfers and data manipulation. Field Book users, for example, can connect to their DeltaBreed program, authenticate, and pull studies and traits directly from DeltaBreed to Field Book on their data collection device. The subsequent step of pushing observations from Field Book to DeltaBreed is straightforward via BrAPI, but is pending implementation until observation transaction handling is improved, intentional and inadvertent repeated measures are differentiated, and a data staging area is implemented for quality control.</p>
<p>Breeding Insight integrated several BrAPI applications to support 2021 phenotypic data collection by USDA-ARS blueberry breeders. DeltaBreed was used to create traits in Breedbase, and Field Book was used to pull studies and traits from Breedbase. The workflow also permitted users to push Field Book observations back to Breedbase via BrAPI. This effort served as a successful proof of concept for multi-application BrAPI integration, but highlighted limitations to the process of accepting BrAPI observations from Field Book. The Breeding Insight team is actively working with the rest of the BrAPI community to correct these limitations in future versions of the specification.</p>
<!-- Submission is expected July 2024. We may need to trim this aspirational list down to reality in final edits.-->
<p>DeltaBreed is integrated with many other BrAPI community projects and tools. There is a BrAPI enabled connection, either in development or production, with all of the following tools: <a href="https://test-server.brapi.org/brapi/v2/">BrAPI Java Test Server</a>, <a href="https://breedbase.org/">BreedBase</a>, <a href="https://play.google.com/store/apps/details?id=com.fieldbook.tracker">Field Book</a>, <a href="https://gigwa.southgreen.fr/gigwa/">Gigwa</a>, <a href="https://icarda-git.github.io/QBMS">QBMS</a>, <a href="https://github.com/AparicioJohan/MrBeanApp">Mr Bean</a>, <a href="https://helium.hutton.ac.uk/#/">Helium</a> and the <a href="https://github.com/solgenomics/BrAPI-Pedigree-Viewer">Pedigree Viewer</a> BrAPP.</p>
<h4 id="bms">BMS</h4>
<!-- Corina -->
<p>The <a href="https://bmspro.io">Breeding Management System (BMS)</a>, developed by the <a href="https://integratedbreeding.net/">Integrated Breeding Platform (IBP)</a>, is a suite of tools designed to enhance the efficiency and effectiveness of plant breeding. BMS covers all stages of the breeding process, with the emphasis on germplasm management and <a href="https://cropontology.org">ontology</a>-harmonized phenotyping. It also features analytics and decision-support tools. With its focus on interoperability, BMS integrates smoothly with BrAPI, facilitating easy connections with a broad array of complementary tools and databases. Notably, the BMS is often deployed together with <a href="https://southgreen.fr/content/gigwa">Gigwa</a> to fulfill the genotyping data management needs of BMS users.</p>
<p>The <a href="https://github.com/IntegratedBreedingPlatform/brapi-sync">brapi-sync</a> tool, a significant component of BMS’s BrAPI capabilities, was developed by the IBP and released as a BrAPP for community use. Brapi-sync is designed to enhance collaboration among partner institutes within a network such as Innovation and Plant Breeding in West Africa (<a href="https://www.iavao.org/en">IAVAO</a>). The tool enables the sharing of germplasm and trial metadata across BrAPI-enabled systems. It helps overcome traditional barriers to collaboration, ensuring data that was once isolated within specific programs or platforms can now be easily shared, integrated, and synchronized.</p>
<p>Additionally, brapi-sync improves data management by maintaining links to the original source of each entity it transmits. This retains the original context of the data and establishes a traceability mechanism for accurate data source attribution and verification. Such practices are crucial for maintaining data integrity and fostering trust among collaborative partners, ensuring access to accurate, reliable, and current information.</p>
<h4 id="breedbase">Breedbase</h4>
<!-- Lukas -->
<p><a href="https://breedbase.org/">Breedbase</a> is a comprehensive breeding data management system<span class="citation" data-cites="al3Hd9Ml 1FDf25dDD"><sup><a href="#ref-al3Hd9Ml" role="doc-biblioref">36</a>,<a href="#ref-1FDf25dDD" role="doc-biblioref">37</a></sup></span> that implements a digital ecosystem for all breeding data, including trial data, phenotypic data, and genotypic data. Data acquisition is supported through data collection apps such as Fieldbook<span class="citation" data-cites="MlqG0w6i"><sup><a href="#ref-MlqG0w6i" role="doc-biblioref">2</a></sup></span>, Coordinate, and InterCross, as well as through drone imagery, Near Infra-Red Spectroscopy (NIRS), and other technologies. Search functions, such as the Search Wizard interface, provide powerful query capabilities. Various breeding-centric analysis tools are available, including mixed models, heritability, stability, PCA, and various clustering algorithms. The original impetus for creating Breedbase was the advent of new breeding paradigms based on genomic information such as genomic prediction algorithms<span class="citation" data-cites="zuq8ri4R"><sup><a href="#ref-zuq8ri4R" role="doc-biblioref">38</a></sup></span> and the accompanying data management challenges. Thus, complete genomic prediction workflow is integrated in the system.</p>
<p>The BrAPI interface is crucial for Breedbase. Breedbase uses BrAPI to connect with the data collection apps, other projects such as CLIMMOB<span class="citation" data-cites="LCpCba21"><sup><a href="#ref-LCpCba21" role="doc-biblioref">4</a></sup></span>, and native BrAPPs built into the Breedbase webpage. Users also appreciate the ability to connect to Breedbase instances using packages such as <a href="https://icarda-git.github.io/QBMS">QBMS</a><span class="citation" data-cites="a9uPTJgO"><sup><a href="#ref-a9uPTJgO" role="doc-biblioref">39</a></sup></span> for data import into R for custom analyses. The Breedbase team has been part of the BrAPI community since its inception, and has continuously adopted and contributed to the BrAPI standard.</p>
<h4 id="bims">BIMS</h4>
<!-- Sook -->
<p><a href="https://wwww.breedwithbims.org">BIMS</a> (Breeding Information Management System)<span class="citation" data-cites="ArPSsaJQ"><sup><a href="#ref-ArPSsaJQ" role="doc-biblioref">28</a></sup></span> is a free, secure, and online breeding management system which allows breeders to store, manage, archive, and analyze their private breeding program data. BIMS enables individual breeders to have complete control of their own breeding data along with access to tools such as data import/export, data analysis, and data archiving for their germplasm, phenotype, genotype, and image data. BIMS is currently implemented in five community databases, the Genome Database for Rosaceae<span class="citation" data-cites="eaDrNpsp"><sup><a href="#ref-eaDrNpsp" role="doc-biblioref">40</a></sup></span>, CottonGEN<span class="citation" data-cites="dcg0274X"><sup><a href="#ref-dcg0274X" role="doc-biblioref">41</a></sup></span>, the Citrus Genome Database, the Pulse Crop Database, and the Genome Database for Vaccinium. BIMS in these five community databases enables individual breeders to import publicly available data so that they can utilize public data in their breeding program. BIMS is also implemented in the public database <a href="https://wwww.breedwithbims.org">breedwithbims.org</a> that any crop breeder can use.</p>
<p>Right now, BIMS primarily utilizes BrAPI to connect with the Field Book Android App<span class="citation" data-cites="MlqG0w6i"><sup><a href="#ref-MlqG0w6i" role="doc-biblioref">2</a></sup></span>, enabling seamless data transfer between BIMS and the app. Data transfer through BrAPI between BIMS and other resources such as BreedBase<span class="citation" data-cites="al3Hd9Ml"><sup><a href="#ref-al3Hd9Ml" role="doc-biblioref">36</a></sup></span>, GIGWA<span class="citation" data-cites="AVjkjraw"><sup><a href="#ref-AVjkjraw" role="doc-biblioref">16</a></sup></span>, and Breeder Genomics Hub is on the way. Hopefully, the BIMS development team can easily reuse some of the solved use cases and workflows created by others in the BrAPI community.</p>
<h4 id="germinate">Germinate</h4>
<!-- Sebastian R -->
<p><a href="https://ics.hutton.ac.uk/get-germinate/">Germinate</a><span class="citation" data-cites="sW9euzzP rPgDlCbt"><sup><a href="#ref-sW9euzzP" role="doc-biblioref">42</a>,<a href="#ref-rPgDlCbt" role="doc-biblioref">43</a></sup></span> is an open-source plant genetic resources database that combines and integrates various kinds of plant breeding data including genotypic data, phenotypic trials data, passport data, images, geographic information and climate data into a single repository. Germinate is tightly linked to the BrAPI specification and supports a majority of BrAPI endpoints for querying, filtering, and submission.</p>
<p>Germinate integrates and connects with other BrAPI-enabled tools such as GridScore for phenotypic data collection, Flapjack for genotypic data visualization, and Helium for pedigree visualization. Additionally, due to the nature of BrAPI, Germinate can act as a data repository for any BrAPI-compatible tool. Thanks to the interoperability provided by BrAPI, the need for manual data handling becomes a rarity with the direct benefit of faster data processing, fewer to no human errors, data security, and data integrity.</p>
<h4 id="faidare">FAIDARE</h4>
<!-- Cyril P, first draft-->
<p><a href="https://urgi.versailles.inrae.fr/faidare/">FAIDARE</a><span class="citation" data-cites="HQFMUyfV"><sup><a href="#ref-HQFMUyfV" role="doc-biblioref">8</a></sup></span> is a data discovery portal providing a biologist friendly search system over a global federation of 40 plant research databases at the time of writing. It allows users to identify data resources using a full text search approach combined with domain specific filters. Each search result contains a link back to the original database for visualization, analysis, and download. The indexed data types are very broad and include genomic features, selected bibliography, QTL, markers, genetic variation studies, phenomic studies, and plant genetic resources. This inclusiveness is achieved thanks to a two stage indexation data model. The first index, more generic, provides basic search functionalities and relies on five fields: name, link back URL, data type, species, and exhaustive description. To provide more advanced filtering, the second stage indexation mechanism takes advantage of BrAPI endpoints to get more detailed metadata on germplasm, genotyping studies and phenotyping studies.</p>
<p>The indexation mechanism relies on a <a href="https://github.com/elixir-europe/plant-brapi-etl-faidare">public software package</a> that allows data resource managers to request the indexation of their database. This BrAPI client is able to extract data from any BrAPI 1.3 and 1.2 endpoint. The development of BrAPI 2.x indexation will be initiated in 2025. Since not all databases are willing to implement BrAPI endpoints, it is possible to generate metadata as static BrAPI compliant JSON files, using the BrAPI standard as a file exchange format.</p>
<p>The FAIDARE architecture has been designed by elaborating on the BrAPI data model mixed with the GnpIS Software Architecture<span class="citation" data-cites="m7yuFGsd"><sup><a href="#ref-m7yuFGsd" role="doc-biblioref">34</a></sup></span>. It uses an Elasticsearch NoSQL engine that searches and serves enriched versions of the BrAPI JSON data model. FAIDARE also includes a BrAPI endpoint using all indexed metadata. It has been adopted by several communities including the ELIXIR and EMPHASIS European infrastructures, and the WheatIS of the Wheat-Initiative. Several databases are added each year to the FAIDARE global federation, allowing to increase both the portal and the BrAPI adoption.</p>
<!-- ![Figure FAIDARE Federation](images/Schema_FAIDARE.png){#fig:Schema_FAIDARE width="100%"} -->
<h3 id="analytics">Analytics</h3>
<p>While other tools listed above have the capability to do specialized analytics on certain types of data, general analytics tools can cover a wide range of data types and analytical models. The tools developed by the BrAPI community can pull in data from multiple BrAPI compatible data sources and provide enhanced analytical functionality. In many cases, there is no longer a need to import and export large data files to a local computational environment just to run standard analytical models. These tools are able to extract the data they need from a data source without much human intervention or human error.</p>
<h4 id="qbms">QBMS</h4>
<!-- Khaled -->
<p>Modern breeding programs can utilize data management systems to maintain both phenotypic and genotypic data. Numerous systems are available for adoption. To fully leverage the benefits of digitalization in this ecosystem, breeders need to utilize data from different sources to make efficient data-driven decisions. With increased computational power at their disposal, scientists can construct more advanced analysis pipelines by combining various data sources.</p>
<p>The <a href="https://icarda-git.github.io/QBMS">QBMS</a><span class="citation" data-cites="a9uPTJgO"><sup><a href="#ref-a9uPTJgO" role="doc-biblioref">39</a></sup></span> R package eliminates technical barriers scientists experience when using the BrAPI specification in their analysis scripts and pipelines. This barrier arises from the complexity of managing API backend processes, such as authentication, tokens, TCP/IP protocol, JSON format, pagination, stateless calls, asynchronous communication, database IDs, and more. To bridge this gap, the QBMS package abstracts the technical complexities, providing breeders with stateful functions familiar to them when navigating their GUI systems. It enables them to query and extract data into a standard data frame structure, consistent with their use of the R language, one of the most common statistical tools in the breeding community.</p>
<p>Since its release on the official CRAN repository in October 2021, the QBMS R package has garnered over 9400 downloads. Several tools, such as MrBean, rely on the QBMS package as their source data adapter. Moreover, the community has started building extended solutions on top of it. QBMS can serve as a cornerstone in the breeding modernization revolution by providing access to actionable data and by enabling the creation of dashboards to reduce the time between harvest and decision-making for the next breeding cycle.</p>
<h4 id="mr.bean">Mr.Bean</h4>
<!-- Johan -->
<p><a href="https://apariciojohan.github.io/MrBeanApp/">Mr.Bean</a><span class="citation" data-cites="7WXz7FUP"><sup><a href="#ref-7WXz7FUP" role="doc-biblioref">44</a></sup></span> is a graphical user interface designed to assist breeders, statisticians, and individuals involved in plant breeding programs with the analysis of field trials. By utilizing innovative methodologies such as SpATS for modeling spatial trends, and autocorrelation models to address spatial variability, Mr.Bean proves highly practical and powerful in facilitating faster and more effective decision-making. Modeling Genotype-by-environment interaction poses its challenges, but Mr.Bean offers the capability to explore various variance-covariance matrices, including Factor Analytic, compound symmetry, and heterogeneous variances. This aids in the assessment of genotype performance across diverse environments.</p>
<p>Mr.Bean boasts flexibility in importing different file types, yet for users managing their data within data management systems (DMS), the process of downloading from their DMS and importing it into Mr.Bean can be cumbersome. To address this issue, QBMS was integrated into the back-end. This feature prompts users to input the URL of a BrAPI compatible server, enter their credentials (if necessary), and select the specific trial they wish to analyze. Subsequently, users can seamlessly access their dataset through BrAPI and utilize it across the entire Mr.Bean interface.</p>
<h4 id="g-crunch">G-Crunch</h4>
<!-- Josh -->
<p>G-Crunch is an upcoming user-facing analysis tool that attempts to fill the space of simple, user driven analytics requests, with a generic user interface and the ability to swap out data sources and analysis tools. G-Crunch hopes to streamline repeatable, debuggable, simple analytic requests and results.</p>
<p>G-Crunch, as a tool, couldn’t feasibly exist without BrAPI. The support of BrAPI interfaces allows G-Crunch to use one unified request method, and adapt to the user’s (BrAPI-compliant) existing network of tools. This lowers the barrier to entry for adoption, and makes analysis pipelines easily repeatable.</p>
<h4 id="shinybrapps">ShinyBrAPPs</h4>
<p>Data management systems are generic and robust centralized applications, with a large community of users, long term development cycles and release plans. BrAPI compliance offers these systems the opportunity to add functionalities in a modular way through the development of external plugin applications that can quickly fulfill specific needs of a group of users. <a href="https://shiny.posit.co/">R-Shiny</a> is a R-package that opens the possibility to develop rich and productive interactive web applications to R users and developers. As such it allows user communities to quickly prototype and produce applications that are finely tailored to their needs, thus improving adoption and daily use of data management systems. The Breeding Management System (BMS) of the IBP and Gigwa are BrAPI compliant and are being widely used by breeding programs, including national breeding programs in Africa. CIRAD and the IBP teams have been working together as part of the <a href="https://www.iavao.org/">IAVAO</a> breeders community to develop the Shiny-BrAPPs, a set of R-Shiny applications using BrAPI as a core technology for data exchange. These applications are connected to BMS and/or Gigwa and provide tools for specific use cases. So far, four applications have been developed covering the fields of trial data quality control, single trial statistical analysis, breeding decision support, and raw genotyping data visual inspection.</p>
<p>The “BMS trial data explorer” retrieves data from a single multi-location trial and displays data counts and summary boxplot for all variables measured in different studies. It also provides an interactive distribution plot to easily select observations that require curation and a report of candidate issues that needs to be addressed by the breeder. The “STABrAPP” tool is an application for single trial mixed model analysis. It basically provides a GUI to the <a href="https://biometris.github.io/statgenSTA/">StatGen-STA</a> R package. The “DSBrAPP” tool is a decision support tool helping breeders to select germplasm according to their various characteristics and save this germplasm list into BMS. These first three apps are grouped into a single repository (<a href="https://github.com/IntegratedBreedingPlatform/ShinyBrAPPs/">ShinyBrAPPs</a>) with modular code shared across applications and make use of the <a href="https://github.com/mverouden/brapir-v1">brapir-v1</a> and <a href="https://github.com/mverouden/brapir-v2">brapir-v2</a> R packages. Finally, the “<a href="https://github.com/jframi/snpclust">snpclust</a>” tool enables a user to check and manually correct the clustering of fluorescence based SNP genotyping data.</p>
<h4 id="sct">SCT</h4>
<!-- Keo -->
<p>The <a href="https://github.com/USDA-ARS-GBRU/SugarcaneCrossingTool">Sugarcane Crossing Tool</a> (SCT) is a lightweight RShiny dashboard application designed to receive, process, and visualize data from a linked BreedBase<span class="citation" data-cites="al3Hd9Ml"><sup><a href="#ref-al3Hd9Ml" role="doc-biblioref">36</a></sup></span> instance. This application is being developed collaboratively with members of the <a href="https://www.amscl.org/sugarcane-integrated-breeding-system/">Sugarcane Integrated Breeding System</a>, who have advocated for an application that assists them in designing crosses based on queried information from a list of available accessions. By leveraging existing community resources, the team has been able to develop a simple, BrAPI-enabled, application without possessing extensive programming knowledge or experience. The SCT is presented as an inspiration for similarly positioned scientists to consider developing custom applications for specific tasks.</p>
<!-- Add this sentence if the repository becomes public -->
<!-- The modularized, open source code is available in a [GitHub repository](https://github.com/USDA-ARS-GBRU/SugarcaneCrossingTool) for community reference and reuse.  -->
<p>The crossing tool utilizes a modified version of the <a href="https://github.com/CIP-RIU/brapi">BrAPI-R</a> library to access a compliant database, and it employs standard R/JavaScript packages to aggregate and visualize data. Modules within the application allow breeders to query the database (through BrAPI) for information relevant to their decision-making process. This includes the number and sex of flowering accessions, deep pedigree and relatedness information, summarized trial data, and the prior frequency and success of potential cross combinations. Future versions of this tool will provide additional decision support (e.g. ranked potential crosses) to enhance the accuracy and efficiency of crossing.</p>
<h3 id="general-infrastructure">General Infrastructure</h3>
<p>Adopting BrAPI compatibility into an existing system can be difficult sometimes. The BrAPI Community has developed several tools to make adoption easier. This includes things like pre built code libraries, connectors to other technology standards, and mappers to alternate data types or data files. The goal is to lower the barrier to entry for the BrAPI community, making it easier for other groups to get started and connect their existing data to the standard.</p>
<h4 id="mira">MIRA</h4>
<!-- Manuel -->
<p>In the plant phenotyping community, <a href="https://www.miappe.org/">MIAPPE</a> (Minimal Information About a Plant Phenotyping Experiment)<span class="citation" data-cites="kWLsm7g6"><sup><a href="#ref-kWLsm7g6" role="doc-biblioref">7</a></sup></span> is an established standard for phenotyping experiments. It is commonly serialized into the ISA-Tab file type.<span class="citation" data-cites="KZc5ayvK"><sup><a href="#ref-KZc5ayvK" role="doc-biblioref">45</a></sup></span> Although ISA-Tab is easy to read for non-technical experts due to its file-based approach, it lacks programmatic accessibility, particularly for web applications. BrAPI, which is aligned with MIAPPE, can help solve this problem.</p>
<p><a href="https://github.com/USDA-ARS-GBRU/SugarcaneCrossingTool">MIRA</a> is a tool that enables the automatic deployment of a BrAPI server on a MIAPPE-compliant dataset in ISA-Tab format. It can be deployed from a Docker image with the dataset mounted. By utilizing the mapping between MIAPPE, ISA-Tab, and BrAPI, there is no need for parsing or manual mapping of datasets that are already compliant with (meta-)data standards. By gaining programmatic access through BrAPI to these datasets, it facilitates the integration of phenotyping datasets into web applications.</p>
<h4 id="brapi2isa">BrAPI2ISA</h4>
<!-- Bert -->
<p>Since the release of BrAPI 1.3, efforts have been made to incorporate support for the MIAPPE standard into the specification<span class="citation" data-cites="kWLsm7g6"><sup><a href="#ref-kWLsm7g6" role="doc-biblioref">7</a></sup></span>. This integration was finalized in BrAPI 2.0, resulting in full compatibility between the two standards. Consequently, BrAPI now encompasses all attributes necessary for MIAPPE compliance, adhering to standardized descriptions in accordance with MIAPPE guidelines.</p>
<p>In some communities and projects, phenotyping data and metadata is archived and published as structured ISA-Tab files, and validated using the <a href="https://github.com/ELIXIR-Belgium/isatab-validation">MIAPPE ISA configuration</a>. The <a href="https://github.com/elixir-europe/plant-brapi-to-isa">BrAPI2ISA</a> service functions as a converter between a BrAPI compatible server and the ISA-Tab format. This simplifies, automates, and facilitates the archiving of data, thereby enhancing data preservation and accessibility. The BrAPI2ISA tool is designed to be compatible with BrAPI 1.3, and is open to contributions from the community to extend support for the latest versions of BrAPI.</p>
<h4 id="brapimapper">BrAPIMapper</h4>
<!-- Valentin G: I'm planning to add a figure in supplementary data that shows data sets on a side, the mapper with some internal parts and the exposed BrAPI services. -->
<p><a href="https://github.com/plantbreeding/BrAPIMapper">BrAPIMapper</a> is a full BrAPI implementation designed to be a convenient wrapper for any breeding related data source. BrAPIMapper is provided as a Docker application that can connect to a variety of external data sources including mySQL or PostgreSQL databases, generic REST services, flat files (XML, JSON, CSV/TSV/GFF3/VCF, YAML), or any combination of these. It provides an administration user interface to map BrAPI data models to external data sources. The interface allows administrators to select the BrAPI specification versions to use and which endpoints to enable. Data mapping configuration import and export features simplify upgrades to future BrAPI versions; administrators only have to map missing fields or make minor adjustments. BrAPIMapper supports the primary BrAPI features including paging, deferred search results, user lists, and authentication. Access restrictions to specific endpoints can be managed through the administration interface as well. This tool aims to accelerate BrAPI services deployment while ensuring specification compliance.</p>
<h4 id="graphql-data-warehouse">GraphQL Data-warehouse</h4>
<!-- Asis Hallab -->
<p>Using the <a href="https://zendro-dev.github.io">Zendro</a> set of automatic software code generators, a fully functional, efficient, and cloud-capable BrAPI data-warehouse has been created for the current version of the BrAPI data models. Unlike most BrAPI compliant data sources, this data-warehouse supports a GraphQL API rather than a RESTful API. This API provides secure access to data read and write functions for all BrAPI data models. It provides create, read, update, and delete (CRUD) functions that are standardized and accept the same parameters for all data models.</p>
<p>The GraphQL server is particularly rich in features. Records are paginated using the highly efficient cursor based pagination model as proposed in the GraphQL standard. Logical filters allow for exhaustive search queries, whose structure is highly intuitive and based around logical triplets. A large collection of operators is available and triplets can be combined to logical search trees using “and” or “or” operators. Searches can be extended over relationships between data models, thus enabling a user to query the warehouse for exactly the required data. Access security is implemented with the OAuth2 user authentication standard<span class="citation" data-cites="KpTrsTH1"><sup><a href="#ref-KpTrsTH1" role="doc-biblioref">46</a></sup></span>. Authorization is based on user roles and can be configured differently for each single data model read or write function. The generated graphical interface allows for the integration of interactive scientific plots and analysis tools written in JavaScript or WebAssembly.</p>
<!-- Asis Hallab: To Do: (i) provide the link to an example running data warehouse, and (ii) include and describe example scientific plots. -->
<p>The Zendro data-warehouse is capable of forming an efficient cloud of data servers. This is achieved simply by linking other Zendro based warehouses that expose the same GraphQL API to the same data models, or a subset of data models. Any network of such Zendro GraphQL servers can be set up using this configuration approach. The code generated then exposes full access to all data records stored on any node of the network, while maintaining full security control at each node. Importantly, the warehouses are programmed in such a way that any number of data servers can be joined without loss of efficiency.</p>
<h2 id="discussion">Discussion</h2>
<!-- * how can BrAPI help breeders (specifically small breeders)
* Looking ahead - what needs to be done further
* Analytics
* HDP
* GraphQL?
* Field management - treatments etc
* Drones, image processing, HTP
* ChatGPT integration - BreedersGPT
* Weather & soil - why it won’t be in BrAPI -->
<h3 id="brapi-for-breeders-and-scientists">BrAPI for Breeders and Scientists</h3>
<p>The BrAPI technical specification document is meant to be read and used by software developers. However, the purpose of the specification, and the community around it, is to make things faster, easier, and cheaper for the breeders and scientists working on breeding, genetics, phenomics and other agricultural projects. BrAPI offers a convenient path to automation, interoperability, and data integration for software tools in those domains. All of the software described above could be made interoperable with manual effort, moving and editing data files by hand from tool to tool. However, when the basic structure and flow of data becomes automated, breeders and scientists can spend less time on data management and more time focusing on the science. For many, the ultimate goal is the development of a digital ecosystem: a collection of software tools and applications that can all work together seamlessly. In this ideal digital ecosystem, data is collected digitally from the beginning, reducing as much human error as possible. The data is checked by quality control and stored automatically, then it can be sent to any internal tool or external lab for further analysis with just the click of a button. This idea might sound too good to be true, but as more tools start sharing a universal data standard, automating data flow becomes easier, and the community gets closer to total interoperability.</p>
<h3 id="looking-ahead">Looking Ahead</h3>
<p>The BrAPI specification will continue a controlled and necessary growth. It will enable more use cases and new types of data that might include newer scientific techniques and technologies. There will be careful evaluation the relevance of using REST web service approaches, and in particular BrAPI, for things like drone imaging data, spectroscopy, LIDAR, metabolomics, transcriptomics, high-throughput phenotyping, pan genomes, and machine learning analysis. All of these technologies can open new avenues for research and development of new crop varieties. All of these technologies also generate more data, and require data sharing between different software applications and data repositories. The BrAPI project leadership and community is committed to building the standards to support these new use cases as they arrive and become accepted by the scientific community. In fact, small groups within the BrAPI community have already start building generic data models and proposed communication standards for many of the technologies listed above. These community efforts will eventually become part of the BrAPI standard in a future version of the specification document.</p>
<p>Expanding the BrAPI specification is important for the community, however it is just as important not to reinvent or compete with existing functional standards. Additions to the BrAPI specification are reviewed thoroughly by the community to make sure BrAPI is compliant with existing standards and data structures. For example, the community has requested compliance with the GFF3 standard for genomic data and the GeoTIFF standard for aerial image data. Pieces of these existing popular data structures might be integrated into the overall BrAPI standard documentation. In some cases, BrAPI will only reference other standards instead of including them in the specification. For example, there have been community discussions around developing connections with the NOAA CDO standard for weather data, or the Galaxy Analytics API for analytics pipeline controls and information. These standards are perfectly adequate on their own, recreating them in the BrAPI standard would be redundant and outside the main scope of the project.</p>
<h3 id="conclusion">Conclusion</h3>
<p>The BrAPI project only exists because of the community of software engineers, biologists, and other scientists who support and use it. While there were many tools and use cases presented here, it is not an exhaustive list of all BrAPI compliant systems. As long as the standard continues to be supported, there is potential for exponential growth of the community. As more groups make their tools BrAPI compliant, these tools can be shared with the community. As more BrAPI compliant tools are shared with the community, more groups can see the value in implementing BrAPI in their own tools. This feedback loop will allow the community to strengthen and grow.</p>
<p>If this manuscript is your first introduction to the BrAPI project, the authors invite you to join the community. More information is always available at <a href="https://brapi.org">brapi.org</a>.</p>
<h2 id="methods">Methods</h2>
<p>The BrAPI Project day-to-day operations are run by the BrAPI Project Coordinator, overseen by the grant PI and Co-PI. The BrAPI Project Coordinator is responsible for the general maintenance of the project infrastructure, as well as coordinating updates to the standard. They organize community events and encourage collaboration between community groups. Long term planning and organization is handled by the BrAPI Advisory board. This is a board of 6 community members who are elected by the community to represent their interest in the project.</p>
<p>The standard documentation is stored and maintained in a GitHub repository. The core documentation is written using the OpenAPI 3 documentation standard, and can be easily transformed into other documentation standards as needed. Copies of the core documentation is published on the API documentation sites Apiary and SwaggerHub for maximum accessability by the community. All these online locations are public, and the specification has a standard MIT open source license.</p>
<p>New versions of the BrAPI standard are developed periodically based on community demand. Problems and enhancements are collected by the community over time, tracked in GitHub issues, then implemented together to create a new stable version. This ensures the standard stays up-to-date and relevant to the community, but maintains a certain level of stability over time. Minor version updates are designed to be backward compatible to further enhance the stability of the standard.</p>
<p>As discussed above, hackathon events are the primary way the BrAPI community stays together. One in-person and one virtual event per year maintain project momentum and social comradery. The hackathons provide a dedicated time to discuss issues and enhancements with the specification, particular use cases, and generally how the community is run and organized.</p>
<h2 class="page_break_before" id="data-availability-statement">Data Availability Statement</h2>
<p>No data was generated from this project.</p>
<h2 id="code-availability-statement">Code Availability Statement</h2>
<p>The all of software and documentation generated by the BrAPI project is free, public, and open source. Everything is available under the MIT open source license, and can be accessed through <a href="https://brapi.org">brapi.org</a>. Most of the other software projects referenced and discussed here are also open source and the text contain links to their repositories or project pages. However, the authors take no responsibility for their availability or licensing.</p>
<h2 id="competing-interests-statement">Competing Interests Statement</h2>
<p>The authors declare no competing interests.</p>
<h2 id="acknowledgements">Acknowledgements</h2>
<p>The authors would like to acknowledge the following funding sources:</p>
<ul>
<li>The BrAPI Project is funded by the USDA grant NIFA-DSFAS 2022-67021-37024.</li>
<li>Innovation Lab for Crop Improvement Cornell, FFAR</li>
<li>The AGENT project is funded by the European Union’s Horizon 2020 research and innovation programme under grant agreement no. 862613.</li>
<li>USDA National Institute of Food and Agriculture National Research Support Project 10</li>
<li>USDA NIFA: 2022-51181-38449</li>
<li>This work was supported by the Agence Nationale de la Recherche, programme Investissements d’avenir, ANR-11-INBS-0012 (Phenome)<br />
</li>
<li>U.S. Department of Agriculture, under agreement numbers [8062-21000-043-004-A, 8062-21000-052-002-A, and 8062-21000-052-003-A]</li>
<li>The Bill and Malinda Gates Foundation in cooperation with the Excellence in Breeding Platform of the CGIAR</li>
<li>This work was funded by ELIXIR, the research infrastructure for life science data.</li>
<li>This work was supported by ELIXIR, the research infrastructure for life science data.</li>
<li>The French Networks of Biological Resource Centres for Agricultural, Environmental and Life Sciences, doi: 10.15454/b4ec-tf49</li>
</ul>
<h2 class="page_break_before" id="author-contributions">Author Contributions</h2>
<ul>
<li><strong>Peter Selby</strong>: Peter wrote and edited a significant portion of the manuscript. Peter has been the BrAPI Project Coordinator since 2017, managing the day-to-day operations of the BrAPI project.</li>
<li><strong>Rafael Abbeloos</strong>: Rafael wrote the section on the PIPPA project, and has made significant contributions to the BrAPI compatibility of the PIPPA project.</li>
<li><strong>Anne-Francoise Adam-Blondon</strong>: Anne-Francoise has made significant contributions to the Florilège, FAIDARE and AGENT projects.</li>
<li><strong>Francisco J. Agosto-Pérez</strong>: Francisco wrote the section on the ImageBreed project, and has made significant contributions to the BrAPI compatibility of the ImageBreed project.</li>
<li><strong>Michael Alaux</strong>: Michael has made significant contributions to the AGENT project.</li>
<li><strong>Isabelle Alic</strong>: Isabelle wrote the section on the PHIS project, and has made significant contributions to the BrAPI compatibility of the PHIS project.</li>
<li><strong>Khaled Al-Shamaa</strong>: Khaled wrote the section on the QBMS project, and has made significant contributions to the BrAPI compatibility of the QBMS project.</li>
<li><strong>Johan Steven Aparicio</strong>: Johan wrote the section on the Mr.Bean project, and has made significant contributions to the BrAPI compatibility of the Mr.Bean project.</li>
<li><strong>Jan Erik Backlund</strong>: Jan Erik has made significant contributions to the BMS project.</li>
<li><strong>Aldrin Batac</strong>: Aldrin has made significant contributions to the BMS project.</li>
<li><strong>Sebastian Beier</strong>: Sebastian wrote the section on the DivBrowse project, and has made significant contributions to the BrAPI compatibility of the DivBrowse project.</li>
<li><strong>Gabriel Besombes</strong>: Gabriel has made significant contributions to the PHIS project.</li>
<li><strong>Alice Boizet</strong>: Alice wrote the section on the Florilège projects, and has made significant contributions to the BrAPI compatibility of the Florilège, Gigwa and ShinyBrAPPs projects.</li>
<li><strong>Matthijs Brouwer</strong>: Matthijs has made significant contributions to the AGENT project.</li>
<li><strong>Terry Casstevens</strong>: Terry has made significant contributions to the PHG project.</li>
<li><strong>Arnaud Charleroy</strong>: Arnaud has made significant contributions to the PHIS project.</li>
<li><strong>Keo Corak</strong>: Keo wrote the section on the SCT project, and has made significant contributions to the BrAPI compatibility of the SCT project.</li>
<li><strong>Chaney Courtney</strong>: Chaney wrote the section on the Field Book project, and has made significant contributions to the BrAPI compatibility of the Field Book project.</li>
<li><strong>Mariano Crimi</strong>: Mariano has made significant contributions to the BMS project.</li>
<li><strong>Gouripriya Davuluri</strong>: Gouripriya has made significant contributions to the AGENT project.</li>
<li><strong>Kauê de Sousa</strong>: Kauê wrote the section on the ClimMob project, and has made significant contributions to the BrAPI compatibility of the ClimMob project.</li>
<li><strong>Jeremy Destin</strong>: Jeremy has made significant contributions to the FAIDARE project.</li>
<li><strong>Stijn Dhondt</strong>: Stijn has made significant contributions to the PIPPA project.</li>
<li><strong>Ajay Dhungana</strong>: Ajay wrote the section introductions for the Phenotyping, Genotyping, and Germplasm topics.</li>
<li><strong>Bert Droesbeke</strong>: Bert wrote the section on the BrAPI2ISA project, and has made significant contributions to the BrAPI compatibility of the BrAPI2ISA project.</li>
<li><strong>Manuel Feser</strong>: Manuel wrote the section on the MIRA project, and has made significant contributions to the BrAPI compatibility of the MIRA project.</li>
<li><strong>Mirella Flores-Gonzalez</strong>: Mirella wrote the section on the Trait Selector BrAPP projects, and has made significant contributions to the BrAPI compatibility of the Trait Selector BrAPP and Breedbase projects.</li>
<li><strong>Valentin Guignon</strong>: Valentin wrote the section on the BrAPI Mapper and MGIS projects, and has made significant contributions to the BrAPI compatibility of the BrAPI Mapper, MGIS, and the Trait Selector BrAPP projects.</li>
<li><strong>Corina Habito</strong>: Corina wrote the section on the BMS project, and has made significant contributions to the BrAPI compatibility of the BMS project.</li>
<li><strong>Asis Hallab</strong>: Asis wrote the section on the Zendro project, and has made significant contributions to the BrAPI compatibility of the Zendro project.</li>
<li><strong>Puthick Hok</strong>: Puthick has made significant contributions to the DArT Sample Submission project.</li>
<li><strong>Lynn Carol Johnson</strong>: Lynn wrote the section on the PHG project, and has made significant contributions to the BrAPI compatibility of the PHG project.</li>
<li><strong>Sook Jung</strong>: Sook wrote the section on the BIMS project, and has made significant contributions to the BrAPI compatibility of the BIMS project.</li>
<li><strong>Paul Kersey</strong>: Paul has made significant contributions to the AGENT project.</li>
<li><strong>Andrzej Kilian</strong>: Andrzej has made significant contributions to the DArT Sample Submission and DArTView projects.</li>
<li><strong>Patrick König</strong>: Patrick has made significant contributions to the AGENT project.</li>
<li><strong>Suman Kumar</strong>: Suman has made significant contributions to the AGENT project.</li>
<li><strong>Josh Lamos-Sweeney</strong>: Josh wrote the section on the G-Crunch project, and has made significant contributions to the BrAPI compatibility of the G-Crunch project.</li>
<li><strong>Laszlo Lang</strong>: Laszlo has made significant contributions to the Zendro project.</li>
<li><strong>Matthias Lange</strong>: Matthias wrote the section on the AGENT project, and has made significant contributions to the BrAPI compatibility of the AGENT project.</li>
<li><strong>Marie-Angélique Laporte</strong>: Marie-Angélique wrote the section on the ClimMob project, and has made significant contributions to the BrAPI compatibility of the ClimMob project.</li>
<li><strong>Taein Lee</strong>: Taein has made significant contributions to the BIMS project.</li>
<li><strong>Erwan Le-Floch</strong>: Erwan has made significant contributions to the AGENT project.</li>
<li><strong>Francisco López</strong>: Francisco has made significant contributions to the GLIS project.</li>
<li><strong>Brandon Madriz</strong>: Brandon wrote the section on the ClimMob project, and has made significant contributions to the BrAPI compatibility of the ClimMob project.</li>
<li><strong>Dorrie Main</strong>: Dorrie has made significant contributions to the BIMS project.</li>
<li><strong>Marco Marsella</strong>: Marco wrote the section on the GLIS project, and has made significant contributions to the BrAPI compatibility of the GLIS project.</li>
<li><strong>Maud Marty</strong>: Maud has made significant contributions to the FAIDARE project.</li>
<li><strong>Célia Michotey</strong>: Célia has made significant contributions to the Florilège and FAIDARE projects.</li>
<li><strong>Zachary Miller</strong>: Zachary has made significant contributions to the PHG project.</li>
<li><strong>Iain Milne</strong>: Iain has made significant contributions to the Flapjack, Gridscore, and Germinate projects.</li>
<li><strong>Lukas A. Mueller</strong>: Lukas wrote the section on the Breedbase project, and has made significant contributions to the BrAPI compatibility of the Breedbase project. Lukas is also Co-PI on the current BrAPI Grant and has supported the BrAPI project since its inception.</li>
<li><strong>Moses Nderitu</strong>: Moses wrote the section on the DArTView project, and has made significant contributions to the BrAPI compatibility of the DArTView project.</li>
<li><strong>Pascal Neveu</strong>: Pascal has made significant contributions to the PHIS project.</li>
<li><strong>Nick Palladino</strong>: Nick has made significant contributions to the DeltaBreed project.</li>
<li><strong>Tim Parsons</strong>: Tim has made significant contributions to the DeltaBreed project.</li>
<li><strong>Cyril Pommier</strong>: Cyril wrote the section on the FAIDARE projects, and has made significant contributions to the BrAPI compatibility of the Florilège, FAIDARE, and AGENT projects. Cyril has also made significant contributions to the Germplasm and Phenotyping modules of the BrAPI specification.</li>
<li><strong>Jean-François Rami</strong>: Jean-François wrote the section on the ShinyBrAPPs project, and has made significant contributions to the BrAPI compatibility of the ShinyBrAPPs project.</li>
<li><strong>Sebastian Raubach</strong>: Sebastian wrote the section on the GridScore and Germinate projects, and has made significant contributions to the BrAPI compatibility of the GridScore, Germinate, and Flapjack projects.</li>
<li><strong>Trevor Rife</strong>: Trevor wrote the section on the Field Book project, and has made significant contributions to the BrAPI compatibility of the Field Book project.</li>
<li><strong>Kelly Robbins</strong>: Kelly is the PI for the BrAPI Project grant, and has supported all technical and scientific aspects of the project.</li>
<li><strong>Mathieu Rouard</strong>: Mathieu wrote the section on the MGIS and GIGWA projects, and has made significant contributions to the BrAPI compatibility of the MGIS and GIGWA projects.</li>
<li><strong>Joseph Ruff</strong>: Joseph has made significant contributions to the AGENT project.</li>
<li><strong>Guilhem Sempéré</strong>: Guilhem wrote the section on the Gigwa project, and has made significant contributions to the BrAPI compatibility of the Gigwa project.</li>
<li><strong>Romil Mayank Shah</strong>: Romil has made significant contributions to the SCT project.</li>
<li><strong>Paul Shaw</strong>: Paul wrote the section on the Helium project, and has made significant contributions to the BrAPI compatibility of the Helium project.</li>
<li><strong>Becky Smith</strong>: Becky has made significant contributions to the Flapjack project.</li>
<li><strong>Nahuel Soldevilla</strong>: Nahuel has made significant contributions to the BMS project.</li>
<li><strong>Anne Tireau</strong>: Anne has made significant contributions to the PHIS project.</li>
<li><strong>Clarysabel Tovar</strong>: Clarysabel has made significant contributions to the BMS project.</li>
<li><strong>Grzegorz Uszynski</strong>: Grzegorz wrote the section on the DArT Sample Submission project, and has made significant contributions to the BrAPI compatibility of the DArT Sample Submission project.</li>
<li><strong>Vivian Bass Vega</strong>: Vivian has made significant contributions to the Zendro project.</li>
<li><strong>Stephan Weise</strong>: Stephan has made significant contributions to the AGENT project.</li>
<li><strong>Shawn C. Yarnes</strong>: Shawn wrote the section on the DeltaBreed project, and has made significant contributions to the BrAPI compatibility of the DeltaBreed project.</li>
</ul>
<h2 class="page_break_before" id="references">References</h2>
<!-- Auto generated references here -->
<div id="refs" class="references csl-bib-body" data-line-spacing="2" role="doc-bibliography">
<div id="ref-QfGEJ6km" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Selby, P. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btz190">BrAPI—an application programming interface for plant breeding applications</a>. <em>Bioinformatics</em> <strong>35</strong>, 4147–4155 (2019).</div>
</div>
<div id="ref-MlqG0w6i" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Rife, T. W. &amp; Poland, J. A. <a href="https://doi.org/10.2135/cropsci2013.08.0579">Field Book: An Open‐Source Application for Field Data Collection on Android</a>. <em>Crop Science</em> <strong>54</strong>, 1624–1627 (2014).</div>
</div>
<div id="ref-Ne0E9HbC" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Raubach, S., Schreiber, M. &amp; Shaw, P. D. <a href="https://doi.org/10.1186/s12859-022-04755-2">GridScore: a tool for accurate, cross-platform phenotypic data collection and visualization</a>. <em>BMC Bioinformatics</em> <strong>23</strong>, (2022).</div>
</div>
<div id="ref-LCpCba21" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Quirós, C. <em>et al.</em> <a href="https://doi.org/10.1016/j.compag.2023.108539">ClimMob: Software to support experimental citizen science in agriculture</a>. <em>Computers and Electronics in Agriculture</em> <strong>217</strong>, 108539 (2024).</div>
</div>
<div id="ref-Dhlmsxng" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Morales, N. <em>et al.</em> <a href="https://doi.org/10.1002/ppj2.20004">ImageBreed: Open‐access plant breeding web–database for image‐based phenotyping</a>. <em>The Plant Phenome Journal</em> <strong>3</strong>, (2020).</div>
</div>
<div id="ref-3TbA7PwE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Neveu, P. <em>et al.</em> <a href="https://doi.org/10.1111/nph.15385">Dealing with multi‐source and multi‐scale information in plant phenomics: the ontology‐driven Phenotyping Hybrid Information System</a>. <em>New Phytologist</em> <strong>221</strong>, 588–601 (2018).</div>
</div>
<div id="ref-kWLsm7g6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Papoutsoglou, E. A. <em>et al.</em> <a href="https://doi.org/10.1111/nph.16544">Enabling reusability of plant phenomic datasets with MIAPPE 1.1</a>. <em>New Phytologist</em> <strong>227</strong>, 260–273 (2020).</div>
</div>
<div id="ref-HQFMUyfV" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline"><a href="https://hal.inrae.fr/hal-04638310">https://hal.inrae.fr/hal-04638310</a>.</div>
</div>
<div id="ref-okf3YI9z" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">PIPPA - PSB Interface for Plant Phenotype Analysis. <a href="https://pippa.psb.ugent.be/">https://pippa.psb.ugent.be/</a>.</div>
</div>
<div id="ref-r84WbX0P" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">WIWAM – Automated systems for plant phenotyping – WIWAM presents phenotyping platforms for plant phenotype analysis. <a href="https://www.wiwam.be/">https://www.wiwam.be/</a>.</div>
</div>
<div id="ref-6DjakjNS" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">Wilkinson, M. D. <em>et al.</em> <a href="https://doi.org/10.1038/sdata.2016.18">The FAIR Guiding Principles for scientific data management and stewardship</a>. <em>Sci Data</em> <strong>3</strong>, (2016).</div>
</div>
<div id="ref-PzHxvkzH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline">Danecek, P. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btr330">The variant call format and VCFtools</a>. <em>Bioinformatics</em> <strong>27</strong>, 2156–2158 (2011).</div>
</div>
<div id="ref-186zclKvK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline"><a href="https://github.com/ga4gh-metadata/SchemaBlocks">ga4gh-metadata/SchemaBlocks</a>. GA4GH Metadata Schema Development Team (2023).</div>
</div>
<div id="ref-Wz8l0HlN" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline">Milne, I. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btq580">Flapjack—graphical genotype visualization</a>. <em>Bioinformatics</em> <strong>26</strong>, 3133–3134 (2010).</div>
</div>
<div id="ref-9fRbvOj2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline">König, P. <em>et al.</em> <a href="https://doi.org/10.1093/gigascience/giad025">DivBrowse—interactive visualization and exploratory data analysis of variant call matrices</a>. <em>GigaScience</em> <strong>12</strong>, (2022).</div>
</div>
<div id="ref-AVjkjraw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline">Sempéré, G. <em>et al.</em> <a href="https://doi.org/10.1093/gigascience/giz051">Gigwa v2—Extended and improved genotype investigator</a>. <em>GigaScience</em> <strong>8</strong>, (2019).</div>
</div>
<div id="ref-bvSFyrtt" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline">Rouard, M. <em>et al.</em> <a href="https://doi.org/10.1002/ppp3.10187">A digital catalog of high‐density markers for banana germplasm collections</a>. <em>Plants People Planet</em> <strong>4</strong>, 61–67 (2021).</div>
</div>
<div id="ref-VFIpO8nU" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline">Bradbury, P. J. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btac410">The Practical Haplotype Graph, a platform for storing and using pangenomes for imputation</a>. <em>Bioinformatics</em> <strong>38</strong>, 3698–3702 (2022).</div>
</div>
<div id="ref-S1uQPBPG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline">Engels, J. M. M. &amp; Ebert, A. W. <a href="https://doi.org/10.3390/plants10081557">A Critical Review of the Current Global Ex Situ Conservation System for Plant Agrobiodiversity. I. History of the Development of the Global System in the Context of the Political/Legal Framework and Its Major Conservation Components</a>. <em>Plants</em> <strong>10</strong>, 1557 (2021).</div>
</div>
<div id="ref-USun3Z8l" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline">Fu, Y. <a href="https://doi.org/10.2135/cropsci2017.01.0014">The Vulnerability of Plant Genetic Resources Conserved Ex Situ</a>. <em>Crop Science</em> <strong>57</strong>, 2314–2328 (2017).</div>
</div>
<div id="ref-i0Aj9IwD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline">Global Information System. <a href="https://glis.fao.org/glis/">https://glis.fao.org/glis/</a>.</div>
</div>
<div id="ref-31NWAHsO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline">Beier, S. <em>et al.</em> <a href="https://doi.org/10.12688/f1000research.109080.2">Recommendations for the formatting of Variant Call Format (VCF) files to make plant genotyping data FAIR</a>. <em>F1000Res</em> <strong>11</strong>, 231 (2022).</div>
</div>
<div id="ref-QRzg4OSs" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline">Adam-Blondon, A.-F. <em>et al.</em> AGENT Guidelines for dataflow. Zenodo <a href="https://doi.org/10.5281/zenodo.12625360">https://doi.org/10.5281/zenodo.12625360</a> (2024).</div>
</div>
<div id="ref-6VGwG9za" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline">Street, K. &amp; Street, K. Genebank mining with FIGS, the Focused Identification of Germplasm Strategy. <em>Unknown</em> (2017) doi:<a href="https://doi.org/10.22004/ag.econ.266624">10.22004/ag.econ.266624</a>.</div>
</div>
<div id="ref-pk5OwSG6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline">Kotni, P., van Hintum, T., Maggioni, L., Oppermann, M. &amp; Weise, S. <a href="https://doi.org/10.1093/nar/gkac852">EURISCO update 2023: the European Search Catalogue for Plant Genetic Resources, a pillar for documentation of genebank material</a>. <em>Nucleic Acids Research</em> <strong>51</strong>, D1465–D1469 (2022).</div>
</div>
<div id="ref-84Yn3FWj" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline">Ruas, M. <em>et al.</em> <a href="https://doi.org/10.1093/database/bax046">MGIS: managing banana (Musa spp.) genetic resources information and high-throughput genotyping data</a>. <em>Database</em> <strong>2017</strong>, (2017).</div>
</div>
<div id="ref-JWi7AKhJ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline">Van den houwe, I. <em>et al.</em> <a href="https://doi.org/10.1186/s43170-020-00015-6">Safeguarding and using global banana diversity: a holistic approach</a>. <em>CABI Agric Biosci</em> <strong>1</strong>, (2020).</div>
</div>
<div id="ref-ArPSsaJQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline">Jung, S. <em>et al.</em> <a href="https://doi.org/10.1093/database/baab054">The Breeding Information Management System (BIMS): an online resource for crop breeding</a>. <em>Database</em> <strong>2021</strong>, (2021).</div>
</div>
<div id="ref-VndIxnl3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline">Shrestha, R. <em>et al.</em> <a href="https://doi.org/10.1093/aobpla/plq008">Multifunctional crop trait ontology for breeders' data: field book, annotation, data discovery and semantic enrichment of the literature</a>. <em>AoB PLANTS</em> <strong>2010</strong>, (2010).</div>
</div>
<div id="ref-X22hwO1v" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline">Shaw, P. D., Graham, M., Kennedy, J., Milne, I. &amp; Marshall, D. F. <a href="https://doi.org/10.1186/1471-2105-15-259">Helium: visualization of large scale plant pedigrees</a>. <em>BMC Bioinformatics</em> <strong>15</strong>, (2014).</div>
</div>
<div id="ref-1CKLIIaug" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline">Paterson, T. &amp; Law, A. <a href="https://doi.org/10.1111/j.1365-2052.2011.02183.x">G&lt;scp&gt;enotypechecker&lt;/scp&gt;: an interactive tool for checking the inheritance consistency of genotyped pedigrees</a>. <em>Animal Genetics</em> <strong>42</strong>, 560–562 (2011).</div>
</div>
<div id="ref-1HmEHgW8h" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline">Bernád, V. <em>et al.</em> <a href="https://doi.org/10.3389/fpls.2024.1268847">Unlocking the genetic diversity and population structure of the newly introduced two-row spring European HerItage Barley collecTion (ExHIBiT)</a>. <em>Front. Plant Sci.</em> <strong>15</strong>, (2024).</div>
</div>
<div id="ref-x8DIc6zn" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline">Elsner-Gearing, F. <em>et al.</em> <a href="https://doi.org/10.1007/s10592-024-01611-z">Admixture and reproductive skew shape the conservation value of ex situ populations of the Critically Endangered eastern black rhino</a>. <em>Conserv Genet</em> <strong>25</strong>, 897–910 (2024).</div>
</div>
<div id="ref-m7yuFGsd" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline">Pommier, C. <em>et al.</em> <a href="https://doi.org/10.34133/2019/1671403">Applying FAIR Principles to Plant Phenotypic Data Management in GnpIS</a>. <em>Plant Phenomics</em> <strong>2019</strong>, (2019).</div>
</div>
<div id="ref-15XwZ2Vb0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline">Adam-Blondon, A.-F. <em>et al.</em> Mining Plant Genomic and Genetic Data Using the GnpIS Information System. in <em>Methods in Molecular Biology</em> 103–117 (Springer New York, 2016). doi:<a href="https://doi.org/10.1007/978-1-4939-6658-5_5">10.1007/978-1-4939-6658-5_5</a>.</div>
</div>
<div id="ref-al3Hd9Ml" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline">Morales, N. <em>et al.</em> <a href="https://doi.org/10.1093/g3journal/jkac078">Breedbase: a digital ecosystem for modern plant breeding</a>. <em>G3 Genes|Genomes|Genetics</em> <strong>12</strong>, (2022).</div>
</div>
<div id="ref-1FDf25dDD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline">Morales, N. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0240059">High density genotype storage for plant breeding in the Chado schema of Breedbase</a>. <em>PLoS ONE</em> <strong>15</strong>, e0240059 (2020).</div>
</div>
<div id="ref-zuq8ri4R" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline">Meuwissen, T. H. E., Hayes, B. J. &amp; Goddard, M. E. <a href="https://doi.org/10.1093/genetics/157.4.1819">Prediction of Total Genetic Value Using Genome-Wide Dense Marker Maps</a>. <em>Genetics</em> <strong>157</strong>, 1819–1829 (2001).</div>
</div>
<div id="ref-a9uPTJgO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline">Khaled Al-Shamaa, Johan Steven Aparicio, Nick &amp; icarda-git. <em>Icarda-Git/QBMS: QBMS Version 1.0.0</em>. (Zenodo, 2024). doi:<a href="https://doi.org/10.5281/zenodo.10791627">10.5281/zenodo.10791627</a>.</div>
</div>
<div id="ref-eaDrNpsp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline">Jung, S. <em>et al.</em> <a href="https://doi.org/10.1093/nar/gky1000">15 years of GDR: New data and functionality in the Genome Database for Rosaceae</a>. <em>Nucleic Acids Research</em> <strong>47</strong>, D1137–D1145 (2018).</div>
</div>
<div id="ref-dcg0274X" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline">Yu, J. <em>et al.</em> <a href="https://doi.org/10.3390/plants10122805">CottonGen: The Community Database for Cotton Genomics, Genetics, and Breeding Research</a>. <em>Plants</em> <strong>10</strong>, 2805 (2021).</div>
</div>
<div id="ref-sW9euzzP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline">Raubach, S. <em>et al.</em> <a href="https://doi.org/10.1002/csc2.20248">From bits to bites: Advancement of the Germinate platform to support prebreeding informatics for crop wild relatives</a>. <em>Crop Science</em> <strong>61</strong>, 1538–1566 (2020).</div>
</div>
<div id="ref-rPgDlCbt" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline">Shaw, P. D. <em>et al.</em> <a href="https://doi.org/10.2135/cropsci2016.09.0814">Germinate 3: Development of a Common Platform to Support the Distribution of Experimental Data on Crop Wild Relatives</a>. <em>Crop Science</em> <strong>57</strong>, 1259–1273 (2017).</div>
</div>
<div id="ref-7WXz7FUP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline">Aparicio, J. <em>et al.</em> <a href="https://doi.org/10.3389/fpls.2023.1290078">Mr.Bean: a comprehensive statistical and visualization application for modeling agricultural field trials data</a>. <em>Front. Plant Sci.</em> <strong>14</strong>, (2024).</div>
</div>
<div id="ref-KZc5ayvK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">45. </div><div class="csl-right-inline">Sansone, S.-A. <em>et al.</em> <a href="https://doi.org/10.1038/ng.1054">Toward interoperable bioscience data</a>. <em>Nat Genet</em> <strong>44</strong>, 121–126 (2012).</div>
</div>
<div id="ref-KpTrsTH1" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">46. </div><div class="csl-right-inline">Hardt, D. <em>The OAuth 2.0 Authorization Framework</em>. <a href="https://datatracker.ietf.org/doc/rfc6749/">https://datatracker.ietf.org/doc/rfc6749/</a> (2012).</div>
</div>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
    font-style: italic;
    font-weight: normal;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen()) ||
        heading.dataset.collapsed === "true"
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script on DOMContentLoaded instead of load to ensure this plugins
  // runs before other plugins
  window.addEventListener("DOMContentLoaded", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script type="module">
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
