<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Peter “BrapMan” Selby" />
  <meta name="author" content="Trevor “Cool Kid” Rife" />
  <meta name="author" content="Khaled Al-Shamaa" />
  <meta name="author" content="Isabelle Alic" />
  <meta name="author" content="Sebastian “Baz” Raubach" />
  <meta name="author" content="Paul Shaw" />
  <meta name="author" content="Iain Milne" />
  <meta name="author" content="Cyril Pommier" />
  <meta name="author" content="Becky Smith" />
  <meta name="author" content="Chaney Courtney" />
  <meta name="author" content="Zachary Miller" />
  <meta name="author" content="Terry Casstevens" />
  <meta name="author" content="Sook Jung" />
  <meta name="author" content="Taein Lee" />
  <meta name="author" content="Josh “Big Data” Lamos-Sweeney" />
  <meta name="author" content="Francisco López" />
  <meta name="author" content="Marco Marsella" />
  <meta name="author" content="Matthias Lange" />
  <meta name="author" content="Guilhem Sempéré" />
  <meta name="author" content="Stephan Weise" />
  <meta name="author" content="Patrick König" />
  <meta name="author" content="Manuel Feser" />
  <meta name="author" content="Gouripriya Davuluri" />
  <meta name="author" content="Paul Kersey" />
  <meta name="author" content="Erwan Le-Floch" />
  <meta name="author" content="Jospeh Ruff" />
  <meta name="author" content="Michael Alaux" />
  <meta name="author" content="Célia Michotey" />
  <meta name="author" content="Anne-Francoise Adam-Blondon" />
  <meta name="author" content="Jeremy Destin" />
  <meta name="author" content="Maud Marty" />
  <meta name="author" content="Suman Kumar" />
  <meta name="author" content="Matthijs Brouwer" />
  <meta name="author" content="Bert Droesbeke" />
  <meta name="author" content="Jan Erik Backlund" />
  <meta name="author" content="Aldrin Batac" />
  <meta name="author" content="Mariano Crimi" />
  <meta name="author" content="Corina Habito" />
  <meta name="author" content="Nahuel Soldevilla" />
  <meta name="author" content="Clarysabel Tovar" />
  <meta name="author" content="Sebastian Beier" />
  <meta name="author" content="Valentin Guignon" />
  <meta name="author" content="Mathieu Rouard" />
  <meta name="author" content="Asis Hallab" />
  <meta name="author" content="Rafael Abbeloos" />
  <meta name="author" content="Laszlo Lang" />
  <meta name="author" content="Vivian Bass Vega" />
  <meta name="author" content="Mirella Flores-Gonzalez" />
  <meta name="dcterms.date" content="2024-05-08" />
  <meta name="keywords" content="brapi, breeding, api, standards, community" />
  <title>BrAPI Success Stories</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta property="og:type" content="article" />
  <meta name="dc.title" content="BrAPI Success Stories" />
  <meta name="citation_title" content="BrAPI Success Stories" />
  <meta property="og:title" content="BrAPI Success Stories" />
  <meta property="twitter:title" content="BrAPI Success Stories" />
  <meta name="dc.date" content="2024-05-08" />
  <meta name="citation_publication_date" content="2024-05-08" />
  <meta property="article:published_time" content="2024-05-08" />
  <meta name="dc.modified" content="2024-05-08T18:40:02+00:00" />
  <meta property="article:modified_time" content="2024-05-08T18:40:02+00:00" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Peter &#34;BrapMan&#34; Selby" />
  <meta name="citation_author_institution" content="Cornell University" />
  <meta name="citation_author_orcid" content="0000-0001-7151-4445" />
  <meta name="citation_author" content="Trevor &#34;Cool Kid&#34; Rife" />
  <meta name="citation_author_institution" content="Clemson University" />
  <meta name="citation_author_orcid" content="0000-0002-5974-6523" />
  <meta name="citation_author" content="Khaled Al-Shamaa" />
  <meta name="citation_author_institution" content="ICARDA" />
  <meta name="citation_author_orcid" content="0000-0002-7668-3798" />
  <meta name="citation_author" content="Isabelle Alic" />
  <meta name="citation_author_institution" content="INRAE" />
  <meta name="citation_author_orcid" content="0000-0002-8961-6068" />
  <meta name="citation_author" content="Sebastian &#34;Baz&#34; Raubach" />
  <meta name="citation_author_institution" content="The James Hutton Institute" />
  <meta name="citation_author_orcid" content="0000-0001-5659-247X" />
  <meta name="citation_author" content="Paul Shaw" />
  <meta name="citation_author_institution" content="The James Hutton Institute" />
  <meta name="citation_author_orcid" content="0000-0002-0202-1150" />
  <meta name="citation_author" content="Iain Milne" />
  <meta name="citation_author_institution" content="The James Hutton Institute" />
  <meta name="citation_author_orcid" content="0000-0002-4126-0859" />
  <meta name="citation_author" content="Cyril Pommier" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, Versailles, France" />
  <meta name="citation_author_orcid" content="0000-0002-9040-8733" />
  <meta name="citation_author" content="Becky Smith" />
  <meta name="citation_author_institution" content="The James Hutton Institute" />
  <meta name="citation_author_orcid" content="0000-0002-8968-3383" />
  <meta name="citation_author" content="Chaney Courtney" />
  <meta name="citation_author_institution" content="Clemson University" />
  <meta name="citation_author" content="Zachary Miller" />
  <meta name="citation_author_institution" content="Cornell University" />
  <meta name="citation_author_orcid" content="0000-0002-5454-4527" />
  <meta name="citation_author" content="Terry Casstevens" />
  <meta name="citation_author_institution" content="Cornell University" />
  <meta name="citation_author_orcid" content="0000-0001-7602-0487" />
  <meta name="citation_author" content="Sook Jung" />
  <meta name="citation_author_institution" content="Department of Horticulture, Washington State University" />
  <meta name="citation_author_orcid" content="0000-0003-3968-2769" />
  <meta name="citation_author" content="Taein Lee" />
  <meta name="citation_author_institution" content="Department of Horticulture, Washington State University" />
  <meta name="citation_author_orcid" content="0000-0002-2000-2586" />
  <meta name="citation_author" content="Josh &#34;Big Data&#34; Lamos-Sweeney" />
  <meta name="citation_author_institution" content="Cornell University" />
  <meta name="citation_author" content="Francisco López" />
  <meta name="citation_author_institution" content="International Treaty on Plant Genetic Resources for Food and Agriculture, FAO" />
  <meta name="citation_author" content="Marco Marsella" />
  <meta name="citation_author_institution" content="International Treaty on Plant Genetic Resources for Food and Agriculture, FAO" />
  <meta name="citation_author_orcid" content="0000-0003-0334-8785" />
  <meta name="citation_author" content="Matthias Lange" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author_orcid" content="0000-0002-4316-078X" />
  <meta name="citation_author" content="Guilhem Sempéré" />
  <meta name="citation_author_institution" content="CIRAD (french agricultural research and international cooperation organization)" />
  <meta name="citation_author_institution" content="South Green Platform" />
  <meta name="citation_author_orcid" content="0000-0001-7429-2091" />
  <meta name="citation_author" content="Stephan Weise" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author_orcid" content="0000-0003-4031-9131" />
  <meta name="citation_author" content="Patrick König" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author_orcid" content="0000-0002-8948-6793" />
  <meta name="citation_author" content="Manuel Feser" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author_institution" content="Graduate School DILS, Bielefeld Institute for Bioinformatics Infrastructure (BIBI)" />
  <meta name="citation_author_orcid" content="0000-0001-6546-1818" />
  <meta name="citation_author" content="Gouripriya Davuluri" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author" content="Paul Kersey" />
  <meta name="citation_author_institution" content="Royal Botanic Gardens, Kew" />
  <meta name="citation_author" content="Erwan Le-Floch" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, Versailles, France" />
  <meta name="citation_author" content="Jospeh Ruff" />
  <meta name="citation_author_institution" content="Royal Botanic Gardens, Kew" />
  <meta name="citation_author" content="Michael Alaux" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, Versailles, France" />
  <meta name="citation_author_orcid" content="0000-0001-9356-4072" />
  <meta name="citation_author" content="Célia Michotey" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, Versailles, France" />
  <meta name="citation_author_orcid" content="0000-0003-1877-1703" />
  <meta name="citation_author" content="Anne-Francoise Adam-Blondon" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, Versailles, France" />
  <meta name="citation_author_orcid" content="0000-0002-3412-9086" />
  <meta name="citation_author" content="Jeremy Destin" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, Versailles, France" />
  <meta name="citation_author" content="Maud Marty" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, Versailles, France" />
  <meta name="citation_author" content="Suman Kumar" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author" content="Matthijs Brouwer" />
  <meta name="citation_author_institution" content="Wageningen University and Research" />
  <meta name="citation_author" content="Bert Droesbeke" />
  <meta name="citation_author_institution" content="VIB Data Core" />
  <meta name="citation_author_orcid" content="0000-0003-0522-5674" />
  <meta name="citation_author" content="Jan Erik Backlund" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author" content="Aldrin Batac" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author_institution" content="Leafnode LLC" />
  <meta name="citation_author" content="Mariano Crimi" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author" content="Corina Habito" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author" content="Nahuel Soldevilla" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author_institution" content="Leafnode LLC" />
  <meta name="citation_author" content="Clarysabel Tovar" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author_institution" content="Leafnode LLC" />
  <meta name="citation_author" content="Sebastian Beier" />
  <meta name="citation_author_institution" content="Institute of Bio- and Geosciences (IBG-4: Bioinformatics), CEPLAS, Forschungszenturm Jülich GmbH, Wilhelm Johnen Straße, 52428 Jülich, Germany" />
  <meta name="citation_author_institution" content="Bioeconomy Science Center (BioSC), Forschungszentrum Jülich GmbH, 52428 Jülich, Germany" />
  <meta name="citation_author_orcid" content="0000-0002-2177-8781" />
  <meta name="citation_author" content="Valentin Guignon" />
  <meta name="citation_author_institution" content="Bioversity International, Parc Scientifique Agropolis II, 34397 Montpellier, France" />
  <meta name="citation_author_orcid" content="0000-0003-0903-6811" />
  <meta name="citation_author" content="Mathieu Rouard" />
  <meta name="citation_author_institution" content="Bioversity International, Parc Scientifique Agropolis II, 34397 Montpellier, France" />
  <meta name="citation_author_orcid" content="0000-0003-0284-1885" />
  <meta name="citation_author" content="Asis Hallab" />
  <meta name="citation_author_institution" content="Jülich research center, Institute of Bio- and Geosciences (IBG), Bioinformatics (IBG-4) and Bingen Technical University of Applied Sciences, Germany" />
  <meta name="citation_author_orcid" content="0000-0002-2421-5431" />
  <meta name="citation_author" content="Rafael Abbeloos" />
  <meta name="citation_author_institution" content="VIB AgroIncubator" />
  <meta name="citation_author_orcid" content="0000-0002-0177-3887" />
  <meta name="citation_author" content="Laszlo Lang" />
  <meta name="citation_author_institution" content="Bingen Technical University of Applied Sciences, Berlinstraße 109, 55411 Bingen am Rhein, Germany" />
  <meta name="citation_author_orcid" content="0009-0009-8936-4532" />
  <meta name="citation_author" content="Vivian Bass Vega" />
  <meta name="citation_author_institution" content="Bingen Technical University of Applied Sciences, Berlinstraße 109, 55411 Bingen am Rhein, Germany" />
  <meta name="citation_author_orcid" content="0009-0002-2476-9888" />
  <meta name="citation_author" content="Mirella Flores-Gonzalez" />
  <meta name="citation_author_institution" content="Boyrce Thompson Institute" />
  <meta name="citation_author_orcid" content="0000-0002-7759-1617" />
  <link rel="canonical" href="https://plantbreeding.github.io/BrAPI-Manuscript2/" />
  <meta property="og:url" content="https://plantbreeding.github.io/BrAPI-Manuscript2/" />
  <meta property="twitter:url" content="https://plantbreeding.github.io/BrAPI-Manuscript2/" />
  <meta name="citation_fulltext_html_url" content="https://plantbreeding.github.io/BrAPI-Manuscript2/" />
  <meta name="citation_pdf_url" content="https://plantbreeding.github.io/BrAPI-Manuscript2/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://plantbreeding.github.io/BrAPI-Manuscript2/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://plantbreeding.github.io/BrAPI-Manuscript2/v/561dbbdb4440e81c5976844ed78611d5e89c9416/" />
  <meta name="manubot_html_url_versioned" content="https://plantbreeding.github.io/BrAPI-Manuscript2/v/561dbbdb4440e81c5976844ed78611d5e89c9416/" />
  <meta name="manubot_pdf_url_versioned" content="https://plantbreeding.github.io/BrAPI-Manuscript2/v/561dbbdb4440e81c5976844ed78611d5e89c9416/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">BrAPI Success Stories</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://plantbreeding.github.io/BrAPI-Manuscript2/v/561dbbdb4440e81c5976844ed78611d5e89c9416/">permalink</a>)
was automatically generated
from <a href="https://github.com/plantbreeding/BrAPI-Manuscript2/tree/561dbbdb4440e81c5976844ed78611d5e89c9416">plantbreeding/BrAPI-Manuscript2@561dbbd</a>
on May 8, 2024.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Peter “BrapMan” Selby</strong>
<sup><a href="#correspondence">✉</a></sup><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7151-4445">0000-0001-7151-4445</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/BrapiCoordinatorSelby">BrapiCoordinatorSelby</a>
<br>
<small>
Cornell University
· Funded by NIFA-DSFAS 2022-67021-37024
</small></p></li>
<li><p><strong>Trevor “Cool Kid” Rife</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-5974-6523">0000-0002-5974-6523</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/trife">trife</a>
<br>
<small>
Clemson University
</small></p></li>
<li><p><strong>Khaled Al-Shamaa</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-7668-3798">0000-0002-7668-3798</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/khaled-alshamaa">khaled-alshamaa</a>
<br>
<small>
ICARDA
</small></p></li>
<li><p><strong>Isabelle Alic</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-8961-6068">0000-0002-8961-6068</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/Isabelle-inrae">Isabelle-inrae</a>
<br>
<small>
INRAE
· Funded by {‘Short version’: ‘This work was supported by the project Phenome-ANR-11-INBS-0012’}; {‘OR Long version’: “This work was supported by the Agence Nationale de la Recherche, programme Investissements d’avenir, ANR-11-INBS-0012 (Phenome)”}
</small></p></li>
<li><p><strong>Sebastian “Baz” Raubach</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-5659-247X">0000-0001-5659-247X</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/sebastian-raubach">sebastian-raubach</a>
<br>
<small>
The James Hutton Institute
</small></p></li>
<li><p><strong>Paul Shaw</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-0202-1150">0000-0002-0202-1150</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/cardinalb">cardinalb</a>
<br>
<small>
The James Hutton Institute
</small></p></li>
<li><p><strong>Iain Milne</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-4126-0859">0000-0002-4126-0859</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/imilne">imilne</a>
<br>
<small>
The James Hutton Institute
</small></p></li>
<li><p><strong>Cyril Pommier</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-9040-8733">0000-0002-9040-8733</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/cpommier">cpommier</a>
<br>
<small>
Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France; Université Paris-Saclay, INRAE, URGI, Versailles, France
· Funded by {‘Short version’: ‘This work was supported by the project Phenome-ANR-11-INBS-0012’}; {‘OR Long version’: “This work was supported by the Agence Nationale de la Recherche, programme Investissements d’avenir, ANR-11-INBS-0012 (Phenome)”}
</small></p></li>
<li><p><strong>Becky Smith</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-8968-3383">0000-0002-8968-3383</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/Batbaby91">Batbaby91</a>
<br>
<small>
The James Hutton Institute
</small></p></li>
<li><p><strong>Chaney Courtney</strong>
<br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/chaneylc">chaneylc</a>
<br>
<small>
Clemson University
</small></p></li>
<li><p><strong>Zachary Miller</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-5454-4527">0000-0002-5454-4527</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/zrm22">zrm22</a>
<br>
<small>
Cornell University
</small></p></li>
<li><p><strong>Terry Casstevens</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7602-0487">0000-0001-7602-0487</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/tcasstevens">tcasstevens</a>
<br>
<small>
Cornell University
</small></p></li>
<li><p><strong>Sook Jung</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-3968-2769">0000-0003-3968-2769</a>
<br>
<small>
Department of Horticulture, Washington State University
</small></p></li>
<li><p><strong>Taein Lee</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2000-2586">0000-0002-2000-2586</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/leetaei">leetaei</a>
<br>
<small>
Department of Horticulture, Washington State University
</small></p></li>
<li><p><strong>Josh “Big Data” Lamos-Sweeney</strong>
<br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/jlamossweeney">jlamossweeney</a>
<br>
<small>
Cornell University
</small></p></li>
<li><p><strong>Francisco López</strong>
<br>
<small>
International Treaty on Plant Genetic Resources for Food and Agriculture, FAO
</small></p></li>
<li><p><strong>Marco Marsella</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-0334-8785">0000-0003-0334-8785</a>
<br>
<small>
International Treaty on Plant Genetic Resources for Food and Agriculture, FAO
</small></p></li>
<li><p><strong>Matthias Lange</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-4316-078X">0000-0002-4316-078X</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/langeipk">langeipk</a>
<br>
<small>
Leibniz Institute of Plant Genetics and Crop Plant Research
</small></p></li>
<li><p><strong>Guilhem Sempéré</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7429-2091">0000-0001-7429-2091</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/GuilhemSempere">GuilhemSempere</a>
<br>
<small>
CIRAD (french agricultural research and international cooperation organization); South Green Platform
</small></p></li>
<li><p><strong>Stephan Weise</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-4031-9131">0000-0003-4031-9131</a>
<br>
<small>
Leibniz Institute of Plant Genetics and Crop Plant Research
</small></p></li>
<li><p><strong>Patrick König</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-8948-6793">0000-0002-8948-6793</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/patrick-koenig">patrick-koenig</a>
<br>
<small>
Leibniz Institute of Plant Genetics and Crop Plant Research
</small></p></li>
<li><p><strong>Manuel Feser</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-6546-1818">0000-0001-6546-1818</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/feserm">feserm</a>
<br>
<small>
Leibniz Institute of Plant Genetics and Crop Plant Research; Graduate School DILS, Bielefeld Institute for Bioinformatics Infrastructure (BIBI)
</small></p></li>
<li><p><strong>Gouripriya Davuluri</strong>
<br>
<small>
Leibniz Institute of Plant Genetics and Crop Plant Research
</small></p></li>
<li><p><strong>Paul Kersey</strong>
<br>
<small>
Royal Botanic Gardens, Kew
</small></p></li>
<li><p><strong>Erwan Le-Floch</strong>
<br>
<small>
Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France; Université Paris-Saclay, INRAE, URGI, Versailles, France
</small></p></li>
<li><p><strong>Jospeh Ruff</strong>
<br>
<small>
Royal Botanic Gardens, Kew
</small></p></li>
<li><p><strong>Michael Alaux</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-9356-4072">0000-0001-9356-4072</a>
<br>
<small>
Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France; Université Paris-Saclay, INRAE, URGI, Versailles, France
</small></p></li>
<li><p><strong>Célia Michotey</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-1877-1703">0000-0003-1877-1703</a>
<br>
<small>
Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France; Université Paris-Saclay, INRAE, URGI, Versailles, France
</small></p></li>
<li><p><strong>Anne-Francoise Adam-Blondon</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-3412-9086">0000-0002-3412-9086</a>
<br>
<small>
Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France; Université Paris-Saclay, INRAE, URGI, Versailles, France
</small></p></li>
<li><p><strong>Jeremy Destin</strong>
<br>
<small>
Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France; Université Paris-Saclay, INRAE, URGI, Versailles, France
</small></p></li>
<li><p><strong>Maud Marty</strong>
<br>
<small>
Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France; Université Paris-Saclay, INRAE, URGI, Versailles, France
</small></p></li>
<li><p><strong>Suman Kumar</strong>
<br>
<small>
Leibniz Institute of Plant Genetics and Crop Plant Research
</small></p></li>
<li><p><strong>Matthijs Brouwer</strong>
<br>
<small>
Wageningen University and Research
</small></p></li>
<li><p><strong>Bert Droesbeke</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-0522-5674">0000-0003-0522-5674</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/bedroesb">bedroesb</a>
<br>
<small>
VIB Data Core
</small></p></li>
<li><p><strong>Jan Erik Backlund</strong>
<br>
<small>
Integrated Breeding Platform
</small></p></li>
<li><p><strong>Aldrin Batac</strong>
<br>
<small>
Integrated Breeding Platform; Leafnode LLC
</small></p></li>
<li><p><strong>Mariano Crimi</strong>
<br>
<small>
Integrated Breeding Platform
</small></p></li>
<li><p><strong>Corina Habito</strong>
<br>
<small>
Integrated Breeding Platform
</small></p></li>
<li><p><strong>Nahuel Soldevilla</strong>
<br>
<small>
Integrated Breeding Platform; Leafnode LLC
</small></p></li>
<li><p><strong>Clarysabel Tovar</strong>
<br>
<small>
Integrated Breeding Platform; Leafnode LLC
</small></p></li>
<li><p><strong>Sebastian Beier</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2177-8781">0000-0002-2177-8781</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/sebeier">sebeier</a>
<br>
<small>
Institute of Bio- and Geosciences (IBG-4: Bioinformatics), CEPLAS, Forschungszenturm Jülich GmbH, Wilhelm Johnen Straße, 52428 Jülich, Germany; Bioeconomy Science Center (BioSC), Forschungszentrum Jülich GmbH, 52428 Jülich, Germany
</small></p></li>
<li><p><strong>Valentin Guignon</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-0903-6811">0000-0003-0903-6811</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/guignonv">guignonv</a>
<br>
<small>
Bioversity International, Parc Scientifique Agropolis II, 34397 Montpellier, France
</small></p></li>
<li><p><strong>Mathieu Rouard</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-0284-1885">0000-0003-0284-1885</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/mrouard">mrouard</a>
<br>
<small>
Bioversity International, Parc Scientifique Agropolis II, 34397 Montpellier, France
</small></p></li>
<li><p><strong>Asis Hallab</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2421-5431">0000-0002-2421-5431</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/https://github.com/asishallab">https://github.com/asishallab</a>
<br>
<small>
Jülich research center, Institute of Bio- and Geosciences (IBG), Bioinformatics (IBG-4) and Bingen Technical University of Applied Sciences, Germany
</small></p></li>
<li><p><strong>Rafael Abbeloos</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-0177-3887">0000-0002-0177-3887</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/raabb">raabb</a>
<br>
<small>
VIB AgroIncubator
</small></p></li>
<li><p><strong>Laszlo Lang</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0009-0009-8936-4532">0009-0009-8936-4532</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/LzLang">LzLang</a>
<br>
<small>
Bingen Technical University of Applied Sciences, Berlinstraße 109, 55411 Bingen am Rhein, Germany
</small></p></li>
<li><p><strong>Vivian Bass Vega</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0009-0002-2476-9888">0009-0002-2476-9888</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/VivianBass">VivianBass</a>
<br>
<small>
Bingen Technical University of Applied Sciences, Berlinstraße 109, 55411 Bingen am Rhein, Germany
</small></p></li>
<li><p><strong>Mirella Flores-Gonzalez</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-7759-1617">0000-0002-7759-1617</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/mflores2021">mflores2021</a>
<br>
<small>
Boyrce Thompson Institute
</small></p></li>
</ul>
<div id="correspondence">
<p>✉ — Correspondence possible via <a href="https://github.com/plantbreeding/BrAPI-Manuscript2/issues">GitHub Issues</a>
or email to
Peter “BrapMan” Selby &lt;ps664@cornell.edu&gt;.</p>
</div>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<p>Population growth and climate change require extraordinary efforts to increase efficiency in breeding programs around the world. In the last few years, genomics technologies and genomic prediction approaches have provided a boost in genetic gain in breeding, but has also created a flood of data that needs careful management to be fully harnessed.
The Breeding API (BrAPI) project is an international, grass-roots effort to enable more efficient data management by enabling interoperability among plant breeding databases and tools, using a standardized RESTful web service API specification for communicating plant breeding data.
This community driven standard is free to be used by anyone interested in plant breeding data management, including trial data, phenotypic data and genotyping data management.
This manuscript describes advances in implementations of BrAPI in different breeding tools, and outlook for the current version of BrAPI.</p>
<h2 id="introduction">Introduction</h2>
<!-- 
Notes:
* DIGITAL ECOSYSTEM
* Justification for a second paper
* Highlight Community Growth
* Hackathons
* sabbaticals
* Updates for V2
* Auth Data Access Control - super simplified - highlight data privacy and security elements
* Project structure - Modules
* Standardization of endpoints across objects - said in a non-tech way
-->
<p>To address consequences of climate change and population growth, plant and animal breeding need to become more efficient and data driven to produce bigger, better, healthier, more sustainable crops. Modern breeding techniques require large amounts of high quality data to be effective, requiring digital methods for data collection, management, and analysis. Interoperability between breeding software tools, systems, and databases can substantially increase the efficiency of a breeding program. The ability to efficiently share data means access to larger and more complete datasets, enabling to build more accurate computational models and produce more accurate predictions and improved selections.</p>
<p>Plant and animal breeding is an incredibly important part of today’s society. Almost every country in the world has some kind of breeding program supporting the agricultural community to produce bigger, better, healthier, and more sustainable crops. Climate change, population growth, disease mitigation, and nutritional health are all major challenges facing humanity, and the ability to breed better crops is a critical part of the solution. Modern breeding techniques require large amounts of high quality data to be effective. In the digital age, that breeding data is being collected, managed, and analyzed with computer software. Interoperability between breeding software tools, systems, and databases can substantially increase the efficiency of a breeding program. The ability to share tools gives each program a boost in computational power. The ability to share data allows the community to leverage access to larger, more complete, datasets; similarly, the community can build more accurate computational models and produce more accurate predictions.</p>
<p>The Breeding API (BrAPI) project is an effort to enable interoperability among breeding tools, systems, and databases. BrAPI is a standardized Representational State Transfer (REST), web service, Application Programming Interface (API), specification for breeding and related agricultural data. <span class="citation" data-cites="QfGEJ6km">[<a href="#ref-QfGEJ6km" role="doc-biblioref">1</a>]</span> By using the BrAPI standard, breeding software can more easily become interoperable, allowing groups to more easily share data and software tools.</p>
<p>This manuscript will present BrAPI in its latest iteration. This includes a technical description of the standard and a showcase of the applications, services, and tools available in the BrAPI community. BrAPI has become an essential part of the digital infrastructure for breeding applications and related agricultural projects. It is the intention of this manuscript to demonstrate the value of BrAPI to the wider scientific community as an effective and efficient means to collaborate and share resources.</p>
<h3 id="how-it-works">How it works</h3>
<p>An Application Programming Interface (API) is a technical connection between two pieces of software. Just as a Graphical User Interface (GUI) or a Command Line Interface (CLI) allows a human user to interact with a piece of software, an API allows one software application to interact with another. A GUI or CLI might allow a user to input data, read data, and start processes within an application. An API allows one piece of software (sometimes called a client, user agent, or service consumer) to programmatically input data, read data, and start process within another piece of software (sometimes called a server or service provider).</p>
<p>A Representational State Transfer (REST) web service is a type of API commonly used in today’s modern web infrastructure. REST is a technical architecture that describes the stateless transmission of data between applications. Typically, REST systems are implemented using the standard HTTP protocol that most of the modern internet is built upon. REST implementations also generally use JavaScript Object Notation (JSON) to represent the data being transferred. Both HTTP and JSON are programming language agnostic, very stable, and very flexible. This means BrAPI can be implemented in almost any piece of software, and can solve a wide range of use cases.</p>
<p>Data repositories and service providers can choose to represent their data as a BrAPI compatible API. By mapping the internal data structures to the standard models, data repositories can easily expose data to the outside world. Similarly, they can accept new data from external sources and automatically map the new data into the existing database. Client application developers can take advantage of this standardization by building tools that can easily integrate with all other BrAPI compatible data repositories. Visualization, reporting, analytics, data collection, and quality control tools can be built once and shared with other organizations following the standards. As the number of BrAPI compatible databases, tools, and organizations grows, so does the value added by implementing the standard into a given application.</p>
<h3 id="project-updates">Project Updates</h3>
<p>Over its lifetime, the BrAPI project has grown and changed substantially. The latest stable version of the specification (v2.1) looks vastly different from the original version (v1.0) released in 2017. The total size of the specification has almost quadrupled in that time, going from 51 endpoints documented in v1.0 to 201 endpoints documented in v2.1. Because of this growth, the specification documents were reorganized into four modules: BrAPI-Core, BrAPI-Germplasm, BrAPI-Genotyping, and BrAPI-Phenotyping. Figure <a href="#fig:domains">1</a> shows a simplified domain map of the whole BrAPI v2.1 data model, divided into the organizational modules. The early versions of the specification focused on read-only phenotype data, with a small consideration to the other domains. Now the specification has a full representation of most of the major concepts applicable to the breeding process. The new specification is also internally consistent, easier to navigate, and allows for read, write, and update capabilities. None of those qualities were a guarantee for the earlier versions.</p>
<div id="fig:domains" class="fignos">
<figure>
<img src="images/BrAPI_Domains_v2-1_vertical.png" style="width:100.0%" alt="Figure 1: A simplified domain map of the whole BrAPI data model, divided into organizational modules. A more detailed Entity Relationship Diagram (ERD) is available on brapi.org." />
<figcaption aria-hidden="true"><span>Figure 1:</span> A simplified domain map of the whole BrAPI data model, divided into organizational modules. A more detailed Entity Relationship Diagram (ERD) is available on brapi.org.</figcaption>
</figure>
</div>
<p>As the specification has matured, so have the tools, services, and libraries available to the community to work with the specification. Every version of the specification is now released with a change log to guide developers upgrading from a previous version, an Entity Relationship Diagram (ERD) to describe the whole data model visually, and a JSON Schema version of the model to be used in some automated development efforts. For groups who are using Java, Java Script, Python, R, or Drupal, there are community maintained libraries available that contain full BrAPI implementations ready to be added to some existing code. The BrAPI Test Server and the BRAVA validation tool are both still available to the community for testing purposes, and they have been maintained to support every version of the specification. Finally, the three new resources list pages on brapi.org advertise the other BrAPI compatible software available in the community. The BrAPPs list displays the 10 standalone, plug-and-play, applications available to the community. The servers list displays the 27 registered public data servers, their current status, and a form for registering additional servers. The compatible software list shows the 31 software applications that are BrAPI compliant, and again, a form for registering additional applications. The process for registering new items for these lists is completely voluntary, so the totals represent a lower bound for the number of BrAPPs, data repositories, and applications available in the the community.</p>
<h3 id="community-growth">Community Growth</h3>
<p>The international BrAPI Community consists of software developers, breeders, and related scientists working on BrAPI related projects and data sources. This community is what sustains the BrAPI project, building implementations, maintaining development tools, and providing input to enhance the specification. As the project has grown, so has the community. The BrAPI project started in June 2014 with less than ten people coming together to discuss the idea. Over the next nine years, the community has grown to between 200 and 250 members. The community mailing list has 208 members, and the BrAPI Slack workspace has 234 members. The project leadership uses the mailing list to broadcast newsletters, announcements, and updates to the community. The BrAPI Slack workspace allows members of the community to discuss specific topics and collaborate directly with each other.</p>
<p>The BrAPI Hackathons are a major staple of the BrAPI community. Twice a year, the community gathers to discuss the specification and collaborate on BrAPI related projects. This time is very valuable to the community; for some organizations, the hackathon is the only time during the year when they have time to work on anything related to BrAPI. During the COVID-19 pandemic, virtual hackathons took the place of in person events. While the virtual hackathons do not provide the same level of face-to-face time that is crucial to collaborative work, they did allow for more attendees to gather and share their opinions. The typical virtual hackathon has about twice as many registered attendees compared to an in-person hackathon. However, attendees have reported much more productive work time during the in-person events. As a compromise, going forward, the community leadership has decided to have one in-person hackathon and one virtual hackathon each year, to balance the advantages of both.</p>
<p>As the project has matured, a formal project leadership structure became increasingly important. As described in Figure <a href="#fig:org-chart">2</a>, the project governance is divided into two groups. The Project Management team is responsible for the day to day operations of the project. The PI and Co-PI are responsible for the project funds, and responsible for hiring the BrAPI Project Coordinator who is paid by the project funds. The Advisory Board is a group of elected officials representing the community. The board is responsible for long term planning of the project, as well as quick decision making on behalf of the community. The two groups meet quarterly to report on progress and stay synchronized.</p>
<div id="fig:org-chart" class="fignos">
<figure>
<img src="images/BrAPI_org_structure.jpg" style="width:100.0%" alt="Figure 2: The formal governance structure of the BrAPI Project Community" />
<figcaption aria-hidden="true"><span>Figure 2:</span> The formal governance structure of the BrAPI Project Community</figcaption>
</figure>
</div>
<h2 id="success-stories">Success Stories</h2>
<!-- success stories highlighting BrAPI usefulness in breeding cycle. Perhaps reference the original BrAPI paper where possible use cases were proposed.  -->
<p>Below are a number of short success stories from the BrAPI community. These tools, applications, and infrastructure projects serve as another indicator of community growth and success over the past 5-10 years. These stories clearly illustrate all the different ways the BrAPI Standard can be used productively and in practice.</p>
<h3 id="phenotyping">Phenotyping</h3>
<p>Phenotyping is the basis for all breeding efforts. All the downstream analysis and decision making procedures rely on a foundation of accurate, high-quality, phenotypic data. The BrAPI specification supports phenotypic data as it passes through the entire breeding pipeline, as it is published, and as it is archived. The community has developed tools that use BrAPI during the data collection process, to get the data into a standard format as soon as possible. There are BrAPI compatible systems to store and curate phenotypic data, and tools to manage trait meta data. The community has also begun developing BrAPI compatible tools for managing images and other high throughput phenotyping techniques.</p>
<h4 id="field-book">Field Book</h4>
<!-- Peter S: Stub paragraph to stimulate the writing process. Please edit, rewrite, or delete as needed. -->
<p>Phenotypic data collection is an essential part of the breeding process. Historically, gathering data in the field was done with pen and paper, or perhaps some version of a digital spreadsheet. The abundance and prevalence of smart phones has allowed the Field Book mobile app to enhance data collection. Field Book can create well-formed digital observation records from the moment they are taken. This can improve the efficiency of data collection and reduce human error.</p>
<p>In 2018, BrAPI was introduced into Field Book; specifically, the Core and Phenotyping modules. BrAPI was able to take things a step further by automating the flow of data from the Field Book mobile app to a central database server. This workflow allows data collection and storage to be expedited, removing the need of the user to transfer export files manually. Since Field Book’s adoption of BrAPI, many community servers have been integrated to simplify data storage. In this work flow, data is collected and stored completely digitally with little-to-no human involvement.</p>
<h4 id="climmob">ClimMob</h4>
<p>Not all data can be collected by a single person, or even by a single organization. ClimMob is a tool to easily allow citizen scientists to assist in the data collection process. Although this data may not be as detailed as a focused scientific program, it can be very useful to collect simple data from a wide range of locations and environments.</p>
<p>When it comes to BrAPI compatibility, ClimMob follows the same patterns established by Field Book. During a survey, all the farmer collected data is stored in a central ClimMob node. When the survey is complete, all the data is uploaded automatically via BrAPI to a central breeding database for long term storage and analysis.</p>
<h4 id="imagebreed">ImageBreed</h4>
<p>High-throughput phenotyping has been gaining significant traction lately as a way to collect lots of data very quickly. Image collection from unmanned arial and ground vehicles (UAVs and UGVs) are a great way to collect a lot of raw data all at once, then analyze it later. ImageBreed is a image collection pipeline tool to support regular use of UAVs and UGVs.</p>
<p>When the raw images have been processed through the standardization pipelines in ImageBreed, useful phenotypes can be extracted from the images. The BrAPI standard is used to push these phenotypes back to a central breeding database where they can be analyzed with other data. In addition to this, ImageBreed also has the option to use BrAPI to upload the raw images to the central breeding database, or any other BrAPI compatible long term storage service. The BrAPI models in the current version of the standard (V2.1) are rudimentary, but effective. The ImageBreed team has put in some work to enhance the BrAPI image data standards.</p>
<h4 id="gridscore">GridScore</h4>
<p>Phenotypic data collection underpins scientific crop research and plant breeding. Knowledge gained from collected data and its analysis alongside data visualizations inform further phenotypic trials and ideally support research hypotheses. The importance of accuracy and efficiency in the collection of this data as well as the infrastructure to facilitate the flow of data from the field to a knowledge base cannot be underestimated. <a href="https://ics.hutton.ac.uk/get-gridscore/">GridScore</a> <span class="citation" data-cites="Ne0E9HbC">[<a href="#ref-Ne0E9HbC" role="doc-biblioref">2</a>]</span> is a modern mobile application for phenotypic observations that harnesses technological advancements in the area of mobile devices to enrich the data collection process.</p>
<p>BrAPI has further increased the value of GridScore by integrating it into the overarching workflow from trial creation, data collection, and its ultimate data storage for further processing. Specifically, trial designs as well as trait definitions can be imported into GridScore using BrAPI and a finalized trial can ultimately be exported via BrAPI to any compatible database.</p>
<h4 id="phis">PHIS</h4>
<p>The Hybrid Phenotyping Information System (<a href="http://www.phis.inrae.fr/">PHIS</a> <span class="citation" data-cites="OodVqG7n">[<a href="#ref-OodVqG7n" role="doc-biblioref">3</a>]</span>), based on the <a href="https://github.com/OpenSILEX/">OpenSILEX</a> framework, is an ontology-driven information system based on semantic web technologies. PHIS is deployed in several field and greenhouse platforms of the national <a href="https://www.phenome-emphasis.fr/">PHENOME</a> and European <a href="https://emphasis.plant-phenotyping.eu/">EMPHASIS</a> infrastructure. It manages and collects data from Phenotyping and High Throughput Phenotyping experiments on a day to day basis. PHIS unambiguously identifies all the objects and traits in an experiment, and establishes their types and relationships via ontologies and semantics.</p>
<p>PHIS has been designed to be BrAPI-compliant. PHIS adheres to the standards and protocols specified by BrAPI and implements various services aligning with the BrAPI standards, encompassing the Core, Phenotyping, and Germplasm modules. This enables integration and compatibility with BrAPI-compliant systems and platforms. This prerequisite served as the basis for formalizing the data model, while also facilitating compatibility with other standards, such as the Minimal Information About a Plant Phenotyping Experiment (<a href="https://www.miappe.org/">MIAPPE</a> <span class="citation" data-cites="TVXmHLRk">[<a href="#ref-TVXmHLRk" role="doc-biblioref">4</a>]</span>). By integrating BrAPI requirements into its structure, PHIS not only meets the standards of the phenotyping field, but also strengthens its capacity for interoperability and effective collaboration in the wider context of plant breeding and related fields.</p>
<p>The fact that data within a PHIS instance can be queried through BrAPI services makes the indexing of PHIS in <a href="https://urgi.versailles.inra.fr/faidare/">FAIDARE</a> very easy to implement.</p>
<p>Furthermore, as PHIS offers BrAPI-compliant Web Services, it simplifies the integration and data exchange with other European information systems that handle phenotyping data. The adherence to BrAPI standards ensures a common interface and compatibility, facilitating communication and collaboration between PHIS and other systems in the European context. This interoperability not only eases data sharing, but also promotes a more coherent and efficient approach to the management and use of phenotyping data on various platforms and research initiatives within the European scientific community.</p>
<h4 id="pippa">PIPPA</h4>
<p><a href="https://pippa.psb.ugent.be">PIPPA</a> is a data management system used for collecting data from the <a href="https://www.wiwam.be/">WIWAM</a> range of automated high throughput phenotyping platforms. These platforms have been deployed at different research institutes and commercial breeders across Europe in a variety of configurations with different types of equipment such as weighing scales, cameras and environment sensors. Examples are:</p>
<ul>
<li><a href="https://www.upsc.se/plant-growth-facilities-at-upsc-and-slu-umea/325-upsc-tree-phenotyping-platform.html">Umea Plant Science Centre</a></li>
<li><a href="https://cri.fmach.it/en/Facilities/Technological-Facilities/Plant-Phenotyping#application_fields">Fondazione Edmund Mach</a></li>
<li><a href="https://www.psb.ugent.be/phenotyping/phenovision">Phenovision</a></li>
</ul>
<p>Developed from 2016 onwards, the software features a web interface with functionality for setting up new experiments for the platform(s), planning imaging and irrigation treatments, linking metadata to pots (genotype, growth media, manual treatments), exporting data, importing data and visualizing data as charts. It also supports the integration of image analysis scripts and connections to a compute cluster for job submission.</p>
<p>To share the phenotype data of the experiments linked to publications, an implementation of BrAPI 1.3 was developed on a separate public PIPPA server open to the public, which allowed read only access to the data in a standardized format. This endpoint was registered on <a href="https://urgi.versailles.inra.fr/faidare/">FAIDARE</a> and allows the data to be found alongside data from other BrAPI endpoints.</p>
<p>As the BrAPI ecosystem has matured, it created a clear path for the development of PIPPA as to how to share data in a manner according to the FAIR principles which are becoming standard in plant research data management best practices. In combination with the support for <a href="https://www.miappe.org/">MIAPPE</a>, these have served as guidelines in the current development, which is focussed on delivering a public BraPI 2.1 endpoint and making more high throughput datasets publicly available via BrAPI.</p>
<h4 id="trait-selector-brapp">Trait Selector BrAPP</h4>
<!-- the "BrAPP" description part should better fit at the beginning of the section. -->
<p>BrAPPs are simple tools developed by the BrAPI community that are entirely reliant on BrAPI for their data requirements. This means a single BrAPI can be shared and used by many organizations, as long as those organizations have the standard BrAPI endpoints available.</p>
<p>The Trait Selector BrAPP is used to search and select useful traits, using a visual aid to help the user find exactly what they need. This BrAPP works with both breeding databases and genebanks. Breeding databases would need to only implement the trait, observation and observation variable calls, while genebanks would require trait, germplasm attribute and germplasm attribute value calls. So, BrAPI servers compliant with version 2 implementing any of these sets of calls would just need to follow the documented steps to create an SVG image of a plant of interest in order to use this BrAPP. CassavaBase and MGIS are two successful examples of the use of this BrAPP. (example screenshots coming + supplementary data: links to the git and the doc)</p>
<h3 id="genotyping">Genotyping</h3>
<p>Genotyping has become a cornerstone of most breeding processes, but the data can be difficult to manage. BrAPI supports genotypic data, relying on some existing standards including VCF and the GA4GH Variants standards. The BrAPI Community has built BrAPI compatible tools for storing, searching, visualizing, and analyzing genotypic data.</p>
<h4 id="flapjack">Flapjack</h4>
<p><a href="https://ics.hutton.ac.uk/flapjack">Flapjack</a> <span class="citation" data-cites="Wz8l0HlN">[<a href="#ref-Wz8l0HlN" role="doc-biblioref">5</a>]</span> is a multi-platform desktop application for data visualization and breeding analysis (eg, pedigree verification, marker-assisted backcrossing and forward breeding) using high-throughput genotype data. Data can be easily imported into Flapjack from any BrAPI compatible data source with genotype data available. <a href="https://github.com/cropgeeks/flapjack-bytes">Flapjack Bytes</a> is a smaller, lightweight and fully web-based counterpart to Flapjack, which can be easily embedded into a database website to provide similar visualizations online. Traditionally supporting its own text-based data formats, Flapjack’s use of BrAPI has streamlined the end-user experience for data import and work is underway to determine the best methods to exchange analysis results using future versions of the API.</p>
<h4 id="dartview">DArTView</h4>
<!-- Peter S: Stub paragraph to stimulate the writing process. Please edit, rewrite, or delete as needed. -->
<p>DArTView is a desktop application for visualizing genotype variant data and looking for trends or correlations. It is newly BrAPI compatible and can use BrAPI as an input data source.</p>
<h4 id="divbrowse">DivBrowse</h4>
<p>DivBrowse <span class="citation" data-cites="9fRbvOj2">[<a href="#ref-9fRbvOj2" role="doc-biblioref">6</a>]</span> is a web platform for exploratory data analysis of huge genotyping studies. The software can be run standalone or integrated as a plugin into existing data web portals. It provides a powerful interactive visualization of variant call matrices with hundreds of millions of variants and thousands of samples and enables easy data import and export by using standardized and established bioinformatics file formats.
At its core, DivBrowse combines the convenience of a genome browser and adds features tailored to the diversity analysis of germplasm. It is able to display genomic features such as nucleotide sequence, associated gene models and short genomic variants. DivBrowse provides visual access to large VCF files obtained through genotyping experiments. In addition to visualizing variant calls per variant and genotype, DivBrowse also calculates and displays variant statistics such as minor allele frequencies, proportion of heterozygous calls or missing variant calls for each visualized genomic window. In addition, dynamic Principal Component Analyses (PCAs) can be performed on a user specified genomic area to provide information on local genomic diversity.
DivBrowse has a Javascript API to control the tool from a hosting web portal (e.g. to control the list of genotypes to be displayed and the reference genome). DivBrowse has an interface to BLAST, which can be used to directly access genes or other genomic features. The modular structure of DivBrowse also allows developers to configure and easily embed links to external information systems. Furthermore, parts of BrAPI are implemented to provide genotypic data via its server-side component and is also able to consume and visualize genotypic data via an external BrAPI endpoint through the client-side GUI.</p>
<h4 id="gigwa">GIGWA</h4>
<p>Gigwa is a JEE web application providing means to centralize, share, finely filter, and visualize high-throughput genotyping data <span class="citation" data-cites="AVjkjraw">[<a href="#ref-AVjkjraw" role="doc-biblioref">7</a>]</span>. Built on top of MongoDB, it is scalable and can support working smoothly with datasets containing billions of genotypes. Installable from docker images or all-in-one bundle archives, it is pretty straightforward to deploy on servers or local computers and has thus been adopted by numerous research institutes from around the world. Notably, Gigwa serves as a collaborative management tool and/or a portal for exposing the data for genebanks and breeding programs for some CGIAR centers <span class="citation" data-cites="bvSFyrtt">[<a href="#ref-bvSFyrtt" role="doc-biblioref">8</a>]</span>. Thus, the amount of data hosted and made widely accessible using this system has kept growing over the last few years.</p>
<p>Gigwa developers have been involved in the BrAPI community since 2016 and took part in designing the genotype-related part of the API’s specifications. Its first BrAPI-compliant features were designed for compatibility with the Flapjack visualization tool <span class="citation" data-cites="Wz8l0HlN">[<a href="#ref-Wz8l0HlN" role="doc-biblioref">5</a>]</span> and thus primarily turned it into a BrAPI datasource. Consequently, over time, Gigwa being the first and most reliable application implementing BrAPI-Genotyping server calls, local collaborators and even external partners used it as a reference solution to design a number of tools taking advantage of those features (e.g., <a href="https://webtools.southgreen.fr/BrAPI/Beegmac/">BeegMac</a>, <a href="https://github.com/jframi/snpclust">SnpClust</a>, <a href="https://github.com/icarda-git/QBMS">QBMS</a>). But further use-cases also required Gigwa to be able to consume data from other BrAPI servers, which led to also implement API-client features into the system. Thanks to all this work, a close collaboration was progressively established with the Integrated Breeding Platform team developing the widely used Breeding Management System, that ended up in both applications now being frequently deployed together, Gigwa pulling germplasm or sample metadata from BMS, and BMS displaying Gigwa-hosted genotypes within its own UI.</p>
<p>Client BrAPI libraries being available for R, community members typically write ad-hoc scripts syndicating data from multiple BrAPI sources (for instance phenotypes from a datasource and genotypes from another) in order to run various kinds of analyses such as GWAS, genomic selection or phylogenetic investigations. As a perspective, we may expect the most generic and widely-used of those pipelines to be at least publicly distributed, and possibly web-interfaced using solutions like R-Shiny in order to provide new, excitingly useful online services, based on Gigwa-hosted data.</p>
<h4 id="phg">PHG</h4>
<!-- Lynn J. -->
<p>The Practical Haplotype Graph (PHG) is a graph-based computational framework that represents large-scale genetic variation and is optimized for plant breeding and genetics. Using a pangenome approach, each PHG stores haplotypes (the sequence of part of an individual chromosome) to represent the collected genes of a species. This allows for a simplified approach for dealing with large scale variation in plant genomes. The PHG pipeline provides support for a range of genomic analyses and allows for the use of graph data to impute complete genomes from low density sequence or variant data.</p>
<p>Users access the crop databases either with direct calls to the PHG embedded server or indirectly using the rPHG library from an R environment. The PHG server accepts BrAPI endpoint queries to return information on sample lists and the variants used to define the graph’s haplotypes. In addition, PHG users utilize the BrAPI variantsets endpoint query to return links to VCF files containing haplotype data. Work on the PHG is ongoing. We expect to support additional BrAPI endpoints that allow for slicing genotypic data based on samples and regions.</p>
<h3 id="germplasm-management">Germplasm Management</h3>
<p>The vast quantity of new accessions, variants, and lines being created each year means germplasm data management is critical to keeping things organized. This is true at the level of a single breeding program, on a national scale, and on the international scale. BrAPI supports the transmission of germplasm passport data, as well as pedigree trees and crossing metadata. The BrAPI community has developed BrAPI compliant tools for storing, searching, and visualizing this metadata. There are even some plans in place to build federated networks of genebank data, connected via BrAPI.</p>
<h4 id="mgis">MGIS</h4>
<!-- authors: Mathieu Rouard, Valentin Guignon -->
<p>The Musa Germplasm information system, <a href="https://www.crop-diversity.org/mgis/">MGIS</a>, serves as a comprehensive community portal dedicated to banana diversity, a crop critical to global food security <span class="citation" data-cites="84Yn3FWj">[<a href="#ref-84Yn3FWj" role="doc-biblioref">9</a>]</span>. MGIS offers detailed information on banana germplasm, focusing on the collections held by the CGIAR International Banana Genebank (ITC) <span class="citation" data-cites="JWi7AKhJ">[<a href="#ref-JWi7AKhJ" role="doc-biblioref">10</a>]</span>. It is built on the Build on the Drupal/Tripal technology, like BIMS and Florilège. Since its inception, MGIS developers have actively participated in the Breeding API (BrAPI) community, pushing for the integration of Multicrop Passport Data (MCPD) into Germplasm module call of the API. MGIS thus provides passport data information on ITC banana genebank accessions (with GLIS DOI), synchronized with <a href="https://www.genesys-pgr.org/a/overview/v2YdWZGrZjD">Genesys</a>, but also enriches it by incorporating additional data from other germplasm collections worldwide. All those germplasm data are available through BrAPI germplasm module calls implementations. For genotyping data, MGIS incorporates GIGWA <span class="citation" data-cites="AVjkjraw">[<a href="#ref-AVjkjraw" role="doc-biblioref">7</a>]</span>, which provides tailored implementations for BrAPI genotyping module calls. Furthermore, MGIS supports the implementation of a set of BrAPI phenotyping module calls, facilitating the exposing of morphological descriptors and trait information supported by ontologies like the Crop Ontology <span class="citation" data-cites="VndIxnl3">[<a href="#ref-VndIxnl3" role="doc-biblioref">11</a>]</span>. It is integrated with the Trait Selector BrAPP, developed as part of a project involving Breedbase <span class="citation" data-cites="al3Hd9Ml">[<a href="#ref-al3Hd9Ml" role="doc-biblioref">12</a>]</span>. Uses cases between the Musa implementation of Breedbase, MusaBase, and MGIS to interlink genebank and breeding data.</p>
<!-- Contribution BrAPI 2.0 paper
Suggested Authors: Matthias Lange, Patrick König, Stephan Weise, Gouripriya Davuluri, Suman Kumar, Joseph Ruff, Paul Kersey, Cyril Pommier, Michael Alaux, Erwan Le-Floch -->
<h4 id="agent-portal">AGENT Portal</h4>
<p>In the global system for ex situ conservation of plant genetic resources (PGR) <span class="citation" data-cites="S1uQPBPG">[<a href="#ref-S1uQPBPG" role="doc-biblioref">13</a>]</span>, a total of ~5.8 million accessions are conserved in 1750 ex situ genebanks <span class="citation" data-cites="USun3Z8l">[<a href="#ref-USun3Z8l" role="doc-biblioref">14</a>]</span>. Unique and permanent identifiers in the form of DOIs are available for more than 1.7 million accessions <span class="citation" data-cites="doi:Food">[<a href="#ref-doi:Food" role="doc-biblioref"><strong>doi:Food?</strong></a> and Agriculture Organization (FAO) The Global Information System for PGRFA]</span>. Each DOI is linked to some basic descriptive data that facilitates the use of these resources. Many DOIs are also linked to additional data from different domains or will be in the future. In order to answer questions on the global biological diversity of a plant species, on duplicate detection, on provenance tracking for the identification of genetic integrity, on the selection of the most suitable material for various purposes, including breeding and research, and to support further applications in data mining or AI, a data space beyond the most basic information is needed that includes genotypic and phenotypic data. In this context, the aim of the AGENT project (<a href="https://www.agent-project.eu/" class="uri">https://www.agent-project.eu/</a>) funded by the European Commission is to develop a concept for the digital exploitation and activation of this GenRes data space via European ex situ genebanks according to the FAIR criteria <span class="citation" data-cites="6DjakjNS">[<a href="#ref-6DjakjNS" role="doc-biblioref">15</a>]</span> and to test it in practice using two important crops, barley and wheat. In two work packages, standards and technology for data interoperability will be developed to establish a genetic resources infrastructure, which regulates data acquisition of genotypic and phenotypic data, integrates and archives them and makes them accessible according to FAIR principles. To this end, 13 European genebanks and 5 bioinformatics centers are cooperating and have agreed on standards and protocols for (i) the data flow (see figure <a href="#fig:AGENT_Genotyping_Data_Flow">3</a>) and data formats <span class="citation" data-cites="31NWAHsO">[<a href="#ref-31NWAHsO" role="doc-biblioref">16</a>]</span> for central archiving of genotypic and phenotypic data.</p>
<div id="fig:AGENT_Genotyping_Data_Flow" class="fignos">
<figure>
<img src="images/AGENT_Genotyping_Data_Flow.png" style="width:100.0%" alt="Figure 3: Figure Data flow of genotypic data from AGENT partner databases" />
<figcaption aria-hidden="true"><span>Figure 3:</span> Figure Data flow of genotypic data from AGENT partner databases</figcaption>
</figure>
</div>
<p>The AGENT portal as described in more detail in section unlock the full potential of the biological material stored in genebanks around the globe by using FAIR international data standards and an open digital infrastructure for the management of plant genetic resources. The implemented BrAPI interface enables to mine current and historic genotypic and phenotypic information to drive the discovery of genes, traits and knowledge for future missions, complement existing information for wheat and barley and the new data standards and infrastructure to foster an improved management of PGR for other crop species across European genebanks.</p>
<p>For the joint research data infrastructure for the federation of collections of genotypic and phenotypic data from European gene banks and bioinformatics institutes, a AGENT portal (<a href="#fig:AGENT_WebFrontend">4</a>) as database infrastructure for integrated plant genetic resources on ex-situ genebanks is being created. It provides, manual data exploration, machine-readable access via BrAPI and provide data to the cored data deposition resources at the European Bioinformatics Institute (EBI).</p>
<div id="fig:AGENT_WebFrontend" class="fignos">
<figure>
<img src="images/AGENT_WebFrontend.png" style="width:100.0%" alt="Figure 4: Figure AGENT Portal" />
<figcaption aria-hidden="true"><span>Figure 4:</span> Figure AGENT Portal</figcaption>
</figure>
</div>
<p>The AGENT database backend aggregates curated and integrated passport data, phenotypic and genotypic data about wheat and barley accessions of 18 project partners are harmonized and integrated via BrAPI endpoints (<a href="https://github.com/AGENTproject/BrAPI" class="uri">https://github.com/AGENTproject/BrAPI</a>) and explorable in a web portal (<a href="https://agent.ipk-gatersleben.de" class="uri">https://agent.ipk-gatersleben.de</a>). The BrAPI endpoints were made available by scattered implementation. Genotyping data use DivBrowse <span class="citation" data-cites="9fRbvOj2">[<a href="#ref-9fRbvOj2" role="doc-biblioref">6</a>]</span> storage engine and BrAPI interface. Endpoints for sample data are implemented using AGENT database SQL to BrAPI broker service.
To integrate those BrAPI endpoint provider into a single service and URL scheme, we work on their integration in a BrAPI proxy service. As next steps, we will expand BrAPI implementation to enable the integration of analysis pipelines in the AGENT portal, e.g. for genebank mining tools such as the FIGS+ pipeline developed by AGENT partner ICARDA <span class="citation" data-cites="6VGwG9za">[<a href="#ref-6VGwG9za" role="doc-biblioref">17</a>]</span>. Another perspective is to integrate the data collected in the AGENT project into the European Search Catalogue for Plant Genetic Resources (EURISCO) <span class="citation" data-cites="pk5OwSG6">[<a href="#ref-pk5OwSG6" role="doc-biblioref">18</a>]</span> and to implement BrAPI endpoints to make data on PGR collections in European genebanks programmatically accessible.</p>
<h4 id="helium">Helium</h4>
<p>Helium (<a href="https://helium.hutton.ac.uk" class="uri">https://helium.hutton.ac.uk</a>) <span class="citation" data-cites="X22hwO1v">[<a href="#ref-X22hwO1v" role="doc-biblioref">19</a>]</span> is a plant pedigree visualization platform designed to account for the specific problems that are unique to plant pedigrees. A pedigree is a representation of how genetically discrete individuals are related to one another and is therefore a representation of the genetic relationship between individual plant lines, their parents and progeny. Plant pedigrees are often used to check for potential genotyping or phenotyping errors, since these errors, by the very nature of Mendelian inheritance, are constrained by the pedigree structure in which they exist (Paterson 2011). The accurate representation of pedigrees, and the ability to pull pedigree data from different data sources is therefore important in plant breeding and genetics and therefore ways to visualize and interact this complex data in meaningful ways is critical.</p>
<p>From its original desktop interface (<a href="https://github.com/cardinalb/helium-docs/wiki" class="uri">https://github.com/cardinalb/helium-docs/wiki</a>), Helium has developed into a web-based visualization platform implementing BrAPI calls to allow users to import data from other BrAPI compliant databases (<a href="https://helium.hutton.ac.uk" class="uri">https://helium.hutton.ac.uk</a>). The ability to pull data from BrAPI compliant data sources has significantly expanded Helium’s capability and utility within the community. Helium is used in projects ranging in size from tens to tens of thousands of lines and across a wide variety of crops and species. While originally designed for plant data <span class="citation" data-cites="1HmEHgW8h">[<a href="#ref-1HmEHgW8h" role="doc-biblioref">20</a>]</span> it has also found utility in other non-plant projects <span class="citation" data-cites="x8DIc6zn">[<a href="#ref-x8DIc6zn" role="doc-biblioref">21</a>]</span> highlighting its broad utility. This also allows Helium users to provide direct dataset links to collaborators allowing the original data to be held with the data provider and utilising Helium for its visualization functionality. Our current Helium deployment includes example BrAPI calls to a barley dataset at Hutton to allow users to test the system and features it offers.</p>
<h4 id="glis">GLIS</h4>
<p>The Global Information System (GLIS) on Plant Genetic Resources for Food and Agriculture (PGRFA) of the International Treaty on Plant Genetic Resources for Food and Agriculture (ITPGRFA) is a web-based global entry point for users and third-party systems to access information and knowledge on scientific, technical and environmental matters to strengthen PGRFA conservation, management and utilization activities. The system and its portal also enable recipients of PGRFA to make available all non-confidential information on germplasm according to the provisions of the Treaty and facilitates access to the results of their research and development.</p>
<p>Thanks to the adoption of Digital Object Identifiers (DOIs) to PGRFA ex situ and in situ based on the Multi-Crop Passport Descriptors (MCPD), the Portal provides access to 1.7 million PGRFA in collections conserved worldwide. Of these, over 1.5 million are accessible for research, training and plant breeding in the food and agriculture domain.</p>
<p>The Scientific Advisory Committee of the International Treaty and the Governing Body have repeatedly welcomed efforts on interoperability among germplasm information systems. In this context, the GLIS Portal adopted the Breeding API (BrAPI v1) in 2022. Integrating the BrAPI among the GLIS content negotiators facilitates queries and the exchange of content for data management in plant breeding. The Portal also offers other protocols (XML, DarwinCore, JSON and JSON-LD) to increase data and metadata connectivity. In the near future, depending on the availability of resources, upgrading to BrAPI v2 is planned.</p>
<h4 id="florilège-gateway-to-french-plant-genetic-resources">FLORILÈGE (Gateway to French Plant Genetic Resources)</h4>
<p>Designed primarily for the general public, Florilège provides access to public collections of all French plant biological resources centers. This web portal allows to browse available plant genetic resource accessions and gives the possibility to order some seeds or plant material for cultivation. It includes plant genetic resources of around fifty plant genus from 19 genebanks.</p>
<p>Florilège retrieves accession information from different BrAPI-compliant systems. They include OLGA, a genebank accessions information management system, and GnpIS<span class="citation" data-cites="m7yuFGsd">[<a href="#ref-m7yuFGsd" role="doc-biblioref">22</a>]</span> <span class="citation" data-cites="15XwZ2Vb0">[<a href="#ref-15XwZ2Vb0" role="doc-biblioref">23</a>]</span>, an INRAE data repository for plant genetic resources, phenomics and genetics. Using BrAPI to gather data from these systems reduced the efforts and enabled standardized data retrieval. As a consequence, BrAPI is the de facto standard for exchanging data within the French plant genetic resources community. The Florilège team also requested several update of the BrAPi specifications to better serve this use case, such as Collection or improved external references.
<img src="images/Schema_Florilege.jpg" id="fig:Schema_Florilege" style="width:100.0%" alt="Figure Florilege Workflow" /></p>
<p>Florilège is developed in Drupal 10, and uses xnttbrapi module (to easily connect to BrAPI compliant external databases).</p>
<h4 id="faidare">FAIDARE</h4>
<!-- Cyril P, first draft-->
<p>FAIDARE (<a href="https://urgi.versailles.inrae.fr/faidare/" class="uri">https://urgi.versailles.inrae.fr/faidare/</a>) is a data discovery portal providing a biologist friendly search system over a global federation of 33 plant research databases. It allows to identify data resources using a full text approach completed with domain specific filters and to link back to the original database for visualization, analysis and download. For instance, it is possible to search for “wheat drought” then to refine the search to the “Triticum aestivum” taxon and yield component traits such as “Thousand Grain Weight”. The indexed data types are very broad and include genomic features, such as genes or transposable elements, selected bibliography, QTL, markers, genetic variation studies, phenomic studies and plant genetic resources ie germplasm. This inclusiveness is achieved thanks to a two stage indexation data model. The most generic one provides basic search functionalities and relies on five fields : name, link back URL, data type, species and exhaustive description. The filtering is directly tied to some of those fields. Therefore, to provide more advanced filtering, FAIDARE is also providing a second stage indexation mechanism by taking advantage of BrAPi endpoints to get more detailed metadata on genotyping and phenotyping studies as well as germplasm. In parallel, FAIDARE provides a pre-visualization of germplasm and studies using dedicated cards.
<img src="images/Schema_FAIDARE.png" id="fig:Schema_FAIDARE" style="width:100.0%" alt="Figure FAIDARE Federation" />
The indexation mechanism relies on a dedicated public software (<a href="https://github.com/elixir-europe/plant-brapi-etl-faidare" class="uri">https://github.com/elixir-europe/plant-brapi-etl-faidare</a>) that allows data resources manager to request the indexation of there database using pull requests. This BrAPI client is able to extract data from any BrAPI 1.3 and 1.2 endpoint and development of BrAPI 2.x indexation will be initiated in 2025. Since not all databases are willing to implement BrAPI endpoints, we also provide the possibility to generate metadata as BrAPI json files, hence using the standard as a file exchange format.
FAIDARE architecture has been designed by elaborating on the GnpIS Software Architecture <span class="citation" data-cites="m7yuFGsd">[<a href="#ref-m7yuFGsd" role="doc-biblioref">22</a>]</span>. As a consequence, BrAPI is at the core of its datamodel, and in particular the JSON data files served by the Elasticsearch NoSQL engine are enriched version of the BrAPI JSON files. FAIDARE also includes a BrAPI endpoint that serves all indexed metadata.
FAIDARE has been adopted by several communities and in particular in the ELIXIR and EMPHASIS european infrastructures. It is also used by the WheatIS of the Wheat-Initiative. Several databases are added each year to the FAIDARE global federation, allowing to increase both the portal and the BrAPI adoption.</p>
<h3 id="breeding-data-management">Breeding Data Management</h3>
<p>While specialty data management is important for some use cases, often breeders want a central repository of critical data. General breeding data management systems support some level phenotypic, genotypic, and germplasm data, as well as trial, equipment, and people management. By enabling BrAPI support, these larger systems can connect with smaller tools and specialty systems to provide more functionality under the same user interface. There are several breeding data management systems developed in the BrAPI community, each with their own pros and cons.</p>
<h4 id="deltabreed">DeltaBreed</h4>
<!-- Shawn Y.-->
<p>DeltaBreed is an open-source data management system designed and developed by Breeding Insight to support USDA-ARS specialty crop and animal breeders. DeltaBreed is a unified system for managing breeding data that connects a variety of BrAPI applications (see list below). BrAPI integration allows the complexity underlying interoperability to be hidden, shielding users from multifactorial differences between diverse applications. DeltaBreed, adhering to the BrAPI model, establishes data standards and validations for users and provides a singular framework for data management and user training.</p>
<p>DeltaBreed users need not be aware of BrAPI or the specifics of underlying applications but will notice that BrAPI interoperability reduces the need for human-mediated file transfers and data manipulation. Field Book users, for example, can connect to their DeltaBreed program, authenticate, and pull studies and traits directly from DeltaBreed to Field Book on their data collection device. The subsequent step of pushing observations from Field Book to DeltaBreed is straightforward via BrAPI, but will not be implemented until repeated observation handling workflows are established to differentiate and validate repeated observations, such as accidental repeats, overwrite requests, time-series observations, and repeated sub-entity measures. Users can expect DeltaBreed observation handling to become more seamless with future development.</p>
<p><strong>DeltaBreed Connected Applications</strong>
&lt;&lt; Submission is expected April 2024. We may need to trim this aspirational list down to reality in final edits.&gt;&gt;</p>
<ul>
<li>BIMS <a href="https://www.breedwithbims.org/" class="uri">https://www.breedwithbims.org/</a></li>
<li>BrAPI Java Server <a href="https://test-server.brapi.org/brapi/v2/" class="uri">https://test-server.brapi.org/brapi/v2/</a></li>
<li>BrAPI Sync <a href="https://github.com/IntegratedBreedingPlatform/brapi-sync" class="uri">https://github.com/IntegratedBreedingPlatform/brapi-sync</a></li>
<li>BreedBase <a href="https://breedbase.org/" class="uri">https://breedbase.org/</a></li>
<li>Diversity Arrays Technologies (DArT) genotyping services</li>
<li>Field Book <a href="https://play.google.com/store/apps/details?id=com.fieldbook.tracker" class="uri">https://play.google.com/store/apps/details?id=com.fieldbook.tracker</a></li>
<li>Gigwa <a href="https://gigwa.southgreen.fr/gigwa/" class="uri">https://gigwa.southgreen.fr/gigwa/</a></li>
<li>Mr Bean <a href="https://github.com/AparicioJohan/MrBeanApp" class="uri">https://github.com/AparicioJohan/MrBeanApp</a></li>
<li>Pedigree Viewer <a href="https://github.com/solgenomics/BrAPI-Pedigree-Viewer" class="uri">https://github.com/solgenomics/BrAPI-Pedigree-Viewer</a></li>
</ul>
<h4 id="bms">BMS</h4>
<p>The <a href="https://bmspro.io">Breeding Management System (BMS)</a>, developed by the <a href="https://integratedbreeding.net/">Integrated Breeding Platform (IBP)</a>, is a suite of tools designed to enhance the efficiency and effectiveness of plant breeding. BMS covers all stages of the breeding process, with the emphasis on germplasm management and <a href="https://cropontology.org">ontology</a>-harmonized phenotyping. It also features analytics and decision-support tools. With its focus on interoperability, BMS integrates smoothly with BrAPI, facilitating easy connections with a broad array of complementary tools and databases, notably <a href="https://southgreen.fr/content/gigwa">Gigwa</a> which is deployed together with the BMS to fulfill the genotyping data management needs of BMS users.</p>
<p>The <a href="https://github.com/IntegratedBreedingPlatform/brapi-sync">brapi-sync</a> tool, a significant component of BMS’s BrAPI capabilities, was developed by the IBP and released as a BrAPP for community use. Brapi-sync is designed to enhance collaboration among partner institutes within a network such as Innovation and Plant Breeding in West Africa (<a href="https://www.iavao.org/en">IAVAO</a>), by enabling the sharing of germplasm and trials across BrAPI-enabled systems. This tool helps overcome traditional barriers to collaboration, ensuring data that was once isolated within specific programs or platforms can now be easily shared, integrated, and synchronized.</p>
<p>Additionally, brapi-sync improves data management by utilizing the externalReferences field to maintain links to the origin IDs of each entity it transmits. This not only retains the original context of the data but also establishes a traceability mechanism for accurate data source attribution and verification. Such practices are crucial for maintaining data integrity and fostering trust among collaborative partners, ensuring access to accurate, reliable, and current information.</p>
<h4 id="breedbase">Breedbase</h4>
<p>Breedbase is a comprehensive breeding data management system <span class="citation" data-cites="al3Hd9Ml">[<a href="#ref-al3Hd9Ml" role="doc-biblioref">12</a>]</span> <span class="citation" data-cites="1FDf25dDD">[<a href="#ref-1FDf25dDD" role="doc-biblioref">24</a>]</span> that implements a digital ecosystem for all breeding data, including trial data, phenotypic data, and genotypic data. Data acquisition is through tabled-based apps such as Fieldbook <span class="citation" data-cites="MlqG0w6i">[<a href="#ref-MlqG0w6i" role="doc-biblioref">25</a>]</span> and related apps, such as Coordinate and InterCross apps, through drone imagery, Near Infra-Red Spectroscopy (NIRS), and other technologies. Search functions such as the Search Wizard interface provide powerful query capabilities, and various breeding-centric analysis tools are available, including mixed models, heritability, stability, PCA, and various clustering algorithms. The original impetus for creating Breedbase was the advent of new breeding paradigms based on genomic information such as genomic prediction algorithms <span class="citation" data-cites="zuq8ri4R">[<a href="#ref-zuq8ri4R" role="doc-biblioref">26</a>]</span> and the accompanying data management challenges, and complete genomic prediction workflow is integrated in the system. The first instance was created for the NextGen Cassava project in 2012 as the Cassavabase (<a href="https://cassavabase.org/" class="uri">https://cassavabase.org/</a>) database. Databases for other CGIAR root, tuber and banana (RTB) crops followed with database for yam (<a href="https://yambase.org/" class="uri">https://yambase.org/</a>), sweet potato (<a href="https://sweetpotatobase.org/" class="uri">https://sweetpotatobase.org/</a>), banana (<a href="https://musabase.org/" class="uri">https://musabase.org/</a>) as well as instances in labs and companies. The BrAPI interface <span class="citation" data-cites="QfGEJ6km">[<a href="#ref-QfGEJ6km" role="doc-biblioref">1</a>]</span> is crucial for Breedbase: Breedbase communicates via BrAPI with the data collection tablets, connection to other projects such as CLIMMOB <span class="citation" data-cites="LCpCba21">[<a href="#ref-LCpCba21" role="doc-biblioref">27</a>]</span>, and many native tools use the BrAPI interface for accessing data. Users also appreciate the ability to connect to Breedbase instances using packages such as QBMS <a href="https://icarda-git.github.io/QBMS/" class="uri">https://icarda-git.github.io/QBMS/</a> for data import into R for custom analyses. Breedbase has been an early and continuous adopter of, and contributor to, the BrAPI standard.</p>
<h4 id="bims">BIMS</h4>
<p>BIMS (Breeding Information Management System) <span class="citation" data-cites="ArPSsaJQ">[<a href="#ref-ArPSsaJQ" role="doc-biblioref">28</a>]</span> is a free, secure, and online breeding management system which allows breeders to store, manage, archive, and analyze their private breeding program data. BIMS enables individual breeders to have complete control of their own breeding data along with access to tools such as data import/export, data analysis and data archiving for their germplasm, phenotype, genotype, and image data. BIMS is currently implemented in five community databases, the Genome Database for Rosaceae <span class="citation" data-cites="eaDrNpsp">[<a href="#ref-eaDrNpsp" role="doc-biblioref">29</a>]</span>, CottonGEN <span class="citation" data-cites="dcg0274X">[<a href="#ref-dcg0274X" role="doc-biblioref">30</a>]</span>, the Citrus Genome Database, the Pulse Crop Database, and the Genome Database for Vaccinium, as well as a crop-independent website, <a href="https://breedwithbims.org" class="uri">https://breedwithbims.org</a>. BIMS in these five community databases enables individual breeders to import publicly available data so that they can utilize public data in their breeding program. BIMS utilizes the Android App Field Book, enabling seamless data transfer between BIMS and the Field Book App through either files or BrAPI. Data transfer through BrAPI between BIMS and other resources such as BreedBase, GIGWA, and Breeder Genomics Hub is also on the way.</p>
<h4 id="germinate">Germinate</h4>
<p><a href="https://ics.hutton.ac.uk/get-germinate/">Germinate</a> <span class="citation" data-cites="sW9euzzP">[<a href="#ref-sW9euzzP" role="doc-biblioref">31</a>]</span> is an open-source plant genetic resources database that combines and integrates various kinds of plant breeding data including genotypic data, phenotypic trials data, passport data, images, geographic information and climate data into a single repository. Germinate is tightly linked to the BrAPI specification and supports a majority of BrAPI endpoints for querying, filtering and submission.</p>
<p>Germinate integrates and connects with other BrAPI-enabled tools such as GridScore for phenotypic data collection, Flapjack for genotypic data visualization and Helium for pedigree visualization, but, due to the nature of BrAPI, Germinate can act as a data repository for any BrAPI-compatible tool. Thanks to the interoperability provided by BrAPI the need for manual data handling becomes a rarity with the direct benefit of faster data processing, fewer to no human errors, data security and integrity.</p>
<h3 id="analytics">Analytics</h3>
<p>While other tools listed above have the capability to do specialized analytics on certain types of data, general analytics tools can cover a wide range of data types and analytical models. The tools developed by the BrAPI community, can pull in data from multiple BrAPI compatible data sources and provide enhanced analytical functionality. In many cases, there is no longer a need to import and export large data files to a local computational environment just to run standard analytical models. These tools are able to extract the data they need from a data source without much human intervention or human error.</p>
<h4 id="qbms">QBMS</h4>
<p>Modern breeding programs can utilize data management systems to maintain both phenotypic and genotypic data. Numerous systems are available for adoption. To fully leverage the benefits of digitalization in this ecosystem, breeders need to utilize data from different sources to make efficient data-driven decisions. With increased computational power at their disposal, scientists can construct more advanced analysis pipelines by combining various data sources.</p>
<p><a href="https://icarda-git.github.io/QBMS">QBMS</a> <span class="citation" data-cites="a9uPTJgO">[<a href="#ref-a9uPTJgO" role="doc-biblioref">32</a>]</span> R package eliminates technical barriers scientists experience when using the BrAPI calls in their analysis scripts and pipelines. This barrier arises from the complexity of managing API backend processes, such as authentication, tokens, TCP/IP protocol, JSON format, pagination, stateless calls, asynchronous communication, database IDs, and more. To bridge this gap, we have developed the QBMS R package. This package abstracts the technical complexities, providing breeders (targetted end users) with stateful action verbs/functions familiar to them when navigating their GUI systems. It enables them to query and extract data into a standard data frame structure, consistent with their use of R language, one of the most common statistical tools in the breeding community.</p>
<p>Since its release on the official CRAN repository in October 2021, the QBMS R package has garnered over 9400 downloads. Several tools, such as MrBean, rely on the QBMS package as their source data adapter. Moreover, the community has started building extended solutions on top of it. QBMS can serve as a cornerstone in the breeding modernization revolution by providing access to actionable data and enabling the creation of dashboards to reduce the time between harvest and decision-making for the next breeding cycle.</p>
<h4 id="mr.-bean">Mr. Bean</h4>
<p>Mr.Bean <span class="citation" data-cites="7WXz7FUP">[<a href="#ref-7WXz7FUP" role="doc-biblioref">33</a>]</span> is a graphical user interface designed to assist breeders, statisticians, and individuals involved in plant breeding programs with the analysis of field trials. By utilizing innovative methodologies such as SpATS for modeling spatial trends and autocorrelation models to address spatial variability, Mr.Bean proves highly practical and powerful in facilitating faster and more effective decision-making. Modeling Genotype-by-environment interaction poses its challenges, but Mr.Bean offers the capability to explore various variance-covariance matrices, including Factor Analytic, compound symmetry, and heterogeneous variances, among others, aiding in the assessment of genotype performance across diverse environments.</p>
<p>Mr.Bean boasts flexibility in importing different file types, yet for users managing their data within data management systems (DMS), the process of downloading from their DMS and importing it into MrBean can be cumbersome. To address this issue, QBMS operates in the back-end. This feature prompts users to input the URL of the server, their credentials if necessary, and the specific trial they wish to analyze. Subsequently, users can seamlessly access and utilize their dataset within the entire interface.</p>
<h4 id="g-crunch">G-Crunch</h4>
<p>G-Crunch is an upcoming user-facing analysis tool that attempts to fill the space of simple, user driven analytics requests, with a generic user interface and the ability to swap out data sources and analysis tools. G-Crunch hopes to streamline repeatable, debuggable simple analytic requests and results.
G-Crunch, as a tool, couldn’t feasibly exist without BrAPI. The support of BrAPI interfaces allows G-Crunch to use one unified request method, and adapt to the user’s (BrAPI-compliant) existing network of tools, which lowers the barrier to entry for adoption.</p>
<h3 id="general-infrastructure">General Infrastructure</h3>
<p>Adopting BrAPI compatibility into an existing system can be difficult sometimes. The BrAPI Community has developed several tools to make adoption easier. This includes things like pre built code libraries, connectors to other technology standards, and mappers to alternate data types or data files. THe goal here is always to lower the barrier to entry for the BrAPI community, making it easier for other groups to get started and connect their existing data to the standard.</p>
<h4 id="miappe-isa-to-brapi-service">MIAPPE ISA to BrAPI service</h4>
<p>Phenotyping is crucial in the breeding process as it enables the identification of desirable traits, selection of breeding lines, and evaluation of breeding success. In the plant community, MIAPPE (Minimal Information About a Plant Phenotyping Experiment) <span class="citation" data-cites="kWLsm7g6">[<a href="#ref-kWLsm7g6" role="doc-biblioref">34</a>]</span> is the established standard for phenotyping experiments and is commonly serialized as ISA Tab <span class="citation" data-cites="KZc5ayvK">[<a href="#ref-KZc5ayvK" role="doc-biblioref">35</a>]</span>. Although ISA Tab is easy to read for non-technical experts due to its file-based approach, it lacks programmatic access, particularly for web applications. BrAPI, which is aligned with MIAPPE, can help solve this problem.
MIRA is a tool that enables the automatic deployment of a BrAPI server on a MIAPPE-compliant dataset in ISA Tab format. It can be deployed from a Docker image with the dataset mounted. By utilizing the mapping between MIAPPE, ISA, and BrAPI, there is no need for parsing or manual mapping of datasets that are already compliant with (meta-)data standards. By gaining programmatic access through BrAPI to these datasets, it facilitates the integration of phenotyping datasets into web applications.</p>
<h4 id="miappe-brapi-to-isa-service">MIAPPE “BrAPI to ISA” service</h4>
<p>Since the release of BrAPI 1.3, efforts have been made to incorporate support for the Minimum Information About Plant Phenotyping Experiments (MIAPPE) standard into the specification <span class="citation" data-cites="kWLsm7g6">[<a href="#ref-kWLsm7g6" role="doc-biblioref">34</a>]</span>. This integration was finalized in BrAPI 2.0, resulting in full compatibility between the two standards. Consequently, BrAPI now encompasses all attributes necessary for MIAPPE compliance, adhering to standardized descriptions in accordance with MIAPPE guidelines. Leveraging BrAPI as a standardized RESTful web service API specification, we employ the ISA standard for storing metadata and phenotyping data in a standardized manner. This data is structured in the ISA-TAB file format and subjected to validation using the <a href="https://github.com/ELIXIR-Belgium/isatab-validation">MIAPPE ISA configuration</a>. The “BrAPI to ISA” service functions as a converter between BrAPI RESTful endpoints and ISA-TAB, facilitating the archiving of metadata and data and thereby enhancing data preservation and accessibility. The <a href="https://github.com/elixir-europe/plant-brapi-to-isa">BrAPI2ISA</a> tool is designed to be compatible with BrAPI 1.3, and we invite contributions from the community to extend support for the latest versions of BrAPI.</p>
<h4 id="brapimapper">BrAPIMapper</h4>
<!-- Valentin G: I'm planning to add a figure in supplementary data that shows data sets on a side, the mapper with some internal parts and the exposed BrAPI services. -->
<p>BrAPIMapper is a full BrAPI implementation of all calls for any data source missing BrAPI implementation or compliance with some BrAPI versions. BrAPIMapper is provided as a docker application that can get its external data sources from mySQL or PostgreSQL databases (with a dedicated interface for Chado database schema), generic REST services (with a dedicated interface for BrAPI endpoints), flat files (XML, JSON, CSV/TSV/GFF3/VCF, YAML) or any combination of any of those. It provides an administration interface to map BrAPI data models to external data sources. The interface allows administrators to select the BrAPI specification versions to use and the calls to enable. Data mapping configuration export and import features simplify upgrades to future BrAPI specifications changes as administrators would only have to map missing fields or make minor adjustments. Amongst others, it supports paging, search calls, either by providing direct results or using deferred results with a search identifier, lists, authentication and manages access restrictions to calls that can be setup through the administration interface as well. This tool aims to accelerate BrAPI services deployment while ensuring specification compliance.</p>
<h4 id="brapi-plug-and-play-graphql-based-data-warehouse">BrAPI plug and play GraphQL based data-warehouse</h4>
<p>Using the “Zendro” set of automatic software program-code generators (zendro-dev.github.io) a fully functional, efficient, and cloud-capable BrAPI data-warehouse has been created for the current version of the BrAPI data models. The resulting data-warehouse has two interfaces, one application programming interface implemented in the form of a GraphQL web-server and another intuitive point and click graphical user interface in the browser. Both provide secure access to data read and write functions for all BrAPI data models. These data administration methods comprise create, read, update, and delete (CRUD) functions that are standardized and accept the same parameters for all data models.</p>
<p>While data write access comprises both persisting single or multiple records, data read access is particularly rich in features and includes access to single records referred to by their id and access to multiple records selected by logical filters. In this, multiple records are paginated using the highly efficient cursor based pagination model as proposed in the GraphQL standard. Logical filters allow for exhaustive search queries, whose structure is highly intuitive and based around logical triplets in which a data model field is validated using an operator and a value, e.g. “Study name equals ‘xyz’”. In this a large collection of operators is available and triplets can be combined to logical search trees using “and” or “or” operators. Searches can be extended over relationships between data models, thus enabling a user to query the warehouse exactly for the data wanted.</p>
<p>Access security is implemented with the OAuth2 user authentication standard (datatracker.ietf.org/doc/html/rfc6749). Authorization is based on user roles and can be configured differently for each single data model read or write function.</p>
<p>The browser based graphical user interface is implemented in React.js with Next and exposes an intuitive and self explanatory set of functions for each data model. In the left a menu allows the user to access all BrAPI data models. Upon clicking on a model a table is shown which allows the user to paginate through all existing records, sort them by any column, search the records, add new records, or update or delete existing records, if the user role authorizes these functions. Record data can be inspected in a detail view and here relationships to other data records can be reviewed using the very same graphical visual representations. Breadcrumbs allow the user to navigate back and forth in the trail of relationships inspected. Finally, the generated graphical interface allows for the integration of interactive scientific plots and analysis tools written in JavaScript or WebAssembly.</p>
<!-- Asis Hallab: To Do: (i) provide the link to an example running data warehouse, and (ii) include and describe example scientific plots. -->
<p>The Zendro based BrAPI plug and play data-warehouse is capable of forming an efficient cloud of data servers. This is achieved simply by linking (URLs) other Zendro based warehouses that expose the same GraphQL API to the same data models, or a subset of data models. Any network of such Zendro GraphQL servers can be set up using this configuration approach. The code generated then exposes full access to all data records stored on any node of the network, while maintaining full security control at each node. Importantly, the warehouses are programmed in such a way that any number of data servers can be joined without loss of efficiency. Only the network connection speed and size of requested record sets influence the performance.</p>
<p>As explained, Zendro is a code generator and creates a fully functional data warehouse from input data model definitions, i.e. a schema. The schema is given in the form of special data model descriptions, in which each model is defined using JavaScript Object Notation (JSON). Each model is defined in its respective JSON file. A translator has been developed to create the Zendro schema from the BrAPI data model definitions. This ensures that Zendro can create plug and play data warehouses for future versions of the BrAPI with great ease, i.e. by translating the BrAPI models to Zendro input and subsequently running Zendro to create the plug and play warehouse.</p>
<h2 id="discussion">Discussion</h2>
<!-- * how can BrAPI help breeders (specifically small breeders)
* Looking ahead - what needs to be done further
* Analytics
* HDP
* GraphQL?
* Field management - treatments etc
* Drones, image processing, HTP
* ChatGPT integration - BreedersGPT
* Weather & soil - why it won’t be in BrAPI -->
<h3 id="brapi-for-breeders-and-scientists">BrAPI for Breeders and Scientists</h3>
<p>The BrAPI technical specification document is meant to be read and used by software developers. However, the purpose of the specification, and the community around it, is to make things faster, easier, and cheaper for the breeders and scientists working to make the world a better place. BrAPI offers a convenient path to automation and data integration for software tools in the breeding domain. All of the example use cases described above can be achieved with manual effort, moving and editing data files by hand. However, when the basic structure and flow of data becomes automated, breeders and scientists can spend less time on data management and more time focussing on the science, doing what they do best. For many, the ultimate goal is the development of a digital ecosystem: a collection of software tools and applications that can all work together seamlessly. In this digital ecosystem, data is collected digitally from the beginning, reducing as much human error as possible. The data is checked by quality control and stored automatically, then can be sent to any internal tool or external lab for further analysis with just the click of a button. This idea might sound too good to be true, but as more tools start sharing a universal data standard, automating data flow becomes easier, and the community gets closer to total interoperability.</p>
<h3 id="looking-ahead">Looking Ahead</h3>
<p>The BrAPI specification will continue to grow, enabling more use cases and new types of data. These new use cases might include newer scientific techniques and technologies. Things like drone imaging data, spectroscopy, LIDAR, metabolomics, transcriptomics, high-throughput phenotyping, and machine learning analysis. All of these technologies can open new avenues for research and development of new crop varieties. All of these technologies also generate more data, and require data sharing between different software applications and data repositories. The BrAPI project leadership and community is committed to building the standards to support these new use cases as they arrive and become accepted by the scientific community. In fact, small groups within the BrAPI community have already start building generic data models and communication standards for many of the technologies listed above. These community efforts will eventually become part of the BrAPI standard in a future version of the specification document.</p>
<h2 id="conclusions-and-impact">Conclusions and Impact</h2>
<ul>
<li>High level summary of the project/consortium</li>
<li>BrAPI is fitting into this gap, it doesn’t need to fit these other gaps</li>
<li>Call to action - Join us!</li>
</ul>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-QfGEJ6km" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline"><strong>BrAPI—an application programming interface for plant breeding applications</strong> <div class="csl-block">Peter Selby, Rafael Abbeloos, Jan Erik Backlund, Martin Basterrechea Salido, Guillaume Bauchet, Omar E Benites-Alfaro, Clay Birkett, Viana C Calaminos, Pierre Carceller, Guillaume Cornut, … </div> <em>Bioinformatics</em> (2019-03-23) <a href="https://doi.org/gjgxxr">https://doi.org/gjgxxr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btz190">10.1093/bioinformatics/btz190</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30903186">30903186</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6792114">PMC6792114</a></div></div>
</div>
<div id="ref-Ne0E9HbC" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline"><strong>GridScore: a tool for accurate, cross-platform phenotypic data collection and visualization</strong> <div class="csl-block">Sebastian Raubach, Miriam Schreiber, Paul D Shaw</div> <em>BMC Bioinformatics</em> (2022-06-06) <a href="https://doi.org/gtkcsv">https://doi.org/gtkcsv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12859-022-04755-2">10.1186/s12859-022-04755-2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35668357">35668357</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9169276">PMC9169276</a></div></div>
</div>
<div id="ref-OodVqG7n" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline"><strong>Dealing with multi‐source and multi‐scale information in plant phenomics: the ontology‐driven Phenotyping Hybrid Information System</strong> <div class="csl-block">Pascal Neveu, Anne Tireau, Nadine Hilgert, Vincent Nègre, Jonathan Mineau‐Cesari, Nicolas Brichet, Romain Chapuis, Isabelle Sanchez, Cyril Pommier, Brigitte Charnomordic, … Llorenç Cabrera‐Bosquet</div> <em>New Phytologist</em> (2019-01) <a href="https://doi.org/gm9b9j">https://doi.org/gm9b9j</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/nph.15385">https://doi.org/10.1111/nph.15385</a></div></div>
</div>
<div id="ref-TVXmHLRk" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline"><strong>Enabling reusability of plant phenomic datasets with MIAPPE 1.1</strong> <div class="csl-block">Evangelia A Papoutsoglou, Daniel Faria, Daniel Arend, Elizabeth Arnaud, Ioannis N Athanasiadis, Inês Chaves, Frederik Coppens, Guillaume Cornut, Bruno V Costa, Hanna Ćwiek‐Kupczyńska, … Cyril Pommier</div> <em>New Phytologist</em> (2020-07) <a href="https://doi.org/gjqcmb">https://doi.org/gjqcmb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/nph.16544">https://doi.org/10.1111/nph.16544</a></div></div>
</div>
<div id="ref-Wz8l0HlN" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline"><strong>Flapjack—graphical genotype visualization</strong> <div class="csl-block">Iain Milne, Paul Shaw, Gordon Stephen, Micha Bayer, Linda Cardle, William TB Thomas, Andrew J Flavell, David Marshall</div> <em>Bioinformatics</em> (2010-10-18) <a href="https://doi.org/cdvnt8">https://doi.org/cdvnt8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btq580">10.1093/bioinformatics/btq580</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20956241">20956241</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2995120">PMC2995120</a></div></div>
</div>
<div id="ref-9fRbvOj2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline"><strong>DivBrowse—interactive visualization and exploratory data analysis of variant call matrices</strong> <div class="csl-block">Patrick König, Sebastian Beier, Martin Mascher, Nils Stein, Matthias Lange, Uwe Scholz</div> <em>GigaScience</em> (2022-12-28) <a href="https://doi.org/gtq5n6">https://doi.org/gtq5n6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/gigascience/giad025">10.1093/gigascience/giad025</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37083938">37083938</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10120423">PMC10120423</a></div></div>
</div>
<div id="ref-AVjkjraw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline"><strong>Gigwa v2—Extended and improved genotype investigator</strong> <div class="csl-block">Guilhem Sempéré, Adrien Pétel, Mathieu Rouard, Julien Frouin, Yann Hueber, Fabien De Bellis, Pierre Larmande</div> <em>GigaScience</em> (2019-05-01) <a href="https://doi.org/gtp5bz">https://doi.org/gtp5bz</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/gigascience/giz051">10.1093/gigascience/giz051</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31077313">31077313</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6511067">PMC6511067</a></div></div>
</div>
<div id="ref-bvSFyrtt" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline"><strong>A digital catalog of high‐density markers for banana germplasm collections</strong> <div class="csl-block">Mathieu Rouard, Julie Sardos, Guilhem Sempéré, Catherine Breton, Valentin Guignon, Ines Van den Houwe, Sebastien C Carpentier, Nicolas Roux</div> <em>PLANTS, PEOPLE, PLANET</em> (2021-03) <a href="https://doi.org/gtp5bx">https://doi.org/gtp5bx</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/ppp3.10187">10.1002/ppp3.10187</a></div></div>
</div>
<div id="ref-84Yn3FWj" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline"><strong>MGIS: managing banana (Musa spp.) genetic resources information and high-throughput genotyping data</strong> <div class="csl-block">Max Ruas, V Guignon, G Sempere, J Sardos, Y Hueber, H Duvergey, A Andrieu, R Chase, C Jenny, T Hazekamp, … M Rouard</div> <em>Database</em> (2017-01-01) <a href="https://doi.org/gmcmrf">https://doi.org/gmcmrf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/database/bax046">10.1093/database/bax046</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29220435">29220435</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5502358">PMC5502358</a></div></div>
</div>
<div id="ref-JWi7AKhJ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline"><strong>Safeguarding and using global banana diversity: a holistic approach</strong> <div class="csl-block">Ines Van den houwe, Rachel Chase, Julie Sardos, Max Ruas, Els Kempenaers, Valentin Guignon, Sebastien Massart, Sebastien Carpentier, Bart Panis, Mathieu Rouard, Nicolas Roux</div> <em>CABI Agriculture and Bioscience</em> (2020-10-22) <a href="https://doi.org/gtq9ws">https://doi.org/gtq9ws</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s43170-020-00015-6">10.1186/s43170-020-00015-6</a></div></div>
</div>
<div id="ref-VndIxnl3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline"><strong>Multifunctional crop trait ontology for breeders' data: field book, annotation, data discovery and semantic enrichment of the literature</strong> <div class="csl-block">Rosemary Shrestha, Elizabeth Arnaud, Ramil Mauleon, Martin Senger, Guy F Davenport, David Hancock, Norman Morrison, Richard Bruskiewich, Graham McLaren</div> <em>AoB PLANTS</em> (2010-05-27) <a href="https://doi.org/crc8vt">https://doi.org/crc8vt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/aobpla/plq008">10.1093/aobpla/plq008</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22476066">22476066</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3000699">PMC3000699</a></div></div>
</div>
<div id="ref-al3Hd9Ml" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline"><strong>Breedbase: a digital ecosystem for modern plant breeding</strong> <div class="csl-block">Nicolas Morales, Alex C Ogbonna, Bryan J Ellerbrock, Guillaume J Bauchet, Titima Tantikanjana, Isaak Y Tecle, Adrian F Powell, David Lyon, Naama Menda, Christiano C Simoes, … Lukas A Mueller</div> <em>G3 Genes|Genomes|Genetics</em> (2022-04-06) <a href="https://doi.org/gpzmnf">https://doi.org/gpzmnf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/g3journal/jkac078">10.1093/g3journal/jkac078</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35385099">35385099</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9258556">PMC9258556</a></div></div>
</div>
<div id="ref-S1uQPBPG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline"><strong>A Critical Review of the Current Global Ex Situ Conservation System for Plant Agrobiodiversity. I. History of the Development of the Global System in the Context of the Political/Legal Framework and Its Major Conservation Components</strong> <div class="csl-block">Johannes MM Engels, Andreas W Ebert</div> <em>Plants</em> (2021-07-29) <a href="https://doi.org/gtq5n9">https://doi.org/gtq5n9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3390/plants10081557">10.3390/plants10081557</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34451602">34451602</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8401695">PMC8401695</a></div></div>
</div>
<div id="ref-USun3Z8l" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline"><strong>The Vulnerability of Plant Genetic Resources Conserved Ex Situ</strong> <div class="csl-block">Yong‐Bi Fu</div> <em>Crop Science</em> (2017-07-27) <a href="https://doi.org/gbzbnd">https://doi.org/gbzbnd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.2135/cropsci2017.01.0014">10.2135/cropsci2017.01.0014</a></div></div>
</div>
<div id="ref-6DjakjNS" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline"><strong>The FAIR Guiding Principles for scientific data management and stewardship</strong> <div class="csl-block">Mark D Wilkinson, Michel Dumontier, IJsbrand Jan Aalbersberg, Gabrielle Appleton, Myles Axton, Arie Baak, Niklas Blomberg, Jan-Willem Boiten, Luiz Bonino da Silva Santos, Philip E Bourne, … Barend Mons</div> <em>Scientific Data</em> (2016-03-15) <a href="https://doi.org/bdd4">https://doi.org/bdd4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/sdata.2016.18">10.1038/sdata.2016.18</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26978244">26978244</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4792175">PMC4792175</a></div></div>
</div>
<div id="ref-31NWAHsO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline"><strong>Recommendations for the formatting of Variant Call Format (VCF) files to make plant genotyping data FAIR</strong> <div class="csl-block">Sebastian Beier, Anne Fiebig, Cyril Pommier, Isuru Liyanage, Matthias Lange, Paul J Kersey, Stephan Weise, Richard Finkers, Baron Koylass, Timothee Cezard, … Uwe Scholz</div> <em>F1000Research</em> (2022-05-19) <a href="https://doi.org/gtq5n7">https://doi.org/gtq5n7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.12688/f1000research.109080.2">10.12688/f1000research.109080.2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35811804">35811804</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9218589">PMC9218589</a></div></div>
</div>
<div id="ref-6VGwG9za" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline"><strong>Genebank mining with FIGS, the Focused Identification of Germplasm Strategy</strong> <div class="csl-block">Ken Street, Ken Street</div> <em>Unknown</em> (2017) <a href="https://doi.org/gtq5n8">https://doi.org/gtq5n8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.22004/ag.econ.266624">10.22004/ag.econ.266624</a></div></div>
</div>
<div id="ref-pk5OwSG6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline"><strong>EURISCO update 2023: the European Search Catalogue for Plant Genetic Resources, a pillar for documentation of genebank material</strong> <div class="csl-block">Pragna Kotni, Theo van Hintum, Lorenzo Maggioni, Markus Oppermann, Stephan Weise</div> <em>Nucleic Acids Research</em> (2022-10-03) <a href="https://doi.org/gqxshq">https://doi.org/gqxshq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkac852">10.1093/nar/gkac852</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36189883">36189883</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9825528">PMC9825528</a></div></div>
</div>
<div id="ref-X22hwO1v" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline"><strong>Helium: visualization of large scale plant pedigrees</strong> <div class="csl-block">Paul D Shaw, Martin Graham, Jessie Kennedy, Iain Milne, David F Marshall</div> <em>BMC Bioinformatics</em> (2014-08-01) <a href="https://doi.org/f6gfzs">https://doi.org/f6gfzs</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1471-2105-15-259">10.1186/1471-2105-15-259</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25085009">25085009</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4133633">PMC4133633</a></div></div>
</div>
<div id="ref-1HmEHgW8h" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline"><strong>Unlocking the genetic diversity and population structure of the newly introduced two-row spring European HerItage Barley collecTion (ExHIBiT)</strong> <div class="csl-block">Villő Bernád, Nadia Al-Tamimi, Patrick Langan, Gary Gillespie, Timothy Dempsey, Joey Henchy, Mary Harty, Luke Ramsay, Kelly Houston, Malcolm Macaulay, … Sónia Negrão</div> <em>Frontiers in Plant Science</em> (2024-03-20) <a href="https://doi.org/gtstdn">https://doi.org/gtstdn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fpls.2024.1268847">10.3389/fpls.2024.1268847</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/38571708">38571708</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10987740">PMC10987740</a></div></div>
</div>
<div id="ref-x8DIc6zn" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline"><strong>Admixture and reproductive skew shape the conservation value of ex situ populations of the Critically Endangered eastern black rhino</strong> <div class="csl-block">Franziska Elsner-Gearing, Petra Kretzschmar, Susanne Shultz, Mark Pilgrim, Deborah Ann Dawson, Gavin John Horsburgh, Jírí Hruby, Jane Hopper, Tony King, Catherine Walton</div> <em>Conservation Genetics</em> (2024-03-22) <a href="https://doi.org/gtstdm">https://doi.org/gtstdm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s10592-024-01611-z">10.1007/s10592-024-01611-z</a></div></div>
</div>
<div id="ref-m7yuFGsd" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline"><strong>Applying FAIR Principles to Plant Phenotypic Data Management in GnpIS</strong> <div class="csl-block">C Pommier, C Michotey, G Cornut, P Roumet, E Duchêne, R Flores, A Lebreton, M Alaux, S Durand, E Kimmel, … AF Adam-Blondon</div> <em>Plant Phenomics</em> (2019-01) <a href="https://doi.org/gtq836">https://doi.org/gtq836</a> <div class="csl-block">DOI: <a href="https://doi.org/10.34133/2019/1671403">10.34133/2019/1671403</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33313522">33313522</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7718628">PMC7718628</a></div></div>
</div>
<div id="ref-15XwZ2Vb0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline"><strong>Mining Plant Genomic and Genetic Data Using the GnpIS Information System</strong> <div class="csl-block">A-F Adam-Blondon, M Alaux, S Durand, T Letellier, G Merceron, N Mohellibi, C Pommier, D Steinbach, F Alfama, J Amselem, … H Quesneville</div> <em>Methods in Molecular Biology</em> (2016-12-17) <a href="https://doi.org/gtq835">https://doi.org/gtq835</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/978-1-4939-6658-5_5">10.1007/978-1-4939-6658-5_5</a></div></div>
</div>
<div id="ref-1FDf25dDD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline"><strong>High density genotype storage for plant breeding in the Chado schema of Breedbase</strong> <div class="csl-block">Nicolas Morales, Guillaume J Bauchet, Titima Tantikanjana, Adrian F Powell, Bryan J Ellerbrock, Isaak Y Tecle, Lukas A Mueller</div> <em>PLOS ONE</em> (2020-11-11) <a href="https://doi.org/gmcmq7">https://doi.org/gmcmq7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pone.0240059">10.1371/journal.pone.0240059</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33175872">33175872</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7657515">PMC7657515</a></div></div>
</div>
<div id="ref-MlqG0w6i" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline"><strong>Field Book: An Open‐Source Application for Field Data Collection on Android</strong> <div class="csl-block">Trevor W Rife, Jesse A Poland</div> <em>Crop Science</em> (2014-07) <a href="https://doi.org/f584rr">https://doi.org/f584rr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.2135/cropsci2013.08.0579">10.2135/cropsci2013.08.0579</a></div></div>
</div>
<div id="ref-zuq8ri4R" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline"><strong>Prediction of Total Genetic Value Using Genome-Wide Dense Marker Maps</strong> <div class="csl-block">THE Meuwissen, BJ Hayes, ME Goddard</div> <em>Genetics</em> (2001-04-01) <a href="https://doi.org/gknztd">https://doi.org/gknztd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/genetics/157.4.1819">10.1093/genetics/157.4.1819</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/11290733">11290733</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1461589">PMC1461589</a></div></div>
</div>
<div id="ref-LCpCba21" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline"><strong>ClimMob: Software to support experimental citizen science in agriculture</strong> <div class="csl-block">Carlos Quirós, Kauê de Sousa, Jonathan Steinke, Brandon Madriz, Marie-Angélique Laporte, Elizabeth Arnaud, Rhys Manners, Berta Ortiz-Crespo, Anna Müller, Jacob van Etten</div> <em>Computers and Electronics in Agriculture</em> (2024-02) <a href="https://doi.org/gtq9h9">https://doi.org/gtq9h9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.compag.2023.108539">10.1016/j.compag.2023.108539</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/38343602">38343602</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10853689">PMC10853689</a></div></div>
</div>
<div id="ref-ArPSsaJQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline"><strong>The Breeding Information Management System (BIMS): an online resource for crop breeding</strong> <div class="csl-block">Sook Jung, Taein Lee, Ksenija Gasic, BTodd Campbell, Jing Yu, Jodi Humann, Sushan Ru, Daniel Edge-Garza, Heidi Hough, Dorrie Main</div> <em>Database</em> (2021-01-01) <a href="https://doi.org/gtj95x">https://doi.org/gtj95x</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/database/baab054">10.1093/database/baab054</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34415997">34415997</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8378516">PMC8378516</a></div></div>
</div>
<div id="ref-eaDrNpsp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline"><strong>15 years of GDR: New data and functionality in the Genome Database for Rosaceae</strong> <div class="csl-block">Sook Jung, Taein Lee, Chun-Huai Cheng, Katheryn Buble, Ping Zheng, Jing Yu, Jodi Humann, Stephen P Ficklin, Ksenija Gasic, Kristin Scott, … Dorrie Main</div> <em>Nucleic Acids Research</em> (2018-10-24) <a href="https://doi.org/gjqq8v">https://doi.org/gjqq8v</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gky1000">10.1093/nar/gky1000</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30357347">30357347</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6324069">PMC6324069</a></div></div>
</div>
<div id="ref-dcg0274X" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline"><strong>CottonGen: The Community Database for Cotton Genomics, Genetics, and Breeding Research</strong> <div class="csl-block">Jing Yu, Sook Jung, Chun-Huai Cheng, Taein Lee, Ping Zheng, Katheryn Buble, James Crabb, Jodi Humann, Heidi Hough, Don Jones, … Dorrie Main</div> <em>Plants</em> (2021-12-18) <a href="https://doi.org/gqqwfm">https://doi.org/gqqwfm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3390/plants10122805">10.3390/plants10122805</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34961276">34961276</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8705096">PMC8705096</a></div></div>
</div>
<div id="ref-sW9euzzP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline"><strong>From bits to bites: Advancement of the Germinate platform to support prebreeding informatics for crop wild relatives</strong> <div class="csl-block">Sebastian Raubach, Benjamin Kilian, Kate Dreher, Ahmed Amri, Filippo M Bassi, Ousmane Boukar, Douglas Cook, Alan Cruickshank, Christian Fatokun, Noureddine El Haddad, … </div> <em>Crop Science</em> (2020-08-20) <a href="https://doi.org/gm66th">https://doi.org/gm66th</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/csc2.20248">10.1002/csc2.20248</a></div></div>
</div>
<div id="ref-a9uPTJgO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline"><strong>icarda-git/QBMS: QBMS Version 1.0.0</strong> <div class="csl-block">Khaled Al-Shamaa, Johan Steven Aparicio, Nick, icarda-git</div> <em>Zenodo</em> (2024-03-07) <a href="https://doi.org/gtq85w">https://doi.org/gtq85w</a> <div class="csl-block">DOI: <a href="https://doi.org/10.5281/zenodo.10791627">10.5281/zenodo.10791627</a></div></div>
</div>
<div id="ref-7WXz7FUP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline"><strong>Mr.Bean: a comprehensive statistical and visualization application for modeling agricultural field trials data</strong> <div class="csl-block">Johan Aparicio, Salvador A Gezan, Daniel Ariza-Suarez, Bodo Raatz, Santiago Diaz, Ana Heilman-Morales, Juan Lobaton</div> <em>Frontiers in Plant Science</em> (2024-01-03) <a href="https://doi.org/gttbvb">https://doi.org/gttbvb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fpls.2023.1290078">10.3389/fpls.2023.1290078</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/38235208">38235208</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10792065">PMC10792065</a></div></div>
</div>
<div id="ref-kWLsm7g6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline"><strong>Enabling reusability of plant phenomic datasets with MIAPPE 1.1</strong> <div class="csl-block">Evangelia A Papoutsoglou, Daniel Faria, Daniel Arend, Elizabeth Arnaud, Ioannis N Athanasiadis, Inês Chaves, Frederik Coppens, Guillaume Cornut, Bruno V Costa, Hanna Ćwiek‐Kupczyńska, … Cyril Pommier</div> <em>New Phytologist</em> (2020-04-25) <a href="https://doi.org/gjqcmb">https://doi.org/gjqcmb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/nph.16544">10.1111/nph.16544</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32171029">32171029</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7317793">PMC7317793</a></div></div>
</div>
<div id="ref-KZc5ayvK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline"><strong>Toward interoperable bioscience data</strong> <div class="csl-block">Susanna-Assunta Sansone, Philippe Rocca-Serra, Dawn Field, Eamonn Maguire, Chris Taylor, Oliver Hofmann, Hong Fang, Steffen Neumann, Weida Tong, Linda Amaral-Zettler, … Winston Hide</div> <em>Nature Genetics</em> (2012-01-27) <a href="https://doi.org/fxp5gk">https://doi.org/fxp5gk</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ng.1054">10.1038/ng.1054</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22281772">22281772</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3428019">PMC3428019</a></div></div>
</div>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen()) ||
        heading.dataset.collapsed === "true"
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script on DOMContentLoaded instead of load to ensure this plugins
  // runs before other plugins
  window.addEventListener("DOMContentLoaded", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script type="module">
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
